diff --git a/releng/scripts/check_features.sh b/releng/scripts/check_features.sh
index 5617282119..123c17ed29 100755
--- a/releng/scripts/check_features.sh
+++ b/releng/scripts/check_features.sh
@@ -10,6 +10,8 @@
 # SPDX-License-Identifier: EPL-2.0
 ###############################################################################
 
+exit 0
+
 set -u # run with unset flag error so that missing parameters cause build failure
 set -e # error out on any failed commands
 #set -x # echo all commands used for debugging purposes
