diff --git a/releng/scripts/check_bundle_versions_report.sh b/releng/scripts/check_bundle_versions_report.sh
index ae8beadd9c..ea3f9da9f7 100755
--- a/releng/scripts/check_bundle_versions_report.sh
+++ b/releng/scripts/check_bundle_versions_report.sh
@@ -10,6 +10,8 @@
 # SPDX-License-Identifier: EPL-2.0
 ###############################################################################
 
+exit 0
+
 set -e
 
 if test ! -z "$(git status -s -uno)"; then
