diff --git a/gcc/gcc.cc b/gcc/gcc.cc
index d07a8e172a4..7e09aa926df 100644
--- a/gcc/gcc.cc
+++ b/gcc/gcc.cc
@@ -8243,6 +8243,63 @@ compare_files (char *cmpfile[])
   return ret;
 }
 
+static void
+add_env_var_paths_to_library_path(const char * env_var) {
+	const char * temp = env.get(env_var);
+	if (temp) {
+		size_t len = strlen(temp);
+		char paths[len + 10];
+		strcpy(paths, temp);
+		char path_separator = PATH_SEPARATOR;
+		size_t i = 0;
+		while (i < len) {
+			size_t path_start, path_end;
+			while (i < len && paths[i] == path_separator) {
+				i++;
+			}
+			// i == len || paths[i] != path_separator
+			path_start = i;
+			while (i < len && paths[i] != path_separator) {
+				i++;
+			}
+			// i == len || paths[i] == path_separator
+			path_end = i;
+
+			bool completed;
+			if (i < len) {
+				// paths[i] == path_separator
+				// see the separator
+				completed = false;
+				i++;
+			} else {
+				// i == len
+				// end of line \0
+				completed = true;
+			}
+
+			size_t path_len = path_end - path_start;
+			if (path_len > 0) {
+				paths[path_end] = 0;
+				char path[path_len + 10];
+				strcpy(path, paths + path_start);
+				path_len = convert_path(path, path_len);
+				(void)path_len;
+
+				add_prefix(&startfile_prefixes, path, NULL, PREFIX_PRIORITY_LAST, 0, 1);
+
+				if (!completed) {
+
+				} else {
+					break;
+				}
+			} else {
+				// empty path, ignore
+			}
+
+		}
+	}
+}
+
 driver::driver (bool can_finalize, bool debug) :
   explicit_link_files (NULL),
   decoded_options (NULL)
