diff --git a/gcc/config/mingw/t-cygming b/gcc/config/mingw/t-cygming
index fe0af545efc..68e992f0ebc 100644
--- a/gcc/config/mingw/t-cygming
+++ b/gcc/config/mingw/t-cygming
@@ -61,4 +61,3 @@ msformat-c.o: $(srcdir)/config/mingw/msformat-c.cc $(CONFIG_H) $(SYSTEM_H) \
 	$(COMPILER) -c $(ALL_COMPILERFLAGS) $(ALL_CPPFLAGS) $(INCLUDES) \
 	$(srcdir)/config/mingw/msformat-c.cc
 
-STMP_FIXINC=stmp-fixinc
