diff --git a/gcc/configure b/gcc/configure
index 05ac3aa97bc..84953ade834 100755
--- a/gcc/configure
+++ b/gcc/configure
@@ -34384,6 +34384,12 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $gcc_cv_no_pie" >&5
 $as_echo "$gcc_cv_no_pie" >&6; }
 
+case $host in
+*-cygwin* | *-mingw*)
+	true
+	;;
+*)
+
 if test x$enable_host_shared = xyes; then
   PICFLAG=-fPIC
 elif test x$enable_host_pie = xyes; then
@@ -34408,6 +34414,8 @@ if test x$enable_host_bind_now = xyes; then
   LD_PICFLAG="$LD_PICFLAG -Wl,-z,now"
 fi
 
+	;;
+esac
 
 
 # Enable Intel CET on Intel CET enabled host if jit is enabled.
diff --git a/gcc/configure.ac b/gcc/configure.ac
index 8b5dca36aa5..b451aff67de 100644
--- a/gcc/configure.ac
+++ b/gcc/configure.ac
@@ -7721,6 +7721,12 @@ AC_CACHE_CHECK([for -no-pie option],
      [gcc_cv_no_pie=no])
    LDFLAGS="$saved_LDFLAGS"])
 
+case $host in
+*-cygwin* | *-mingw*)
+	true
+	;;
+*)
+
 if test x$enable_host_shared = xyes; then
   PICFLAG=-fPIC
 elif test x$enable_host_pie = xyes; then
@@ -7731,8 +7737,6 @@ else
   PICFLAG=
 fi
 
-AC_SUBST([PICFLAG])
-
 if test x$enable_host_pie = xyes; then
   LD_PICFLAG=-pie
 elif test x$gcc_cv_no_pie = xyes; then
@@ -7745,6 +7749,12 @@ if test x$enable_host_bind_now = xyes; then
   LD_PICFLAG="$LD_PICFLAG -Wl,-z,now"
 fi
 
+	;;
+esac
+
+
+AC_SUBST([PICFLAG])
+
 AC_SUBST([LD_PICFLAG])
 
 # Enable Intel CET on Intel CET enabled host if jit is enabled.
