diff --git a/gcc/configure b/gcc/configure
index 84953ade834..32f039849a2 100755
--- a/gcc/configure
+++ b/gcc/configure
@@ -34322,6 +34322,12 @@ cat > gcc-driver-name.h <<EOF
 #define GCC_DRIVER_NAME "${target_noncanonical}-gcc-${gcc_driver_version}${exeext}"
 EOF
 
+case $host in
+*-cygwin* | *-mingw*)
+	true
+	;;
+*)
+
 # Check whether --enable-default-pie was given.
 # Check whether --enable-default-pie was given.
 if test "${enable_default_pie+set}" = set; then :
@@ -34336,6 +34342,9 @@ $as_echo "#define ENABLE_DEFAULT_PIE 1" >>confdefs.h
 
 fi
 
+	;;
+esac
+
 
 # Check if -fno-PIE works.
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for -fno-PIE option" >&5
diff --git a/gcc/configure.ac b/gcc/configure.ac
index b451aff67de..5f0e8bd5f36 100644
--- a/gcc/configure.ac
+++ b/gcc/configure.ac
@@ -7689,6 +7689,12 @@ cat > gcc-driver-name.h <<EOF
 #define GCC_DRIVER_NAME "${target_noncanonical}-gcc-${gcc_driver_version}${exeext}"
 EOF
 
+case $host in
+*-cygwin* | *-mingw*)
+	true
+	;;
+*)
+
 # Check whether --enable-default-pie was given.
 AC_ARG_ENABLE(default-pie,
 [AS_HELP_STRING([--enable-default-pie],
@@ -7699,6 +7705,9 @@ if test x$enable_default_pie = xyes ; then
   AC_DEFINE(ENABLE_DEFAULT_PIE, 1,
       [Define if your target supports default PIE and it is enabled.])
 fi
+
+	;;
+esac
 AC_SUBST([enable_default_pie])
 
 # Check if -fno-PIE works.
