diff --git a/gcc/configure b/gcc/configure
index b2db5a154ed..6ce6e4c17ac 100755
--- a/gcc/configure
+++ b/gcc/configure
@@ -32212,6 +32212,12 @@ if test "$gcc_cv_c_no_fpie" = "yes"; then
   NO_PIE_CFLAGS="-fno-PIE"
 fi
 
+case $host in
+  *-cygwin | *-mingw*)
+    NO_PIE_CFLAGS=
+    ;;
+esac
+
 
 # Check if -no-pie works.
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for -no-pie option" >&5
diff --git a/gcc/configure.ac b/gcc/configure.ac
index c7a253bd560..70e25eab70d 100644
--- a/gcc/configure.ac
+++ b/gcc/configure.ac
@@ -7524,6 +7524,13 @@ AC_CACHE_CHECK([for -fno-PIE option],
 if test "$gcc_cv_c_no_fpie" = "yes"; then
   NO_PIE_CFLAGS="-fno-PIE"
 fi
+
+case $host in
+  *-cygwin | *-mingw*)
+    NO_PIE_CFLAGS=
+    ;;
+esac
+
 AC_SUBST([NO_PIE_CFLAGS])
 
 # Check if -no-pie works.
