diff --git a/gcc/configure b/gcc/configure
index 23da7d55d62..eede4be1d8a 100755
--- a/gcc/configure
+++ b/gcc/configure
@@ -29033,32 +29033,6 @@ fi
     ;;
 
   i[34567]86-*-* | x86_64-*-*)
-    case $target_os in
-      cygwin*)
-	# Full C++ conformance when using a shared libstdc++-v3 requires some
-	# support from the Cygwin DLL, which in more recent versions exports
-	# wrappers to aid in interposing and redirecting operators new, delete,
-	# etc., as per n2800 #17.6.4.6 [replacement.functions].  Check if we
-	# are configuring for a version of Cygwin that exports the wrappers.
-	if test x$host = x$target && test x$host_cpu = xi686; then
-	  ac_fn_cxx_check_func "$LINENO" "__wrap__Znaj" "ac_cv_func___wrap__Znaj"
-if test "x$ac_cv_func___wrap__Znaj" = xyes; then :
-  gcc_ac_cygwin_dll_wrappers=yes
-else
-  gcc_ac_cygwin_dll_wrappers=no
-fi
-
-	else
-	  # Can't check presence of libc functions during cross-compile, so
-	  # we just have to assume we're building for an up-to-date target.
-	  gcc_ac_cygwin_dll_wrappers=yes
-	fi
-
-cat >>confdefs.h <<_ACEOF
-#define USE_CYGWIN_LIBSTDCXX_WRAPPERS `if test $gcc_ac_cygwin_dll_wrappers = yes; then echo 1; else echo 0; fi`
-_ACEOF
-
-    esac
     case $target_os in
       cygwin* | pe | mingw32*)
 	# Recent binutils allows the three-operand form of ".comm" on PE.  This
