diff --git a/gcc/incpath.cc b/gcc/incpath.cc
index 64cdd2f4a1b..d819b301f3c 100644
--- a/gcc/incpath.cc
+++ b/gcc/incpath.cc
@@ -502,6 +502,10 @@ register_include_chains (cpp_reader *pfile, const char *sysroot,
     add_standard_paths (sysroot, iprefix, imultilib, cxx_stdinc);
 
   target_c_incpath.extra_includes (sysroot, iprefix, stdinc);
+  
+#if defined(_WIN32) && !defined(CROSS_DIRECTORY_STRUCTURE)
+	add_env_var_paths ("INCLUDE", INC_AFTER);
+#endif
 
   merge_include_chains (sysroot, pfile, verbose);
 
