diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index 1bb9bcfe6aab..2d270f687f86 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -554,6 +554,30 @@ toolchains::MinGW::MinGW(const Driver &D, const llvm::Triple &Triple,
       getDriver().SysRoot.size())
     getFilePaths().push_back(Base + "lib");
 
+#ifdef __CYGWIN__
+  // for Cygwin's crt0.o
+  if (!::isCrossCompiling(getTriple(), /*RequireArchMatch=*/true) ||
+      getDriver().SysRoot.size()) {
+    getFilePaths().push_back(getDriver().SysRoot + "/lib");
+  } else {
+    // must add library directories by -L
+  }
+#else
+  // for MinGW's crt2.o
+  if (!::isCrossCompiling(getTriple(), /*RequireArchMatch=*/true)) {
+      const char * path = getenv("LIB");
+      if (path) {
+        getFilePaths().push_back(path);
+      } else {
+
+      }
+  } else if (getDriver().SysRoot.size()) {
+      getFilePaths().push_back(getDriver().SysRoot + "/lib");
+  } else {
+    // must add library directories by -L
+  }
+#endif
+
   NativeLLVMSupport =
       Args.getLastArgValue(options::OPT_fuse_ld_EQ, D.getPreferredLinker())
           .equals_insensitive("lld");
