diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index c81a7ed17029..d194ff1ac8d8 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -475,11 +517,17 @@ findClangRelativeSysroot(const Driver &D, const llvm::Triple &LiteralTriple,
 }
 
 static bool looksLikeMinGWSysroot(const std::string &Directory) {
-  StringRef Sep = llvm::sys::path::get_separator();
-  if (!llvm::sys::fs::exists(Directory + Sep + "include" + Sep + "_mingw.h"))
+#ifdef __CYGWIN__
+  if (!llvm::sys::fs::exists(Directory + "/usr/include/cygwin/version.h"))
+    return false;
+  if (!llvm::sys::fs::exists(Directory + "/lib/libcygwin.a"))
     return false;
-  if (!llvm::sys::fs::exists(Directory + Sep + "lib" + Sep + "libkernel32.a"))
+#else
+  if (!llvm::sys::fs::exists(Directory + "/include/_mingw.h"))
     return false;
+  if (!llvm::sys::fs::exists(Directory + "/lib/libkernel32.a"))
+    return false;
+#endif
   return true;
 }
 
