diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index c81a7ed17029..d194ff1ac8d8 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -83,6 +85,9 @@ void tools::MinGW::Linker::AddLibGCC(const ArgList &Args,
     AddRunTimeLibs(getToolChain(), getToolChain().getDriver(), CmdArgs, Args);
   }
 
+#ifdef __CYGWIN__
+  CmdArgs.push_back("-lcygwin");
+#else
   CmdArgs.push_back("-lmoldname");
   CmdArgs.push_back("-lmingwex");
   for (auto Lib : Args.getAllArgValues(options::OPT_l))
@@ -91,6 +96,7 @@ void tools::MinGW::Linker::AddLibGCC(const ArgList &Args,
         StringRef(Lib).starts_with("crtdll"))
       return;
   CmdArgs.push_back("-lmsvcrt");
+#endif
 }
 
 void tools::MinGW::Linker::ConstructJob(Compilation &C, const JobAction &JA,
