diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index c81a7ed17029..d194ff1ac8d8 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -162,12 +168,31 @@ void tools::MinGW::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back("-Bdynamic");
   if (Args.hasArg(options::OPT_mdll) || Args.hasArg(options::OPT_shared)) {
     CmdArgs.push_back("-e");
+#ifdef __CYGWIN__
+    if (TC.getArch() == llvm::Triple::x86)
+      CmdArgs.push_back("__cygwin_dll_entry@12");
+    else
+      CmdArgs.push_back("_cygwin_dll_entry");
+#else
     if (TC.getArch() == llvm::Triple::x86)
       CmdArgs.push_back("_DllMainCRTStartup@12");
     else
       CmdArgs.push_back("DllMainCRTStartup");
+#endif
     CmdArgs.push_back("--enable-auto-image-base");
+  } else {
+#ifndef __CYGWIN__
+    if (Args.hasArg(options::OPT_no_pie)) {
+      CmdArgs.push_back("--disable-dynamicbase");
+    }
+#endif
+    if (TC.getArch() == llvm::Triple::x86) {
+      CmdArgs.push_back("--large-address-aware");
+    }
   }
+#ifdef __CYGWIN__
+  CmdArgs.push_back("--dynamicbase");
+#endif
 
   if (Args.hasArg(options::OPT_Z_Xlinker__no_demangle))
     CmdArgs.push_back("--no-demangle");
