diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index c81a7ed17029..d194ff1ac8d8 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -371,8 +409,13 @@ void tools::MinGW::Linker::ConstructJob(Compilation &C, const JobAction &JA,
 
 static bool isCrossCompiling(const llvm::Triple &T, bool RequireArchMatch) {
   llvm::Triple HostTriple(llvm::Triple::normalize(LLVM_HOST_TRIPLE));
+#ifdef __CYGWIN__
+  if (!HostTriple.isWindowsCygwinEnvironment())
+    return true;
+#else
   if (HostTriple.getOS() != llvm::Triple::Win32)
     return true;
+#endif
   if (RequireArchMatch && HostTriple.getArch() != T.getArch())
     return true;
   return false;
