diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index c81a7ed17029..bafa3f7475cf 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -455,14 +503,15 @@ static llvm::ErrorOr<std::string> findGcc(const llvm::Triple &LiteralTriple,
 
 static llvm::ErrorOr<std::string>
 findClangRelativeSysroot(const Driver &D, const llvm::Triple &LiteralTriple,
-                         const llvm::Triple &T, std::string &SubdirName) {
+                         const llvm::Triple &T, std::string &SubdirName, bool isCygwin) {
   llvm::SmallVector<llvm::SmallString<32>, 4> Subdirs;
-  Subdirs.emplace_back(LiteralTriple.str());
   Subdirs.emplace_back(T.str());
   Subdirs.emplace_back(T.getArchName());
-  Subdirs.back() += "-w64-mingw32";
-  Subdirs.emplace_back(T.getArchName());
-  Subdirs.back() += "-w64-mingw32ucrt";
+  if (isCygwin) {
+    Subdirs.back() += "-pc-cygwin";
+  } else {
+    Subdirs.back() += "-pc-mingw64";
+  }
   StringRef ClangRoot = llvm::sys::path::parent_path(D.Dir);
   StringRef Sep = llvm::sys::path::get_separator();
   for (StringRef CandidateSubdir : Subdirs) {
