diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index 067758c05e97..11f31cbcffd6 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -471,19 +471,25 @@ findClangRelativeSysroot(const Driver &D, const llvm::Triple &LiteralTriple,
   return make_error_code(std::errc::no_such_file_or_directory);
 }
 
-static bool looksLikeMinGWSysroot(const std::string &Directory) {
-  StringRef Sep = llvm::sys::path::get_separator();
-  if (!llvm::sys::fs::exists(Directory + Sep + "include" + Sep + "_mingw.h"))
-    return false;
-  if (!llvm::sys::fs::exists(Directory + Sep + "lib" + Sep + "libkernel32.a"))
-    return false;
+static bool looksLikeMinGWSysroot(const std::string &Directory, bool isCygwin) {
+  if (isCygwin) {
+    if (!llvm::sys::fs::exists(Directory + "/usr/include/cygwin/version.h"))
+      return false;
+    if (!llvm::sys::fs::exists(Directory + "/lib/libcygwin.a"))
+      return false;
+  } else {
+    if (!llvm::sys::fs::exists(Directory + "/include/_mingw.h"))
+      return false;
+    if (!llvm::sys::fs::exists(Directory + "/lib/libkernel32.a"))
+      return false;
+  }
   return true;
 }
 
 toolchains::MinGW::MinGW(const Driver &D, const llvm::Triple &Triple,
-                         const ArgList &Args)
+                         const ArgList &Args, bool isCygwin)
     : ToolChain(D, Triple, Args), CudaInstallation(D, Triple, Args),
-      RocmInstallation(D, Triple, Args) {
+      RocmInstallation(D, Triple, Args), isCygwin(isCygwin) {
   getProgramPaths().push_back(getDriver().getInstalledDir());
 
   std::string InstallBase =
