diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index c81a7ed17029..bafa3f7475cf 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -822,13 +857,13 @@ static bool testTriple(const Driver &D, const llvm::Triple &Triple,
   llvm::Triple LiteralTriple = getLiteralTriple(D, Triple);
   std::string InstallBase = std::string(llvm::sys::path::parent_path(D.Dir));
   if (llvm::ErrorOr<std::string> TargetSubdir =
-          findClangRelativeSysroot(D, LiteralTriple, Triple, SubdirName))
+          findClangRelativeSysroot(D, LiteralTriple, Triple, SubdirName, isCygwin))
     return true;
   // If the install base itself looks like a mingw sysroot, we'll use that
   // - don't use any potentially unrelated gcc to influence what triple to use.
-  if (looksLikeMinGWSysroot(InstallBase))
+  if (looksLikeMinGWSysroot(InstallBase, isCygwin))
     return false;
-  if (llvm::ErrorOr<std::string> GPPName = findGcc(LiteralTriple, Triple))
+  if (llvm::ErrorOr<std::string> GPPName = findGcc(LiteralTriple, Triple, isCygwin))
     return true;
   // If we neither found a colocated sysroot or a matching gcc executable,
   // conclude that we can't know if this is the correct spelling of the triple.
