diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index c81a7ed17029..bafa3f7475cf 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -855,7 +890,7 @@ static llvm::Triple adjustTriple(const Driver &D, const llvm::Triple &Triple,
   for (auto A : Archs) {
     llvm::Triple TestTriple(Triple);
     TestTriple.setArchName(A);
-    if (testTriple(D, TestTriple, Args))
+    if (testTriple(D, TestTriple, Args, isCygwin))
       return TestTriple;
   }
   // If none was found, just proceed with the original value.
