diff --git a/clang/lib/Driver/ToolChains/MinGW.h b/clang/lib/Driver/ToolChains/MinGW.h
index a9963d8d06c2..7754a9a0a0b1 100644
--- a/clang/lib/Driver/ToolChains/MinGW.h
+++ b/clang/lib/Driver/ToolChains/MinGW.h
@@ -59,10 +59,10 @@ namespace toolchains {
 class LLVM_LIBRARY_VISIBILITY MinGW : public ToolChain {
 public:
   MinGW(const Driver &D, const llvm::Triple &Triple,
-        const llvm::opt::ArgList &Args);
+        const llvm::opt::ArgList &Args, bool isCygwin);
 
   static void fixTripleArch(const Driver &D, llvm::Triple &Triple,
-                            const llvm::opt::ArgList &Args);
+                            const llvm::opt::ArgList &Args, bool isCygwin);
 
   bool HasNativeLLVMSupport() const override;
 
@@ -117,6 +117,8 @@ private:
   void findGccLibDir(const llvm::Triple &LiteralTriple);
 
   bool NativeLLVMSupport;
+public:
+  bool isCygwin;
 };
 
 } // end namespace toolchains
