diff --git a/fixincludes/configure b/fixincludes/configure
index 662c94dc112..a1404d6ab29 100755
--- a/fixincludes/configure
+++ b/fixincludes/configure
@@ -4839,12 +4839,21 @@ $as_echo "#define SEPARATE_FIX_PROC 1" >>confdefs.h
 
 fi
 
+case $host in
+  *-cygwin | *-mingw*)
+    true
+    ;;
+  *)
+
 # Enable --enable-host-pie.
 # Check whether --enable-host-pie was given.
 if test "${enable_host_pie+set}" = set; then :
   enableval=$enable_host_pie; PICFLAG=-fPIE; LD_PICFLAG=-pie
 fi
 
+    ;;
+esac
+
 
 
 
diff --git a/fixincludes/configure.ac b/fixincludes/configure.ac
index 4e78511d20f..fa91fc38d7f 100644
--- a/fixincludes/configure.ac
+++ b/fixincludes/configure.ac
@@ -68,11 +68,21 @@ if test $TARGET = twoprocess; then
 		  [Define if testing and fixing are done by separate process])
 fi
 
+case $host in
+  *-cygwin | *-mingw*)
+    true
+    ;;
+  *)
+
 # Enable --enable-host-pie.
 AC_ARG_ENABLE(host-pie,
 [AS_HELP_STRING([--enable-host-pie],
        [build host code as PIE])],
 [PICFLAG=-fPIE; LD_PICFLAG=-pie], [])
+
+    ;;
+esac
+
 AC_SUBST(PICFLAG)
 AC_SUBST(LD_PICFLAG)
 
