diff --git a/sim/erc32/func.c b/sim/erc32/func.c
index 972a636e619..80f416d87d7 100644
--- a/sim/erc32/func.c
+++ b/sim/erc32/func.c
@@ -697,7 +697,7 @@ init_signals(void)
 extern struct disassemble_info dinfo;
 
 struct estate   ebase;
-struct evcell   evbuf[EVENT_MAX];
+struct evcell   evbuf[MAX_EVENTS];
 struct irqcell  irqarr[16];
 
 static int
@@ -864,10 +864,10 @@ init_event(void)
 
     ebase.eq.nxt = NULL;
     ebase.freeq = evbuf;
-    for (i = 0; i < EVENT_MAX; i++) {
+    for (i = 0; i < MAX_EVENTS; i++) {
 	evbuf[i].nxt = &evbuf[i + 1];
     }
-    evbuf[EVENT_MAX - 1].nxt = NULL;
+    evbuf[MAX_EVENTS - 1].nxt = NULL;
 }
 
 void
diff --git a/sim/erc32/sis.h b/sim/erc32/sis.h
index 504d7e69c77..cf44ddfb7fe 100644
--- a/sim/erc32/sis.h
+++ b/sim/erc32/sis.h
@@ -31,7 +31,7 @@
 #define I_ACC_EXC 1
 
 /* Maximum events in event queue */
-#define EVENT_MAX	256
+#define MAX_EVENTS	256
 
 /* Maximum # of floating point queue */
 #define FPUQN	1
