diff --git a/sim/ppc/hw_sem.c b/sim/ppc/hw_sem.c
index 9495b8591b9..a1b167eb20b 100644
--- a/sim/ppc/hw_sem.c
+++ b/sim/ppc/hw_sem.c
@@ -24,8 +24,10 @@
 #include "device_table.h"
 
 #include <string.h>
+#if !defined(_WIN32) || defined(__CYGWIN__)
 #include <sys/ipc.h>
 #include <sys/sem.h>
+#endif
 
 #include <errno.h>
 
diff --git a/sim/ppc/hw_shm.c b/sim/ppc/hw_shm.c
index 0f78ae8c4a9..03c162cb70b 100644
--- a/sim/ppc/hw_shm.c
+++ b/sim/ppc/hw_shm.c
@@ -24,8 +24,10 @@
 #include "device_table.h"
 
 #include <string.h>
+#if !defined(_WIN32) || defined(__CYGWIN__)
 #include <sys/ipc.h>
 #include <sys/shm.h>
+#endif
 
 
 /* DEVICE
