--- origsrc/Python-3.8.7/Python/dynload_shlib.c	2020-12-21 17:25:24.000000000 +0100
+++ src/Python-3.8.7/Python/dynload_shlib.c	2021-01-12 14:49:06.902341400 +0100
@@ -35,6 +35,8 @@
 
 const char *_PyImport_DynLoadFiletab[] = {
 #ifdef __CYGWIN__
+    "." SOABI ".dll",
+    ".abi" PYTHON_ABI_STRING ".dll",
     ".dll",
 #else  /* !__CYGWIN__ */
     "." SOABI ".so",
--- origsrc/Python-3.8.7/configure.ac	2021-01-12 11:57:01.476420200 +0100
+++ src/Python-3.8.7/configure.ac	2021-01-12 14:49:06.961841600 +0100
@@ -845,6 +845,14 @@ cat >> conftest.c <<EOF
         darwin
 #elif defined(__VXWORKS__)
         vxworks
+#elif defined(__CYGWIN__)
+# if defined(__x86_64__)
+        x86_64-cygwin
+# elif defined(__i386__)
+        i386-cygwin
+# else
+#   error unknown platform triplet
+# endif
 #else
 # error unknown platform triplet
 #endif
