diff --git a/builtin/index-pack.c b/builtin/index-pack.c
index f176dd28c8..d4d8e70bf2 100644
--- a/builtin/index-pack.c
+++ b/builtin/index-pack.c
@@ -42,7 +42,7 @@ struct base_data {
 	int ofs_first, ofs_last;
 };
 
-struct thread_local {
+struct thread_local_data {
 	pthread_t thread;
 	struct base_data *base_cache;
 	size_t base_cache_used;
@@ -67,7 +67,7 @@ static struct object_entry *objects;
 static struct object_stat *obj_stat;
 static struct ofs_delta_entry *ofs_deltas;
 static struct ref_delta_entry *ref_deltas;
-static struct thread_local nothread_data;
+static struct thread_local_data nothread_data;
 static int nr_objects;
 static int nr_ofs_deltas;
 static int nr_ref_deltas;
@@ -97,7 +97,7 @@ static uint32_t input_crc32;
 static int input_fd, output_fd;
 static const char *curr_pack;
 
-static struct thread_local *thread_data;
+static struct thread_local_data *thread_data;
 static int nr_dispatched;
 static int threads_active;
 
@@ -347,7 +347,7 @@ static NORETURN void bad_object(off_t offset, const char *format, ...)
 	    (uintmax_t)offset, buf);
 }
 
-static inline struct thread_local *get_thread_data(void)
+static inline struct thread_local_data *get_thread_data(void)
 {
 	if (HAVE_THREADS) {
 		if (threads_active)
@@ -358,7 +358,7 @@ static inline struct thread_local *get_thread_data(void)
 	return &nothread_data;
 }
 
-static void set_thread_data(struct thread_local *data)
+static void set_thread_data(struct thread_local_data *data)
 {
 	if (threads_active)
 		pthread_setspecific(key, data);
@@ -383,7 +383,7 @@ static void free_base_data(struct base_data *c)
 static void prune_base_data(struct base_data *retain)
 {
 	struct base_data *b;
-	struct thread_local *data = get_thread_data();
+	struct thread_local_data *data = get_thread_data();
 	for (b = data->base_cache;
 	     data->base_cache_used > delta_base_cache_limit && b;
 	     b = b->child) {
