diff --git a/gcc/config.in b/gcc/config.in
index 7fcabbe5061..714814cf576 100644
--- a/gcc/config.in
+++ b/gcc/config.in
@@ -2564,13 +2564,6 @@
 #endif
 
 
-/* Define if you want to generate code by default that assumes that the Cygwin
-   DLL exports wrappers to support libstdc++ function replacement. */
-#ifndef USED_FOR_TARGET
-#undef USE_CYGWIN_LIBSTDCXX_WRAPPERS
-#endif
-
-
 /* Define to 1 if the 'long long' type is wider than 'long' but still
    efficiently supported by the host hardware. */
 #ifndef USED_FOR_TARGET
