diff --git a/clang/lib/CodeGen/Address.h b/clang/lib/CodeGen/Address.h
index bddeac1d6dcb..af872689be40 100644
--- a/clang/lib/CodeGen/Address.h
+++ b/clang/lib/CodeGen/Address.h
@@ -46,6 +46,14 @@ template <typename T> class AddressImpl<T, true> {
   llvm::PointerIntPair<llvm::Value *, 3, unsigned> Pointer;
   // Int portion stores lower 3 bits of the log of the alignment.
   llvm::PointerIntPair<llvm::Type *, 3, unsigned> ElementType;
+  union {
+    unsigned AlignLog1;
+    uintptr_t pad1;
+  };
+  union {
+    bool IsKnownNonNull1;
+    uintptr_t pad2;
+  };
 
 public:
   AddressImpl(llvm::Value *Pointer, llvm::Type *ElementType,
