diff --git a/clang/lib/CodeGen/CGCall.h b/clang/lib/CodeGen/CGCall.h
index aee86a3242fd..5c181b8c0086 100644
--- a/clang/lib/CodeGen/CGCall.h
+++ b/clang/lib/CodeGen/CGCall.h
@@ -254,7 +254,7 @@ public:
 
 /// CallArgList - Type for representing both the value and type of
 /// arguments in a call.
-class CallArgList : public SmallVector<CallArg, 8> {
+class CallArgList : public SmallVector<CallArg, 2> {
 public:
   CallArgList() = default;
 
