diff --git a/gcc/gcc.cc b/gcc/gcc.cc
index d07a8e172a4..296500f80d3 100644
--- a/gcc/gcc.cc
+++ b/gcc/gcc.cc
@@ -3149,13 +3149,22 @@ add_prefix (struct path_prefix *pprefix, const char *prefix,
 
   /* Keep track of the longest prefix.  */
 
-  prefix = update_path (prefix, component);
-  len = strlen (prefix);
+  char * prefix2 = update_path (prefix, component);
+  len = strlen (prefix2);
+  len = convert_path(prefix2, len);
+  char prefix3[len + 10];
+  strcpy(prefix3, prefix2);
+  free(prefix2);
+  prefix2 = 0;
+  if (len > 0 && prefix3[len - 1] != '/') {
+    prefix3[len++] = '/';
+  }
+  prefix3[len] = 0;
   if (len > pprefix->max_len)
     pprefix->max_len = len;
 
   pl = XNEW (struct prefix_list);
-  pl->prefix = prefix;
+  pl->prefix = xstrdup(prefix3);
   pl->require_machine_suffix = require_machine_suffix;
   pl->priority = priority;
   pl->os_multilib = os_multilib;
@@ -4930,6 +4939,57 @@ process_command (unsigned int decoded_options_count,
 	}
     }
 
+#if defined(_WIN32) && !defined(__CYGWIN__) && !defined(CROSS_COMPILE)
+	temp = env.get ("LIB");
+	if (temp) {
+		char * paths = xstrdup(temp);
+		size_t len = strlen(paths);
+		char path_separator = PATH_SEPARATOR;
+		size_t i = 0;
+		while (i < len) {
+			size_t path_start, path_end;
+			path_start = i;
+			while (i < len && paths[i] != path_separator) {
+				i++;
+			}
+			// i == len || paths[i] == path_separator
+			path_end = i;
+
+			bool completed;
+			if (i < len) {
+				// paths[i] == path_separator
+				// see the separator
+				completed = false;
+				i++;
+			} else {
+				// i == len
+				// end of line \0
+				completed = true;
+			}
+
+			size_t path_len = path_end > path_start;
+			if (path_len > 0) {
+				paths[path_end] = 0;
+
+				add_prefix (&startfile_prefixes, paths + path_start, NULL, PREFIX_PRIORITY_LAST, 0, 1);
+
+				if (!completed) {
+
+				} else {
+					break;
+				}
+			} else {
+				// empty path, ignore
+			}
+
+			while (i < len && paths[i] == path_separator) {
+				i++;
+			}
+			// i == len || paths[i] != path_separator
+		}
+	}
+#endif
+
   /* Use LPATH like LIBRARY_PATH (for the CMU build program).  */
   temp = env.get ("LPATH");
   if (temp && *cross_compile == '0')
