diff --git a/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/Command.java b/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/Command.java
index f15a0dd7af..4a27475a3d 100644
--- a/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/Command.java
+++ b/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/Command.java
@@ -17,6 +17,7 @@ import java.io.File;
 import java.io.IOException;
 
 import org.eclipse.cdt.make.core.makefile.ICommand;
+import org.eclipse.cdt.utils.spawner.ProcessFactory;
 
 /**
  * Makefile : ( statement ) *
@@ -122,7 +123,7 @@ public class Command extends Directive implements ICommand {
 	@Override
 	public Process execute(String shell, String[] envp, File dir) throws IOException {
 		String[] cmdArray = new String[] { shell, "-c", command }; //$NON-NLS-1$
-		return Runtime.getRuntime().exec(cmdArray, envp, dir);
+		return ProcessFactory.getFactory().exec(cmdArray, envp, dir);
 	}
 
 }
