diff --git a/build/org.eclipse.cdt.autotools.core/src/org/eclipse/cdt/internal/autotools/core/PkgconfigErrorResolution.java b/build/org.eclipse.cdt.autotools.core/src/org/eclipse/cdt/internal/autotools/core/PkgconfigErrorResolution.java
index 624b401a1d..2f3027cea5 100644
--- a/build/org.eclipse.cdt.autotools.core/src/org/eclipse/cdt/internal/autotools/core/PkgconfigErrorResolution.java
+++ b/build/org.eclipse.cdt.autotools.core/src/org/eclipse/cdt/internal/autotools/core/PkgconfigErrorResolution.java
@@ -93,7 +93,7 @@ public class PkgconfigErrorResolution implements IMarkerResolution {
 				"--queryformat", //$NON-NLS-1$
 				"%{NAME}", //$NON-NLS-1$
 				"--whatprovides", //$NON-NLS-1$
-				pkgconfigPath.toOSString() };
+				pkgconfigPath.toString() };
 		try {
 			// Use CDT launcher to run rpm to query the package that provides
 			// the pkgconfig .pc file for the package in question.
