diff --git a/build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/internal/ui/language/settings/providers/CompilationDatabaseParserOptionPage.java b/build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/internal/ui/language/settings/providers/CompilationDatabaseParserOptionPage.java
index 4063eaaca2..b9fca5e04b 100644
--- a/build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/internal/ui/language/settings/providers/CompilationDatabaseParserOptionPage.java
+++ b/build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/internal/ui/language/settings/providers/CompilationDatabaseParserOptionPage.java
@@ -129,7 +129,7 @@ public final class CompilationDatabaseParserOptionPage extends AbstractLanguageS
 				dialog.setText(Messages.CompilationDatabaseParserOptionPage_ChooseFile);
 				String fileName = fCompileCommandsPath.getText();
 				IPath folder = new Path(fileName).removeLastSegments(1);
-				dialog.setFilterPath(folder.toOSString());
+				dialog.setFilterPath(folder.toString());
 				String chosenFile = dialog.open();
 				if (chosenFile != null) {
 					fCompileCommandsPath.setText(chosenFile);
