diff --git a/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/SetConfigurationParameterTest.java b/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/SetConfigurationParameterTest.java
index 2cc4a0af76..edbef116ca 100644
--- a/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/SetConfigurationParameterTest.java
+++ b/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/SetConfigurationParameterTest.java
@@ -105,7 +105,7 @@ public class SetConfigurationParameterTest extends AbstractTest {
 		assertNotNull(project);
 		IPath path = project.getLocation();
 		path = path.append(".autotools");
-		File f = new File(path.toOSString());
+		File f = new File(path.toString());
 		assertTrue(f.exists());
 
 		DocumentBuilder db = XmlProcessorFactoryCdt.createDocumentBuilderWithErrorOnDOCTYPE();
@@ -155,9 +155,9 @@ public class SetConfigurationParameterTest extends AbstractTest {
 		IProject project = root.getProject(projectName);
 		assertNotNull(project);
 		IPath path = project.getLocation();
-		File f = new File(path.append("src/a.out").toOSString());
+		File f = new File(path.append("src/a.out").toString());
 		assertTrue(f.exists());
-		f = new File(path.append("config.status").toOSString());
+		f = new File(path.append("config.status").toString());
 		assertTrue(f.exists());
 
 		try (BufferedReader r = new BufferedReader(new FileReader(f))) {
@@ -202,13 +202,13 @@ public class SetConfigurationParameterTest extends AbstractTest {
 		// sleep a bit and look for it...give up after 120 seconds
 		for (int i = 0; i < 240; ++i) {
 			bot.sleep(500);
-			f = new File(path.append("src/a.out").toOSString());
+			f = new File(path.append("src/a.out").toString());
 			if (f.exists()) {
 				break;
 			}
 		}
 		assertTrue(f.exists());
-		f = new File(path.append("config.status").toOSString());
+		f = new File(path.append("config.status").toString());
 		assertTrue(f.exists());
 		long newDate = f.lastModified();
 		assertEquals(newDate, oldDate);
@@ -278,13 +278,13 @@ public class SetConfigurationParameterTest extends AbstractTest {
 		File f = null;
 		for (int i = 0; i < 40; ++i) {
 			bot.sleep(500);
-			f = new File(path.append("build-debug/src/a.out").toOSString());
+			f = new File(path.append("build-debug/src/a.out").toString());
 			if (f.exists()) {
 				break;
 			}
 		}
 		assertTrue(f.exists());
-		f = new File(path.append("build-debug/config.status").toOSString());
+		f = new File(path.append("build-debug/config.status").toString());
 		assertTrue(f.exists());
 		try (BufferedReader r = new BufferedReader(new FileReader(f))) {
 			int ch;
@@ -314,10 +314,10 @@ public class SetConfigurationParameterTest extends AbstractTest {
 		// Verify we cleaned out the top-level build directory (i.e. that there
 		// is no config.status there anymore).
 		path = project.getLocation().append("config.status");
-		f = new File(path.toOSString());
+		f = new File(path.toString());
 		assertTrue(!f.exists());
 		path = project.getLocation().append(".autotools");
-		f = new File(path.toOSString());
+		f = new File(path.toString());
 		assertTrue(f.exists());
 		DocumentBuilder db = XmlProcessorFactoryCdt.createDocumentBuilderWithErrorOnDOCTYPE();
 		Document d = db.parse(f);
