diff --git a/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/TestMakeTargets.java b/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/TestMakeTargets.java
index bb89f85a90..3fcf566c51 100644
--- a/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/TestMakeTargets.java
+++ b/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/TestMakeTargets.java
@@ -56,9 +56,9 @@ public class TestMakeTargets extends AbstractTest {
 		assertNotNull(project);
 		IPath path = project.getLocation();
 		path = path.append("config.status");
-		File f = new File(path.toOSString());
+		File f = new File(path.toString());
 		assertTrue(f.exists());
-		f = new File(path.toOSString());
+		f = new File(path.toString());
 		assertTrue(f.exists());
 
 		projectExplorer.bot().tree().getTreeItem(projectName).select();
