diff --git a/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/TestEnvironmentVars.java b/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/TestEnvironmentVars.java
index d5dca2a894..8a54cca9c8 100644
--- a/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/TestEnvironmentVars.java
+++ b/build/org.eclipse.cdt.autotools.ui.tests/src/org/eclipse/cdt/autotools/ui/tests/TestEnvironmentVars.java
@@ -60,7 +60,7 @@ public class TestEnvironmentVars extends AbstractTest {
 		// sleep a bit and look for it...give up after 40 seconds
 		for (int i = 0; i < 80; ++i) {
 			bot.sleep(500);
-			f = new File(path.append("config.status").toOSString());
+			f = new File(path.append("config.status").toString());
 			if (f.exists()) {
 				break;
 			}
@@ -104,7 +104,7 @@ public class TestEnvironmentVars extends AbstractTest {
 		// sleep a bit and look for it...give up after 40 seconds
 		for (int i = 0; i < 80; ++i) {
 			bot.sleep(500);
-			f = new File(path.append("config.status").toOSString());
+			f = new File(path.append("config.status").toString());
 			if (f.exists()) {
 				break;
 			}
@@ -124,7 +124,7 @@ public class TestEnvironmentVars extends AbstractTest {
 		IPath path = checkProject().getLocation();
 		// Create a fake configure script which prints out the values of
 		// envvars some_var1, some_var2, and some_var3
-		File f = new File(path.append("fake_configure").toOSString());
+		File f = new File(path.append("fake_configure").toString());
 		try (BufferedWriter w = new BufferedWriter(new FileWriter(f))) {
 			w.append("echo VAR1 is ${some_var1}");
 			w.newLine();
