diff --git a/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/language/settings/providers/AbstractBuiltinSpecsDetector.java b/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/language/settings/providers/AbstractBuiltinSpecsDetector.java
index cdac186a77..c800e19f3e 100644
--- a/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/language/settings/providers/AbstractBuiltinSpecsDetector.java
+++ b/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/language/settings/providers/AbstractBuiltinSpecsDetector.java
@@ -877,7 +877,7 @@ public abstract class AbstractBuiltinSpecsDetector extends AbstractLanguageSetti
 		IPath workingLocation = ManagedBuilderCorePlugin.getDefault().getStateLocation();
 		IPath fileLocation = workingLocation.append(specFileName);
 
-		specFile = new java.io.File(fileLocation.toOSString());
+		specFile = new java.io.File(fileLocation.toString());
 		// will preserve spec file if it was already there otherwise will delete upon finishing
 		preserveSpecFile = specFile.exists();
 		if (!preserveSpecFile) {
