diff --git a/build/org.eclipse.cdt.make.core/src/org/eclipse/cdt/make/internal/core/scannerconfig/gnu/GCCScannerInfoConsoleParser.java b/build/org.eclipse.cdt.make.core/src/org/eclipse/cdt/make/internal/core/scannerconfig/gnu/GCCScannerInfoConsoleParser.java
index ee008d9357..51d67a3eef 100644
--- a/build/org.eclipse.cdt.make.core/src/org/eclipse/cdt/make/internal/core/scannerconfig/gnu/GCCScannerInfoConsoleParser.java
+++ b/build/org.eclipse.cdt.make.core/src/org/eclipse/cdt/make/internal/core/scannerconfig/gnu/GCCScannerInfoConsoleParser.java
@@ -157,7 +157,7 @@ public class GCCScannerInfoConsoleParser extends AbstractGCCBOPConsoleParser {
 		}
 
 		if (fileName != null && fileName.startsWith("/cygdrive/")) { //$NON-NLS-1$
-			fileName = AbstractGCCBOPConsoleParserUtility.convertCygpath(new Path(fileName)).toOSString();
+			fileName = AbstractGCCBOPConsoleParserUtility.convertCygpath(new Path(fileName)).toString();
 		}
 		if (fileName == null || fileName.trim().length() == 0) {
 			return false; // return when no file was given (analogous to GCCPerFileBOPConsoleParser)
