diff --git a/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/AutomakeDocumentProvider.java b/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/AutomakeDocumentProvider.java
index 1b693072f5..e874861d06 100644
--- a/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/AutomakeDocumentProvider.java
+++ b/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/AutomakeDocumentProvider.java
@@ -46,7 +46,7 @@ public class AutomakeDocumentProvider extends TextFileDocumentProvider implement
 		if (element instanceof IFileEditorInput) {
 			IFileEditorInput input = (IFileEditorInput) element;
 			if (input.getFile().exists())
-				original = createMakefile(input.getFile().getLocation().toOSString());
+				original = createMakefile(input.getFile().getLocation().toString());
 		} else if (element instanceof IURIEditorInput) {
 			IURIEditorInput input = (IURIEditorInput) element;
 			original = createMakefile(input.getURI().getPath().toString());
