diff --git a/build/org.eclipse.cdt.meson.ui/src/org/eclipse/cdt/meson/ui/properties/MesonPropertyPage.java b/build/org.eclipse.cdt.meson.ui/src/org/eclipse/cdt/meson/ui/properties/MesonPropertyPage.java
index 9b7faf40ad..82eea7935a 100644
--- a/build/org.eclipse.cdt.meson.ui/src/org/eclipse/cdt/meson/ui/properties/MesonPropertyPage.java
+++ b/build/org.eclipse.cdt.meson.ui/src/org/eclipse/cdt/meson/ui/properties/MesonPropertyPage.java
@@ -88,7 +88,7 @@ public class MesonPropertyPage extends PropertyPage {
 			configName = ((CBuildConfiguration) project.getActiveBuildConfig().getAdapter(ICBuildConfiguration.class))
 					.getName();
 			IPath sourceDir = project.getLocation();
-			String buildDir = project.getLocation().append("build").append(configName).toOSString(); //$NON-NLS-1$
+			String buildDir = project.getLocation().append("build").append(configName).toString(); //$NON-NLS-1$
 			IPath buildPath = new Path(buildDir).append("build.ninja"); //$NON-NLS-1$
 			configured = buildPath.toFile().exists();
 			if (configured) {
@@ -275,7 +275,7 @@ public class MesonPropertyPage extends PropertyPage {
 				String configName = ((CBuildConfiguration) project.getActiveBuildConfig()
 						.getAdapter(ICBuildConfiguration.class)).getName();
 				IPath sourceDir = project.getLocation();
-				String buildDir = project.getLocation().append("build").append(configName).toOSString(); //$NON-NLS-1$
+				String buildDir = project.getLocation().append("build").append(configName).toString(); //$NON-NLS-1$
 				ICommandLauncher launcher = CommandLauncherManager.getInstance()
 						.getCommandLauncher(project.getActiveBuildConfig().getAdapter(ICBuildConfiguration.class));
 				launcher.setProject(project);
