diff --git a/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/ExternalEditorInputFactory.java b/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/ExternalEditorInputFactory.java
index 694b59ba74..33e5d509f7 100644
--- a/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/ExternalEditorInputFactory.java
+++ b/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/ExternalEditorInputFactory.java
@@ -74,7 +74,7 @@ public class ExternalEditorInputFactory implements IElementFactory {
 	static void saveState(IMemento memento, ExternalEditorInput input) {
 		IPath location = input.getPath(input);
 		if (location != null) {
-			memento.putString(TAG_PATH, location.toOSString());
+			memento.putString(TAG_PATH, location.toString());
 		}
 		IProject project = null;
 		ITranslationUnit unit = input.getTranslationUnit();
