diff --git a/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/ManagedBuildInfo.java b/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/ManagedBuildInfo.java
index c13311800e..45b4bbe285 100644
--- a/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/ManagedBuildInfo.java
+++ b/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/ManagedBuildInfo.java
@@ -1020,7 +1020,7 @@ public class ManagedBuildInfo implements IManagedBuildInfo, IScannerInfo {
 								if (null != optionPathConverter) {
 									IPath platformPath = optionPathConverter.convertToPlatformLocation(paths[i], null,
 											null);
-									paths[i] = platformPath.toOSString();
+									paths[i] = platformPath.toString();
 								}
 							}
 							list.add(checkPath(paths[i]));
