diff --git a/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/templateengine/processes/CreateIncludeFolder.java b/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/templateengine/processes/CreateIncludeFolder.java
index 4d186eb6d4..abf613fda2 100644
--- a/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/templateengine/processes/CreateIncludeFolder.java
+++ b/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/templateengine/processes/CreateIncludeFolder.java
@@ -53,7 +53,7 @@ public class CreateIncludeFolder extends CreateSourceFolder {
 		try {
 			IConfiguration[] configs = info.getManagedProject().getConfigurations();
 			for (IConfiguration config : configs) {
-				String path = projectHandle.getFolder(targetPath).getLocation().toOSString();
+				String path = projectHandle.getFolder(targetPath).getLocation().toString();
 				IToolChain toolChain = config.getToolChain();
 				setIncludePathOptionForConfig(path, config, toolChain.getOptions(), toolChain);
 
