diff --git a/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/makegen/internal/DefaultIndexerDependencyCalculator.java b/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/makegen/internal/DefaultIndexerDependencyCalculator.java
index 6d6289c225..6b1627c4d3 100644
--- a/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/makegen/internal/DefaultIndexerDependencyCalculator.java
+++ b/build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/makegen/internal/DefaultIndexerDependencyCalculator.java
@@ -30,7 +30,7 @@ public class DefaultIndexerDependencyCalculator implements IManagedDependencyGen
 	public IResource[] findDependencies(IResource resource, IProject project) {
 		//		PathCollector pathCollector = new PathCollector();
 		//		ICSearchScope scope = SearchEngine.createWorkspaceScope();
-		//		CSearchPattern pattern = CSearchPattern.createPattern(resource.getLocation().toOSString(), ICSearchConstants.INCLUDE, ICSearchConstants.REFERENCES, ICSearchConstants.EXACT_MATCH, true);
+		//		CSearchPattern pattern = CSearchPattern.createPattern(resource.getLocation().toString(), ICSearchConstants.INCLUDE, ICSearchConstants.REFERENCES, ICSearchConstants.EXACT_MATCH, true);
 		//		IndexManager indexManager = CCorePlugin.getDefault().getCoreModel().getIndexManager();
 		//		indexManager.performConcurrentJob(
 		//			new PatternSearchJob(
