diff --git a/build/org.eclipse.cdt.make.ui/src/org/eclipse/cdt/make/ui/dialogs/GCCPerFileSCDProfilePage.java b/build/org.eclipse.cdt.make.ui/src/org/eclipse/cdt/make/ui/dialogs/GCCPerFileSCDProfilePage.java
index db3441a3d4..eab175f579 100644
--- a/build/org.eclipse.cdt.make.ui/src/org/eclipse/cdt/make/ui/dialogs/GCCPerFileSCDProfilePage.java
+++ b/build/org.eclipse.cdt.make.ui/src/org/eclipse/cdt/make/ui/dialogs/GCCPerFileSCDProfilePage.java
@@ -137,7 +137,7 @@ public class GCCPerFileSCDProfilePage extends AbstractDiscoveryPage {
 					IPath filePath = new Path(fileName);
 					filterPath = filePath.removeLastSegments(1).makeAbsolute();
 				}
-				dialog.setFilterPath(filterPath.toOSString());
+				dialog.setFilterPath(filterPath.toString());
 				String res = dialog.open();
 				if (res == null) {
 					return;
