diff --git a/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/OpenIncludeAction.java b/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/OpenIncludeAction.java
index c7e8e1604d..74c14dbc27 100644
--- a/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/OpenIncludeAction.java
+++ b/build/org.eclipse.cdt.autotools.ui/src/org/eclipse/cdt/internal/autotools/ui/editors/automake/OpenIncludeAction.java
@@ -110,7 +110,7 @@ public class OpenIncludeAction extends Action {
 							// search in current directory
 							IPath location = include.getTranslationUnit().getLocation();
 							if (location != null) {
-								String currentDir = location.removeLastSegments(1).toOSString();
+								String currentDir = location.removeLastSegments(1).toString();
 								findFile(new String[] { currentDir }, includeName, filesFound);
 							}
 							if (filesFound.isEmpty()) {
@@ -231,7 +231,7 @@ public class OpenIncludeAction extends Action {
 				int numSegToRemove = rPath.segmentCount() - name.segmentCount();
 				IPath sPath = rPath.removeFirstSegments(numSegToRemove);
 				sPath = sPath.setDevice(name.getDevice());
-				if (Platform.getOS().equals(Platform.OS_WIN32) ? sPath.toOSString().equalsIgnoreCase(name.toOSString())
+				if (Platform.getOS().equals(Platform.OS_WIN32) ? sPath.toString().equalsIgnoreCase(name.toString())
 						: sPath.equals(name)) {
 					list.add(rPath);
 				}
