diff --git a/build/org.eclipse.cdt.make.core/src/org/eclipse/cdt/make/internal/core/scannerconfig/util/CygpathTranslator.java b/build/org.eclipse.cdt.make.core/src/org/eclipse/cdt/make/internal/core/scannerconfig/util/CygpathTranslator.java
index 4ece541085..38bde97d79 100644
--- a/build/org.eclipse.cdt.make.core/src/org/eclipse/cdt/make/internal/core/scannerconfig/util/CygpathTranslator.java
+++ b/build/org.eclipse.cdt.make.core/src/org/eclipse/cdt/make/internal/core/scannerconfig/util/CygpathTranslator.java
@@ -141,7 +141,7 @@ public class CygpathTranslator {
 						String drive = realPath.segment(1);
 						if (drive.length() == 1) {
 							translatedPath = realPath.removeFirstSegments(2).makeAbsolute()
-									.setDevice(drive.toUpperCase() + ':').toOSString();
+									.setDevice(drive.toUpperCase() + ':').toString();
 						}
 					}
 				}
