diff --git a/codan/org.eclipse.cdt.codan.core.cxx/src/org/eclipse/cdt/codan/core/cxx/externaltool/InvocationParametersProvider.java b/codan/org.eclipse.cdt.codan.core.cxx/src/org/eclipse/cdt/codan/core/cxx/externaltool/InvocationParametersProvider.java
index 05003894da..15829c975a 100644
--- a/codan/org.eclipse.cdt.codan.core.cxx/src/org/eclipse/cdt/codan/core/cxx/externaltool/InvocationParametersProvider.java
+++ b/codan/org.eclipse.cdt.codan.core.cxx/src/org/eclipse/cdt/codan/core/cxx/externaltool/InvocationParametersProvider.java
@@ -35,7 +35,7 @@ public class InvocationParametersProvider implements IInvocationParametersProvid
 	 */
 	@Override
 	public InvocationParameters createParameters(IResource fileToProcess) {
-		String path = fileToProcess.getLocation().toOSString();
+		String path = fileToProcess.getLocation().toString();
 		return new InvocationParameters(fileToProcess, fileToProcess, path, null);
 	}
 }
