diff --git a/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/text/selection/CPPSelectionNoIndexerTests.java b/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/text/selection/CPPSelectionNoIndexerTests.java
index e1fd71660a..66c30db9e0 100644
--- a/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/text/selection/CPPSelectionNoIndexerTests.java
+++ b/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/text/selection/CPPSelectionNoIndexerTests.java
@@ -99,7 +99,7 @@ public class CPPSelectionNoIndexerTests extends BaseSelectionTests {
 		project = cPrj.getProject();
 
 		IPath pathLoc = CCorePlugin.getDefault().getStateLocation();
-		File indexFile = new File(pathLoc.append(INDEX_FILE_ID + ".index").toOSString()); //$NON-NLS-1$
+		File indexFile = new File(pathLoc.append(INDEX_FILE_ID + ".index").toString()); //$NON-NLS-1$
 		if (indexFile.exists())
 			indexFile.delete();
 		assertNotNull("Unable to create project"); //$NON-NLS-1$
@@ -140,7 +140,7 @@ public class CPPSelectionNoIndexerTests extends BaseSelectionTests {
 
 		IPath location = ResourceHelper.createTemporaryFolder().append(fileName);
 
-		File linkFile = new File(location.toOSString());
+		File linkFile = new File(location.toString());
 		if (!linkFile.exists()) {
 			linkFile.createNewFile();
 		}
@@ -162,7 +162,7 @@ public class CPPSelectionNoIndexerTests extends BaseSelectionTests {
 	protected IFile importFileInsideLinkedFolder(String fileName, String contents, String folderName) throws Exception {
 		IFolder linkedFolder = project.getFolder(folderName);
 		IPath folderLocation = new Path(
-				project.getLocation().toOSString() + File.separator + folderName + "_this_is_linked"); //$NON-NLS-1$
+				project.getLocation().toString() + File.separator + folderName + "_this_is_linked"); //$NON-NLS-1$
 		IFolder actualFolder = project.getFolder(folderName + "_this_is_linked"); //$NON-NLS-1$
 		if (!actualFolder.exists())
 			actualFolder.create(true, true, monitor);
