diff --git a/core/org.eclipse.cdt.ui/templateengine/org/eclipse/cdt/ui/templateengine/ProjectSelectionPage.java b/core/org.eclipse.cdt.ui/templateengine/org/eclipse/cdt/ui/templateengine/ProjectSelectionPage.java
index 783e2bb573..97df10b384 100644
--- a/core/org.eclipse.cdt.ui/templateengine/org/eclipse/cdt/ui/templateengine/ProjectSelectionPage.java
+++ b/core/org.eclipse.cdt.ui/templateengine/org/eclipse/cdt/ui/templateengine/ProjectSelectionPage.java
@@ -275,7 +275,7 @@ public class ProjectSelectionPage extends WizardPage implements IWizardDataPage
 		ICProject cProject = chooseProject();
 		if (cProject != null) {
 			IPath path = cProject.getProject().getFullPath().makeRelative();
-			projectName = path.toOSString();
+			projectName = path.toString();
 			projectNameText.setText(projectName);
 		}
 	}
