diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/search/actions/SelectionParseAction.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/search/actions/SelectionParseAction.java
index 4b41f83778..532cefa5e8 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/search/actions/SelectionParseAction.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/search/actions/SelectionParseAction.java
@@ -113,7 +113,7 @@ public class SelectionParseAction extends Action {
 
 	protected void reportSourceFileOpenFailure(IPath path) {
 		showStatusLineMessage(MessageFormat.format(CSearchMessages.SelectionParseAction_FileOpenFailure_format,
-				new Object[] { path.toOSString() }));
+				new Object[] { path.toString() }));
 	}
 
 	protected void reportSelectionMatchFailure() {
