diff --git a/core/org.eclipse.cdt.core/model/org/eclipse/cdt/internal/core/model/TranslationUnit.java b/core/org.eclipse.cdt.core/model/org/eclipse/cdt/internal/core/model/TranslationUnit.java
index dcf563dc47..f1089dd7f6 100644
--- a/core/org.eclipse.cdt.core/model/org/eclipse/cdt/internal/core/model/TranslationUnit.java
+++ b/core/org.eclipse.cdt.core/model/org/eclipse/cdt/internal/core/model/TranslationUnit.java
@@ -1037,15 +1037,15 @@ public class TranslationUnit extends Openable implements ITranslationUnit {
 		if (location == null)
 			return new org.eclipse.cdt.core.parser.CodeReader(getContents());
 		if (isWorkingCopy()) {
-			return new org.eclipse.cdt.core.parser.CodeReader(location.toOSString(), getContents());
+			return new org.eclipse.cdt.core.parser.CodeReader(location.toString(), getContents());
 		}
 
 		IResource res = getResource();
 		try {
 			if (res instanceof IFile)
-				return InternalParserUtil.createWorkspaceFileReader(location.toOSString(), (IFile) res, null);
+				return InternalParserUtil.createWorkspaceFileReader(location.toString(), (IFile) res, null);
 			else
-				return InternalParserUtil.createExternalFileReader(location.toOSString(), null);
+				return InternalParserUtil.createExternalFileReader(location.toString(), null);
 		} catch (CoreException e) {
 			CCorePlugin.log(e);
 		} catch (IOException e) {
