diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/util/ExternalEditorInputFactory.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/util/ExternalEditorInputFactory.java
index dcf3f868b4..ead6632e3f 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/util/ExternalEditorInputFactory.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/util/ExternalEditorInputFactory.java
@@ -76,7 +76,7 @@ public class ExternalEditorInputFactory implements IElementFactory {
 	static void saveState(IMemento memento, ExternalEditorInput input) {
 		IPath location = input.getPath();
 		if (location != null) {
-			memento.putString(TAG_PATH, location.toOSString());
+			memento.putString(TAG_PATH, location.toString());
 		}
 		IProject project = null;
 		ITranslationUnit unit = input.getTranslationUnit();
