diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/utils/NamespaceHelper.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/utils/NamespaceHelper.java
index 1d9f660dc9..fe316bcd43 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/utils/NamespaceHelper.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/utils/NamespaceHelper.java
@@ -86,7 +86,7 @@ public class NamespaceHelper {
 		if (fileLoc == null) {
 			return false;
 		}
-		boolean fileNameOk = fileLoc.getFileName().endsWith(path.toOSString());
+		boolean fileNameOk = fileLoc.getFileName().endsWith(path.toString());
 		if (!fileNameOk) {
 			return false;
 		}
