diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/navigator/CNavigatorDragAdapterAssistant.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/navigator/CNavigatorDragAdapterAssistant.java
index 559df1409c..d170190c46 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/navigator/CNavigatorDragAdapterAssistant.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/navigator/CNavigatorDragAdapterAssistant.java
@@ -80,7 +80,7 @@ public class CNavigatorDragAdapterAssistant extends CommonDragAdapterAssistant {
 						ITranslationUnit tu = (ITranslationUnit) element;
 						IPath location = tu.getLocation();
 						if (location != null) {
-							files.add(location.toOSString());
+							files.add(location.toString());
 						}
 					}
 				}
