diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/changes/CreateFileChange.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/changes/CreateFileChange.java
index ca4b56c32e..69785a8e6a 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/changes/CreateFileChange.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/changes/CreateFileChange.java
@@ -64,7 +64,7 @@ public class CreateFileChange extends ResourceChange {
 	@Override
 	public String getName() {
 		if (name == null) {
-			return NLS.bind(Messages.CreateFileChange_create_file, path.toOSString());
+			return NLS.bind(Messages.CreateFileChange_create_file, path.toString());
 		}
 		return name;
 	}
