diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/buildconsole/BuildConsoleManager.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/buildconsole/BuildConsoleManager.java
index 9e1dc20db1..b06b06cc08 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/buildconsole/BuildConsoleManager.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/buildconsole/BuildConsoleManager.java
@@ -527,7 +527,7 @@ public class BuildConsoleManager implements IBuildConsoleManager, IResourceChang
 	public String getDefaultConsoleLogLocation(IProject project) {
 		IPath defaultLogLocation = CUIPlugin.getDefault().getStateLocation()
 				.append(project.getName() + PROJECT_LOG_EXT);
-		return defaultLogLocation.toOSString();
+		return defaultLogLocation.toString();
 	}
 
 	/**
