diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/preferences/BuildLogPreferencePage.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/preferences/BuildLogPreferencePage.java
index 23f4d10e0a..8fe78f9f6a 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/preferences/BuildLogPreferencePage.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/preferences/BuildLogPreferencePage.java
@@ -105,7 +105,7 @@ public class BuildLogPreferencePage extends PropertyPage implements ICOptionCont
 					dialog.setText(PreferencesMessages.BuildLogPreferencePage_ChooseLogFile);
 					String fileName = logLocationText.getText();
 					IPath logFolder = new Path(fileName).removeLastSegments(1);
-					dialog.setFilterPath(logFolder.toOSString());
+					dialog.setFilterPath(logFolder.toString());
 					String chosenFile = dialog.open();
 					if (chosenFile != null) {
 						logLocationText.setText(chosenFile);
