diff --git a/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/refactoring/includes/IncludesTestBase.java b/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/refactoring/includes/IncludesTestBase.java
index 880882dd20..7fbdb9960b 100644
--- a/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/refactoring/includes/IncludesTestBase.java
+++ b/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/refactoring/includes/IncludesTestBase.java
@@ -91,7 +91,7 @@ public abstract class IncludesTestBase extends BaseTestCase {
 				: CProjectHelper.createCProject(getName() + System.currentTimeMillis(), "bin",
 						IPDOMManager.ID_NO_INDEXER);
 		IProject project = cproject.getProject();
-		TestScannerProvider.sLocalIncludes = new String[] { project.getLocation().toOSString() };
+		TestScannerProvider.sLocalIncludes = new String[] { project.getLocation().toString() };
 		QualifiedName scannerInfoProviderName = new QualifiedName(CCorePlugin.PLUGIN_ID, SCANNER_INFO_PROVIDER2_NAME);
 		project.setSessionProperty(scannerInfoProviderName, new TestScannerProvider());
 
