diff --git a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/som/parser/SOMBinaryObject.java b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/som/parser/SOMBinaryObject.java
index 646b8f5679..0f49ea8310 100644
--- a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/som/parser/SOMBinaryObject.java
+++ b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/som/parser/SOMBinaryObject.java
@@ -127,9 +127,9 @@ public class SOMBinaryObject extends BinaryObjectAdapter {
 
 	protected SOM getSOM() throws IOException {
 		if (header != null) {
-			return new SOM(getPath().toOSString(), header.getObjectDataOffset());
+			return new SOM(getPath().toString(), header.getObjectDataOffset());
 		}
-		return new SOM(getPath().toOSString());
+		return new SOM(getPath().toString());
 	}
 
 	protected void loadAll() throws IOException {
