diff --git a/core/org.eclipse.cdt.core.tests/misc/org/eclipse/cdt/core/internal/errorparsers/tests/ErrorParserFileMatchingTest.java b/core/org.eclipse.cdt.core.tests/misc/org/eclipse/cdt/core/internal/errorparsers/tests/ErrorParserFileMatchingTest.java
index fe2575cd3b..19c39635a2 100644
--- a/core/org.eclipse.cdt.core.tests/misc/org/eclipse/cdt/core/internal/errorparsers/tests/ErrorParserFileMatchingTest.java
+++ b/core/org.eclipse.cdt.core.tests/misc/org/eclipse/cdt/core/internal/errorparsers/tests/ErrorParserFileMatchingTest.java
@@ -530,7 +530,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 	@Test
 	public void testAbsolutePathSingle() throws Exception {
 		ResourceHelper.createFile(fProject, "testAbsolutePathSingle.c");
-		String fullName = fProject.getLocation().append("testAbsolutePathSingle.c").toOSString();
+		String fullName = fProject.getLocation().append("testAbsolutePathSingle.c").toString();
 
 		parseOutput(fullName + ":1:error");
 		assertEquals(1, errorList.size());
@@ -549,7 +549,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 	public void testAbsolutePathInOtherProject() throws Exception {
 		IProject anotherProject = ResourceHelper.createCDTProject("ProjectAbsolutePathInOtherProject");
 		ResourceHelper.createFile(anotherProject, "testAbsolutePathInOtherProject.c");
-		String fullName = anotherProject.getLocation().append("testAbsolutePathInOtherProject.c").toOSString();
+		String fullName = anotherProject.getLocation().append("testAbsolutePathInOtherProject.c").toString();
 
 		parseOutput(fullName + ":1:error");
 		assertEquals(1, errorList.size());
@@ -572,7 +572,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 		IPath outsideFile = ResourceHelper.createWorkspaceFile("OutsideFolder/testAbsolutePathOutsideWorkspace.c");
 
 		String fullName = ResourcesPlugin.getWorkspace().getRoot().getLocation()
-				.append("OutsideFolder/testAbsolutePathOutsideWorkspace.c").toOSString();
+				.append("OutsideFolder/testAbsolutePathOutsideWorkspace.c").toString();
 
 		parseOutput(fullName + ":1:error");
 		assertEquals(1, errorList.size());
@@ -919,7 +919,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 	@Test
 	public void testCygwinCygdrive() throws Exception {
 		String fileName = "testCygwinCygdrive.c";
-		String windowsFileName = fProject.getLocation().append(fileName).toOSString();
+		String windowsFileName = fProject.getLocation().append(fileName).toString();
 		String cygwinFileName;
 		try {
 			cygwinFileName = Cygwin.windowsToCygwinPath(windowsFileName);
@@ -983,7 +983,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 		String fileName = "testCygwinAnotherProject.c";
 		IProject anotherProject = ResourceHelper.createCDTProject("AnotherProject");
 
-		String windowsFileName = anotherProject.getLocation().append(fileName).toOSString();
+		String windowsFileName = anotherProject.getLocation().append(fileName).toString();
 		String cygwinFileName;
 		try {
 			cygwinFileName = Cygwin.windowsToCygwinPath(windowsFileName);
@@ -1034,7 +1034,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 	@Test
 	public void testCygwinAndMakeErrorParserBug270772() throws Exception {
 		String fileName = "testCygwinAndMakeErrorParser.c";
-		String windowsFileName = fProject.getLocation().append(fileName).toOSString();
+		String windowsFileName = fProject.getLocation().append(fileName).toString();
 		String cygwinFileName;
 		try {
 			cygwinFileName = Cygwin.windowsToCygwinPath(windowsFileName);
@@ -1104,7 +1104,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 		ResourceHelper.createFolder(fProject, "BuildDir");
 		ResourceHelper.createFile(fProject, "BuildDir/testBuildDir.c");
 
-		String buildDir = fProject.getLocation().append("BuildDir").toOSString();
+		String buildDir = fProject.getLocation().append("BuildDir").toString();
 		parseOutput(fProject, buildDir, "testBuildDir.c:1:error");
 		assertEquals(1, errorList.size());
 
@@ -1124,7 +1124,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 		ResourceHelper.createFolder(fProject, "BuildDir");
 		ResourceHelper.createFile(fProject, "BuildDir/testBuildDirVsProjectRoot.c");
 
-		String buildDir = fProject.getLocation().append("BuildDir").toOSString();
+		String buildDir = fProject.getLocation().append("BuildDir").toString();
 		parseOutput(fProject, buildDir, "testBuildDirVsProjectRoot.c:1:error");
 		assertEquals(1, errorList.size());
 
@@ -1143,7 +1143,7 @@ public class ErrorParserFileMatchingTest extends BaseTestCase5 {
 	public void testAbsoluteFileVsLink() throws Exception {
 		ResourceHelper.createFolder(fProject, "Folder");
 		IFile file = ResourceHelper.createFile(fProject, "Folder/testAbsoluteFileVsLink.c");
-		String fullName = file.getLocation().toOSString();
+		String fullName = file.getLocation().toString();
 		ResourceHelper.createLinkedFile(fProject, "testAbsoluteFileVsLink.c", file.getLocation());
 
 		parseOutput(fullName + ":1:error");
