diff --git a/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/typehierarchy/TypeHierarchyAcrossProjectsTest.java b/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/typehierarchy/TypeHierarchyAcrossProjectsTest.java
index 3633488b15..1fc1c7e62e 100644
--- a/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/typehierarchy/TypeHierarchyAcrossProjectsTest.java
+++ b/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/typehierarchy/TypeHierarchyAcrossProjectsTest.java
@@ -51,8 +51,8 @@ public class TypeHierarchyAcrossProjectsTest extends TypeHierarchyBaseTest {
 
 		CCorePlugin.getIndexManager().reindex(fCProject2);
 		fIndex = CCorePlugin.getIndexManager().getIndex(new ICProject[] { fCProject, fCProject2 });
-		TestScannerProvider.sIncludes = new String[] { fCProject.getProject().getLocation().toOSString(),
-				fCProject2.getProject().getLocation().toOSString() };
+		TestScannerProvider.sIncludes = new String[] { fCProject.getProject().getLocation().toString(),
+				fCProject2.getProject().getLocation().toString() };
 	}
 
 	@Override
