diff --git a/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/core/parser/FileContent.java b/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/core/parser/FileContent.java
index 73f10c50dc..2a9cbf29a2 100644
--- a/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/core/parser/FileContent.java
+++ b/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/core/parser/FileContent.java
@@ -111,13 +111,13 @@ public abstract class FileContent {
 		if (location == null) {
 			fileContent = new InternalFileContent(tu.getElementName(), new CharArray(tu.getContents()));
 		} else if (tu.isWorkingCopy()) {
-			fileContent = new InternalFileContent(location.toOSString(), new CharArray(tu.getContents()));
+			fileContent = new InternalFileContent(location.toString(), new CharArray(tu.getContents()));
 		} else {
 			IResource res = tu.getResource();
 			if (res instanceof IFile) {
 				fileContent = InternalParserUtil.createWorkspaceFileContent((IFile) res);
 			} else {
-				fileContent = InternalParserUtil.createExternalFileContent(location.toOSString(),
+				fileContent = InternalParserUtil.createExternalFileContent(location.toString(),
 						InternalParserUtil.SYSTEM_DEFAULT_ENCODING);
 			}
 		}
