diff --git a/core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/CommandLauncher.java b/core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/CommandLauncher.java
index 0f7984a6e1..5d90b3dead 100644
--- a/core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/CommandLauncher.java
+++ b/core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/CommandLauncher.java
@@ -164,7 +164,7 @@ public class CommandLauncher implements ICommandLauncher {
 		String envPathValue = getEnvironmentProperty(PATH_ENV);
 
 		Boolean isFound = null;
-		String command = commandPath.toOSString();
+		String command = commandPath.toString();
 		fCommandArgs = constructCommandArray(command, args);
 		if (Platform.getOS().equals(Platform.OS_WIN32)) {
 			// Handle cygwin link
