diff --git a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/xcoff/parser/XCOFFBinaryArchive.java b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/xcoff/parser/XCOFFBinaryArchive.java
index d0cdf5f729..2494bb5b86 100644
--- a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/xcoff/parser/XCOFFBinaryArchive.java
+++ b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/xcoff/parser/XCOFFBinaryArchive.java
@@ -39,7 +39,7 @@ public class XCOFFBinaryArchive extends BinaryFile implements IBinaryArchive {
 	 */
 	public XCOFFBinaryArchive(IBinaryParser parser, IPath path) throws IOException {
 		super(parser, path, IBinaryFile.ARCHIVE);
-		try (AR ar = new AR(path.toOSString())) {
+		try (AR ar = new AR(path.toString())) {
 			// create the object just to check file type
 		}
 		children = new ArrayList<>(5);
@@ -51,7 +51,7 @@ public class XCOFFBinaryArchive extends BinaryFile implements IBinaryArchive {
 			children.clear();
 			AR ar = null;
 			try {
-				ar = new AR(getPath().toOSString());
+				ar = new AR(getPath().toString());
 				AR.MemberHeader[] headers = ar.getHeaders();
 				for (int i = 0; i < headers.length; i++) {
 					IBinaryObject bin = new XCOFFBinaryObject(getBinaryParser(), getPath(), headers[i]);
