diff --git a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/elf/parser/ElfBinaryArchive.java b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/elf/parser/ElfBinaryArchive.java
index a4affdd11c..bfa0fc23f1 100644
--- a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/elf/parser/ElfBinaryArchive.java
+++ b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/elf/parser/ElfBinaryArchive.java
@@ -34,7 +34,7 @@ public class ElfBinaryArchive extends BinaryFile implements IBinaryArchive {
 
 	public ElfBinaryArchive(IBinaryParser parser, IPath path) throws IOException {
 		super(parser, path, IBinaryFile.ARCHIVE);
-		try (AR ar = new AR(path.toOSString())) {
+		try (AR ar = new AR(path.toString())) {
 			// create the object just to check file type
 		}
 		children = new ArrayList<>(5);
@@ -46,7 +46,7 @@ public class ElfBinaryArchive extends BinaryFile implements IBinaryArchive {
 			children.clear();
 			AR ar = null;
 			try {
-				ar = new AR(getPath().toOSString());
+				ar = new AR(getPath().toString());
 				AR.ARHeader[] headers = ar.getHeaders();
 				IBinaryObject[] bobjs = createArchiveMembers(headers);
 				children.addAll(Arrays.asList(bobjs));
