diff --git a/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/pdom/indexer/ProjectIndexerInputAdapter.java b/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/pdom/indexer/ProjectIndexerInputAdapter.java
index 3e199f69ee..f0bd407a8c 100644
--- a/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/pdom/indexer/ProjectIndexerInputAdapter.java
+++ b/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/pdom/indexer/ProjectIndexerInputAdapter.java
@@ -150,7 +150,7 @@ public class ProjectIndexerInputAdapter extends IndexerInputAdapter {
 	public String getASTPath(IIndexFileLocation ifl) {
 		IPath path = IndexLocationFactory.getAbsolutePath(ifl);
 		if (path != null) {
-			return path.toOSString();
+			return path.toString();
 		}
 		return ifl.getURI().getPath();
 	}
