diff --git a/core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/parser/CodeReaderCache.java b/core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/parser/CodeReaderCache.java
index e846d5258d..b7afd3d661 100644
--- a/core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/parser/CodeReaderCache.java
+++ b/core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/parser/CodeReaderCache.java
@@ -105,7 +105,7 @@ public class CodeReaderCache implements ICodeReaderCache {
 						removeKeys(delta.getAffectedChildren());
 					} else if (delta.getResource() instanceof IFile
 							&& ((IFile) delta.getResource()).getLocation() != null) {
-						removeKey(((IFile) delta.getResource()).getLocation().toOSString());
+						removeKey(((IFile) delta.getResource()).getLocation().toString());
 					}
 				}
 			}
