diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/search/CSearchResult.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/search/CSearchResult.java
index a433ef12f2..f15c927a80 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/search/CSearchResult.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/search/CSearchResult.java
@@ -97,7 +97,7 @@ public class CSearchResult extends AbstractTextSearchResult implements IEditorMa
 			}
 		}
 		if (path != null)
-			return path.toOSString();
+			return path.toString();
 
 		return null;
 	}
@@ -144,7 +144,7 @@ public class CSearchResult extends AbstractTextSearchResult implements IEditorMa
 	@Override
 	public Match[] computeContainedMatches(AbstractTextSearchResult result, IFile file) {
 		try {
-			String filename = file.getLocation().toOSString();
+			String filename = file.getLocation().toString();
 			return computeContainedMatches(result, filename);
 		} catch (CoreException e) {
 			CUIPlugin.log(e);
