diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/AddIncludeAction.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/AddIncludeAction.java
index 8e772d139c..785322e7bd 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/AddIncludeAction.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/AddIncludeAction.java
@@ -113,7 +113,7 @@ public class AddIncludeAction extends TextEditorAction {
 						IASTTranslationUnit ast = tu.getAST(index, ASTCache.PARSE_MODE);
 						if (ast == null) {
 							return CUIPlugin.createErrorStatus(
-									NLS.bind(CEditorMessages.AddInclude_ast_not_available, tu.getPath().toOSString()));
+									NLS.bind(CEditorMessages.AddInclude_ast_not_available, tu.getPath().toString()));
 						}
 						IncludeCreator creator = new IncludeCreator(tu, index, fAmbiguityResolver);
 						holder[0] = creator.createInclude(ast, (ITextSelection) selection);
