diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/includes/InclusionRequest.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/includes/InclusionRequest.java
index e0c7865738..6d60c053ff 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/includes/InclusionRequest.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/includes/InclusionRequest.java
@@ -116,11 +116,11 @@ class InclusionRequest {
 		for (int i = 0; i < fCandidatePaths.size(); i++) {
 			if (i != 0)
 				buf.append(", "); //$NON-NLS-1$
-			buf.append(fCandidatePaths.get(i).toOSString());
+			buf.append(fCandidatePaths.get(i).toString());
 		}
 		if (fResolvedPath != null) {
 			buf.append(" represented by "); //$NON-NLS-1$
-			buf.append(fResolvedPath.toOSString());
+			buf.append(fResolvedPath.toString());
 		}
 		return buf.toString();
 	}
