diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/CElementIncludeResolver.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/CElementIncludeResolver.java
index abebbdd70b..7910a83c25 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/CElementIncludeResolver.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/CElementIncludeResolver.java
@@ -78,7 +78,7 @@ public class CElementIncludeResolver {
 						// search in current directory
 						IPath location = include.getTranslationUnit().getLocation();
 						if (location != null) {
-							String currentDir = location.removeLastSegments(1).toOSString();
+							String currentDir = location.removeLastSegments(1).toString();
 							findFile(new String[] { currentDir }, includeName, filesFound);
 						}
 						if (filesFound.isEmpty()) {
@@ -149,7 +149,7 @@ public class CElementIncludeResolver {
 						IPath sPath = rPath.removeFirstSegments(numSegToRemove);
 						sPath = sPath.setDevice(name.getDevice());
 						if (Platform.getOS().equals(Platform.OS_WIN32)
-								? sPath.toOSString().equalsIgnoreCase(name.toOSString())
+								? sPath.toString().equalsIgnoreCase(name.toString())
 								: sPath.equals(name)) {
 							list.add(rPath);
 						}
