diff --git a/core/org.eclipse.cdt.core.tests/parser/org/eclipse/cdt/internal/index/tests/IndexBindingResolutionTestBase.java b/core/org.eclipse.cdt.core.tests/parser/org/eclipse/cdt/internal/index/tests/IndexBindingResolutionTestBase.java
index f0b037c928..974ed70df6 100644
--- a/core/org.eclipse.cdt.core.tests/parser/org/eclipse/cdt/internal/index/tests/IndexBindingResolutionTestBase.java
+++ b/core/org.eclipse.cdt.core.tests/parser/org/eclipse/cdt/internal/index/tests/IndexBindingResolutionTestBase.java
@@ -858,7 +858,7 @@ public abstract class IndexBindingResolutionTestBase extends SemanticTestBase {
 
 			referenced = createReferencedContent();
 
-			TestScannerProvider.sIncludes = new String[] { referenced.getProject().getLocation().toOSString() };
+			TestScannerProvider.sIncludes = new String[] { referenced.getProject().getLocation().toString() };
 			IFile references = TestSourceReader.createFile(cproject.getProject(),
 					new Path("refs.c" + (cpp ? "pp" : "")), sourceContents);
 
