diff --git a/core/org.eclipse.cdt.core/src/org/eclipse/cdt/internal/core/dom/SavedCodeReaderFactory.java b/core/org.eclipse.cdt.core/src/org/eclipse/cdt/internal/core/dom/SavedCodeReaderFactory.java
index bf995f159e..07c32315e2 100644
--- a/core/org.eclipse.cdt.core/src/org/eclipse/cdt/internal/core/dom/SavedCodeReaderFactory.java
+++ b/core/org.eclipse.cdt.core/src/org/eclipse/cdt/internal/core/dom/SavedCodeReaderFactory.java
@@ -84,7 +84,7 @@ public class SavedCodeReaderFactory extends AbstractCodeReaderFactory {
 	}
 
 	public CodeReader createCodeReaderForTranslationUnit(ITranslationUnit tu) {
-		return new CodeReader(tu.getResource().getLocation().toOSString(), tu.getContents());
+		return new CodeReader(tu.getResource().getLocation().toString(), tu.getContents());
 	}
 
 	@Override
