diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/includes/IncludeUtil.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/includes/IncludeUtil.java
index cc7c87859d..30c3dfee88 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/includes/IncludeUtil.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/refactoring/includes/IncludeUtil.java
@@ -111,7 +111,7 @@ public class IncludeUtil {
 	 * @return The path.
 	 */
 	public static String getPath(IIndexFileLocation fileLocation) {
-		return IndexLocationFactory.getAbsolutePath(fileLocation).toOSString();
+		return IndexLocationFactory.getAbsolutePath(fileLocation).toString();
 	}
 
 	public static boolean isContainedInRegion(IASTNode node, IRegion region) {
