diff --git a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/DefaultCygwinToolFactory.java b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/DefaultCygwinToolFactory.java
index 770c067daa..9f538f058f 100644
--- a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/DefaultCygwinToolFactory.java
+++ b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/DefaultCygwinToolFactory.java
@@ -35,7 +35,7 @@ public class DefaultCygwinToolFactory extends DefaultGnuToolFactory implements I
 		CygPath cygpath = null;
 		if (cygPathPath != null && !cygPathPath.isEmpty()) {
 			try {
-				cygpath = new CygPath(cygPathPath.toOSString());
+				cygpath = new CygPath(cygPathPath.toString());
 			} catch (IOException e1) {
 			}
 		}
