diff --git a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/coff/parser/PEBinaryArchive64.java b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/coff/parser/PEBinaryArchive64.java
index c37b477450..c56d1e207c 100644
--- a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/coff/parser/PEBinaryArchive64.java
+++ b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/coff/parser/PEBinaryArchive64.java
@@ -34,7 +34,7 @@ public class PEBinaryArchive64 extends BinaryFile implements IBinaryArchive {
 
 	public PEBinaryArchive64(PEParser64 parser, IPath path) throws IOException {
 		super(parser, path, IBinaryFile.ARCHIVE);
-		try (AR ar = new AR(path.toOSString())) {
+		try (AR ar = new AR(path.toString())) {
 			// create the object just to check file type
 		}
 		children = new ArrayList<>(5);
@@ -49,7 +49,7 @@ public class PEBinaryArchive64 extends BinaryFile implements IBinaryArchive {
 			children.clear();
 			AR ar = null;
 			try {
-				ar = new AR(getPath().toOSString());
+				ar = new AR(getPath().toString());
 				AR.ARHeader[] headers = ar.getHeaders();
 				addArchiveMembers(headers, children);
 			} catch (IOException e) {
