diff --git a/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/refactoring/RefactoringTestBase.java b/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/refactoring/RefactoringTestBase.java
index b353571122..b4c0a2ebbe 100644
--- a/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/refactoring/RefactoringTestBase.java
+++ b/core/org.eclipse.cdt.ui.tests/ui/org/eclipse/cdt/ui/tests/refactoring/RefactoringTestBase.java
@@ -94,7 +94,7 @@ public abstract class RefactoringTestBase extends BaseTestCase {
 						IPDOMManager.ID_NO_INDEXER)
 				: CProjectHelper.createCProject(getName() + System.currentTimeMillis(), "bin",
 						IPDOMManager.ID_NO_INDEXER);
-		TestScannerProvider.sLocalIncludes = new String[] { cproject.getProject().getLocation().toOSString() };
+		TestScannerProvider.sLocalIncludes = new String[] { cproject.getProject().getLocation().toString() };
 
 		Bundle bundle = CTestPlugin.getDefault().getBundle();
 		CharSequence[] testData = TestSourceReader.getContentsForTest(bundle, "ui", getClass(), getName(), 0);
