diff --git a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/internal/core/MinGW.java b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/internal/core/MinGW.java
index 7d516ee7df..a067a03ee5 100644
--- a/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/internal/core/MinGW.java
+++ b/core/org.eclipse.cdt.core/utils/org/eclipse/cdt/internal/core/MinGW.java
@@ -68,7 +68,7 @@ public class MinGW {
 		if (envMinGWHomeValue != null && !envMinGWHomeValue.isEmpty()) {
 			IPath mingwBinDir = new Path(envMinGWHomeValue + "\\bin"); //$NON-NLS-1$
 			if (mingwBinDir.toFile().isDirectory()) {
-				rootValue = mingwBinDir.removeLastSegments(1).toOSString();
+				rootValue = mingwBinDir.removeLastSegments(1).toString();
 			}
 		}
 
@@ -78,7 +78,7 @@ public class MinGW {
 			IPath installPath = new Path(Platform.getInstallLocation().getURL().getFile());
 			IPath mingwBinDir = installPath.append("mingw\\bin"); //$NON-NLS-1$
 			if (mingwBinDir.toFile().isDirectory()) {
-				rootValue = mingwBinDir.removeLastSegments(1).toOSString();
+				rootValue = mingwBinDir.removeLastSegments(1).toString();
 			}
 		}
 
@@ -151,7 +151,7 @@ public class MinGW {
 		if (rootValue == null) {
 			IPath mingwBinDir = new Path("C:\\MinGW"); //$NON-NLS-1$
 			if (mingwBinDir.toFile().isDirectory()) {
-				rootValue = mingwBinDir.toOSString();
+				rootValue = mingwBinDir.toString();
 			}
 		}
 
@@ -161,7 +161,7 @@ public class MinGW {
 	private static Optional<String> findMinGwInstallationLocation(String exeName, String envPath) {
 		IPath exeLocation = PathUtil.findProgramLocation(exeName, envPath);
 		if (exeLocation != null) {
-			return Optional.of(exeLocation.removeLastSegments(2).toOSString());
+			return Optional.of(exeLocation.removeLastSegments(2).toString());
 		}
 		return Optional.empty();
 	}
@@ -209,7 +209,7 @@ public class MinGW {
 		IPath installPath = new Path(Platform.getInstallLocation().getURL().getFile());
 		IPath installMsysBin = installPath.append("msys\\bin"); //$NON-NLS-1$
 		if (installMsysBin.toFile().isDirectory()) {
-			msysHome = installMsysBin.removeLastSegments(1).toOSString();
+			msysHome = installMsysBin.removeLastSegments(1).toString();
 		}
 
 		// Look under $MINGW_HOME
@@ -217,7 +217,7 @@ public class MinGW {
 			if (envMinGWHomeValue != null && !envMinGWHomeValue.isEmpty()) {
 				IPath minGwMsysBin = new Path(envMinGWHomeValue + "\\msys\\1.0\\bin"); //$NON-NLS-1$
 				if (minGwMsysBin.toFile().isDirectory()) {
-					msysHome = minGwMsysBin.removeLastSegments(1).toOSString();
+					msysHome = minGwMsysBin.removeLastSegments(1).toString();
 				}
 			}
 		}
@@ -270,7 +270,7 @@ public class MinGW {
 		if (msysHome == null) {
 			IPath minGwMsysBin = new Path("C:\\MinGW\\msys\\1.0\\bin"); //$NON-NLS-1$
 			if (minGwMsysBin.toFile().isDirectory()) {
-				msysHome = minGwMsysBin.removeLastSegments(1).toOSString();
+				msysHome = minGwMsysBin.removeLastSegments(1).toString();
 			}
 		}
 
@@ -278,7 +278,7 @@ public class MinGW {
 		if (msysHome == null) {
 			IPath defaultMsysBin = new Path("C:\\msys\\1.0\\bin"); //$NON-NLS-1$
 			if (defaultMsysBin.toFile().isDirectory()) {
-				msysHome = defaultMsysBin.removeLastSegments(1).toOSString();
+				msysHome = defaultMsysBin.removeLastSegments(1).toString();
 			}
 		}
 		return msysHome;
