diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/corext/util/Resources.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/corext/util/Resources.java
index 583368e9d3..6526f49eae 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/corext/util/Resources.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/corext/util/Resources.java
@@ -181,7 +181,7 @@ public class Resources {
 		for (int i = 0; i < resources.length; i++) {
 			IPath location = resources[i].getLocation();
 			if (location != null)
-				result.add(location.toOSString());
+				result.add(location.toString());
 		}
 		return result.toArray(new String[result.size()]);
 	}
