diff --git a/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/pdom/indexer/PDOMUpdateTask.java b/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/pdom/indexer/PDOMUpdateTask.java
index f96164e0f7..140cc13067 100644
--- a/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/pdom/indexer/PDOMUpdateTask.java
+++ b/core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/pdom/indexer/PDOMUpdateTask.java
@@ -200,7 +200,7 @@ public class PDOMUpdateTask implements IPDOMIndexerTask {
 	private static boolean canResolveUnresolvedInclude(IIndexFile file, IScannerInfo scannerInfo,
 			ProjectIndexerIncludeResolutionHeuristics includeResolutionHeuristics) {
 		try {
-			String filePath = IndexLocationFactory.getAbsolutePath(file.getLocation()).toOSString();
+			String filePath = IndexLocationFactory.getAbsolutePath(file.getLocation()).toString();
 			long fileReadTime = file.getSourceReadTime();
 			IncludeSearchPath includeSearchPath = CPreprocessor
 					.configureIncludeSearchPath(new File(filePath).getParentFile(), scannerInfo);
