diff --git a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/OrganizeIncludesAction.java b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/OrganizeIncludesAction.java
index 55bfffcc8f..fd044755a6 100644
--- a/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/OrganizeIncludesAction.java
+++ b/core/org.eclipse.cdt.ui/src/org/eclipse/cdt/internal/ui/editor/OrganizeIncludesAction.java
@@ -83,8 +83,8 @@ public class OrganizeIncludesAction extends TextEditorAction {
 						index.acquireReadLock();
 						IASTTranslationUnit ast = tu.getAST(index, ASTCache.PARSE_MODE);
 						if (ast == null) {
-							return CUIPlugin.createErrorStatus(NLS.bind(
-									CEditorMessages.OrganizeIncludes_ast_not_available, tu.getPath().toOSString()));
+							return CUIPlugin.createErrorStatus(NLS
+									.bind(CEditorMessages.OrganizeIncludes_ast_not_available, tu.getPath().toString()));
 						}
 						IncludeOrganizer organizer = new IncludeOrganizer(tu, index, headerChooser);
 						holder[0] = organizer.organizeIncludes(ast);
