diff --git a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/executables/StandardSourceFilesProvider.java b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/executables/StandardSourceFilesProvider.java
index 29752f98f4..2b493c2f95 100644
--- a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/executables/StandardSourceFilesProvider.java
+++ b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/executables/StandardSourceFilesProvider.java
@@ -51,7 +51,7 @@ public class StandardSourceFilesProvider extends PlatformObject implements ISour
 		if (!Executable.isBinaryFile(executable.getPath()))
 			return null;
 
-		File f = new File(path.toOSString());
+		File f = new File(path.toString());
 		if (f.length() == 0) {
 			return null;
 		}
