diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/sourcelookup/CSourceNotFoundEditor.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/sourcelookup/CSourceNotFoundEditor.java
index f1ef15d560..7354dffa46 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/sourcelookup/CSourceNotFoundEditor.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/sourcelookup/CSourceNotFoundEditor.java
@@ -172,7 +172,7 @@ public class CSourceNotFoundEditor extends CommonSourceNotFoundEditor {
 				tunit = (ITranslationUnit) artifact;
 				IPath tuPath = tunit.getLocation();
 				if (tuPath != null) {
-					missingFile = tuPath.toOSString();
+					missingFile = tuPath.toString();
 				}
 			} else {
 				missingFile = ""; //$NON-NLS-1$
@@ -422,7 +422,7 @@ public class CSourceNotFoundEditor extends CommonSourceNotFoundEditor {
 		if (tu != null) {
 			IPath tuPath = tu.getLocation();
 			if (tuPath != null) {
-				String filePath = tuPath.toOSString();
+				String filePath = tuPath.toString();
 				try {
 					Object[] foundElements = CDebugCorePlugin.getDefault().getCommonSourceLookupDirector()
 							.findSourceElements(filePath);
