diff --git a/debug/org.eclipse.cdt.debug.application.tests/src/org/eclipse/cdt/debug/application/tests/StandaloneTest1.java b/debug/org.eclipse.cdt.debug.application.tests/src/org/eclipse/cdt/debug/application/tests/StandaloneTest1.java
index 0f243b1ebd..7db4d11943 100644
--- a/debug/org.eclipse.cdt.debug.application.tests/src/org/eclipse/cdt/debug/application/tests/StandaloneTest1.java
+++ b/debug/org.eclipse.cdt.debug.application.tests/src/org/eclipse/cdt/debug/application/tests/StandaloneTest1.java
@@ -17,15 +17,16 @@ import static org.eclipse.swtbot.swt.finder.matchers.WidgetMatcherFactory.withMn
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 
+import java.awt.MenuItem;
+import java.util.concurrent.TimeoutException;
+
 import org.eclipse.core.runtime.IPath;
-import org.eclipse.swt.widgets.MenuItem;
 import org.eclipse.swtbot.swt.finder.junit.SWTBotJunit4ClassRunner;
 import org.eclipse.swtbot.swt.finder.waits.Conditions;
 import org.eclipse.swtbot.swt.finder.waits.WaitForObjectCondition;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotMenu;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotShell;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotText;
-import org.eclipse.swtbot.swt.finder.widgets.TimeoutException;
 import org.hamcrest.Matcher;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
@@ -78,7 +79,7 @@ public class StandaloneTest1 extends StandaloneTest {
 		shell.setFocus();
 		// Try and have two open debug sessions on same binary
 		IPath projectPath = Utilities.getDefault().getProjectPath(PROJECT_NAME).append("a.out");
-		shell.bot().textWithLabel("Binary: ").setText(projectPath.toOSString());
+		shell.bot().textWithLabel("Binary: ").setText(projectPath.toString());
 		shell.bot().textWithLabel("Arguments: ").setText("1 2 3");
 		bot.sleep(2000);
 
