diff --git a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/MapEntrySourceContainerType.java b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/MapEntrySourceContainerType.java
index 4d8c28d499..49600c4faa 100644
--- a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/MapEntrySourceContainerType.java
+++ b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/MapEntrySourceContainerType.java
@@ -68,7 +68,7 @@ public class MapEntrySourceContainerType extends AbstractSourceContainerTypeDele
 		Document document = newDocument();
 		Element element = document.createElement(ELEMENT_NAME);
 		element.setAttribute(BACKEND_PATH, entry.getBackendPathStr());
-		element.setAttribute(LOCAL_PATH, entry.getLocalPath().toOSString());
+		element.setAttribute(LOCAL_PATH, entry.getLocalPath().toString());
 		document.appendChild(element);
 		return serializeDocument(document);
 	}
