diff --git a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/CDirectorySourceLocation.java b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/CDirectorySourceLocation.java
index 004e28c97c..bba2e87852 100644
--- a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/CDirectorySourceLocation.java
+++ b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/CDirectorySourceLocation.java
@@ -103,7 +103,7 @@ public class CDirectorySourceLocation implements IDirectorySourceLocation {
 				IPath path = new Path(name);
 				if (path.segmentCount() > 1 && getAssociation().isPrefixOf(path)) {
 					path = getDirectory().append(path.removeFirstSegments(getAssociation().segmentCount()));
-					result = findFileByAbsolutePath(path.toOSString());
+					result = findFileByAbsolutePath(path.toString());
 				}
 			}
 		}
@@ -282,9 +282,9 @@ public class CDirectorySourceLocation implements IDirectorySourceLocation {
 			document = XmlProcessorFactoryCdt.createDocumentBuilderWithErrorOnDOCTYPE().newDocument();
 			Element node = document.createElement(ELEMENT_NAME);
 			document.appendChild(node);
-			node.setAttribute(ATTR_DIRECTORY, getDirectory().toOSString());
+			node.setAttribute(ATTR_DIRECTORY, getDirectory().toString());
 			if (getAssociation() != null)
-				node.setAttribute(ATTR_ASSOCIATION, getAssociation().toOSString());
+				node.setAttribute(ATTR_ASSOCIATION, getAssociation().toString());
 			node.setAttribute(ATTR_SEARCH_SUBFOLDERS, String.valueOf(searchSubfolders()));
 			return CDebugUtils.serializeDocument(document);
 		} catch (ParserConfigurationException e) {
@@ -294,7 +294,7 @@ public class CDirectorySourceLocation implements IDirectorySourceLocation {
 		} catch (TransformerException e) {
 			ex = e;
 		}
-		abort(NLS.bind(InternalSourceLookupMessages.CDirectorySourceLocation_0, getDirectory().toOSString()), ex);
+		abort(NLS.bind(InternalSourceLookupMessages.CDirectorySourceLocation_0, getDirectory().toString()), ex);
 		// execution will not reach here
 		return null;
 	}
@@ -384,8 +384,8 @@ public class CDirectorySourceLocation implements IDirectorySourceLocation {
 		int segCount = prefix.segmentCount();
 		if (segCount >= path.segmentCount())
 			return false;
-		String prefixString = prefix.toOSString();
-		String pathString = path.removeLastSegments(path.segmentCount() - segCount).toOSString();
+		String prefixString = prefix.toString();
+		String pathString = path.removeLastSegments(path.segmentCount() - segCount).toString();
 		return prefixString.equalsIgnoreCase(pathString);
 	}
 
@@ -458,7 +458,7 @@ public class CDirectorySourceLocation implements IDirectorySourceLocation {
 	 */
 	@Override
 	public String toString() {
-		return (getDirectory() != null) ? getDirectory().toOSString() : ""; //$NON-NLS-1$
+		return (getDirectory() != null) ? getDirectory().toString() : ""; //$NON-NLS-1$
 	}
 
 	/* (non-Javadoc)
