diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/breakpoints/CBreakpointPropertyPage.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/breakpoints/CBreakpointPropertyPage.java
index ec2ef12875..c8222822aa 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/breakpoints/CBreakpointPropertyPage.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/breakpoints/CBreakpointPropertyPage.java
@@ -270,7 +270,7 @@ public class CBreakpointPropertyPage extends FieldEditorPreferencePage implement
 							IFile file = (IFile) result[0];
 							IPath location = file.getRawLocation();
 							if (location != null) {
-								String newValue = location.makeAbsolute().toOSString();
+								String newValue = location.makeAbsolute().toString();
 								setStringValue(newValue);
 							}
 						}
