diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/SourceFilesViewer.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/SourceFilesViewer.java
index c9e0a9fed2..61460cb2f6 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/SourceFilesViewer.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/SourceFilesViewer.java
@@ -266,7 +266,7 @@ public class SourceFilesViewer extends BaseViewer {
 				// java.io.File would be wrong since that class makes arbitrary
 				// assumptions about where the file should be locally. See
 				// similar comment in Executable.getSourceFiles()
-				if (Util.isNativeAbsolutePath(info.location.toOSString())) {
+				if (Util.isNativeAbsolutePath(info.location.toString())) {
 					File file = info.location.toFile();
 					info.exists = file.exists();
 					if (info.exists) {
@@ -283,7 +283,7 @@ public class SourceFilesViewer extends BaseViewer {
 				}
 
 				info.originalLocation = new Path(executable.getOriginalLocation(tu));
-				info.originalExists = Util.isNativeAbsolutePath(info.originalLocation.toOSString())
+				info.originalExists = Util.isNativeAbsolutePath(info.originalLocation.toString())
 						&& info.originalLocation.toFile().exists();
 			} else {
 				info.exists = false;
