diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/DebugMarkerAnnotationModelFactory.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/DebugMarkerAnnotationModelFactory.java
index 854d662820..d2de468764 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/DebugMarkerAnnotationModelFactory.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/DebugMarkerAnnotationModelFactory.java
@@ -32,7 +32,7 @@ public class DebugMarkerAnnotationModelFactory extends ResourceMarkerAnnotationM
 		if (file != null) {
 			return super.createAnnotationModel(location);
 		}
-		File osFile = new File(location.toOSString());
+		File osFile = new File(location.toString());
 		if (osFile.exists()) {
 			return new DebugMarkerAnnotationModel(osFile);
 		}
