diff --git a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/core/executables/Executable.java b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/core/executables/Executable.java
index bb806fa6c9..2d7fe63373 100644
--- a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/core/executables/Executable.java
+++ b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/core/executables/Executable.java
@@ -136,7 +136,7 @@ public class Executable extends PlatformObject {
 	public Executable(IPath path, IProject project, IResource resource, ISourceFileRemapping[] sourceFileRemappings) {
 		this.executablePath = path;
 		this.project = project;
-		this.name = new File(path.toOSString()).getName();
+		this.name = new File(path.toString()).getName();
 		this.resource = resource;
 		this.remappers = sourceFileRemappings;
 		remappedPaths = new HashMap<>();
@@ -320,7 +320,7 @@ public class Executable extends PlatformObject {
 	public synchronized String getOriginalLocation(ITranslationUnit tu) {
 		String orgLocation = remappedPaths.get(tu);
 		if (orgLocation == null)
-			orgLocation = tu.getLocation().toOSString();
+			orgLocation = tu.getLocation().toString();
 		return orgLocation;
 	}
 
