diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/CDebugModelPresentation.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/CDebugModelPresentation.java
index b039409951..bbf9a45b23 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/CDebugModelPresentation.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/CDebugModelPresentation.java
@@ -681,7 +681,7 @@ public class CDebugModelPresentation extends LabelProvider implements IDebugMode
 				if (frame.getFile() != null) {
 					IPath path = new Path(frame.getFile());
 					if (!path.isEmpty()) {
-						label.append((qualified ? path.toOSString() : path.lastSegment()));
+						label.append((qualified ? path.toString() : path.lastSegment()));
 						label.append(':');
 						if (frame.getFrameLineNumber() != 0)
 							label.append(frame.getFrameLineNumber());
