diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/SourceFilesLabelProvider.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/SourceFilesLabelProvider.java
index e8cf86d07b..27717f1051 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/SourceFilesLabelProvider.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/SourceFilesLabelProvider.java
@@ -80,7 +80,7 @@ public class SourceFilesLabelProvider extends TreeColumnViewerLabelProvider impl
 			cell.setText(null);
 			if (tuInfo != null) {
 				if (tuInfo.location != null) {
-					cell.setText(tuInfo.location.toOSString());
+					cell.setText(tuInfo.location.toString());
 					if (tuInfo.exists)
 						cell.setForeground(Display.getDefault().getSystemColor(SWT.COLOR_BLACK));
 					else
@@ -91,7 +91,7 @@ public class SourceFilesLabelProvider extends TreeColumnViewerLabelProvider impl
 		} else if (orgColumnIndex == 2) {
 			cell.setText(null);
 			if (tuInfo != null && tuInfo.originalLocation != null) {
-				cell.setText(tuInfo.originalLocation.toOSString());
+				cell.setText(tuInfo.originalLocation.toString());
 				if (tuInfo.originalExists)
 					cell.setForeground(Display.getDefault().getSystemColor(SWT.COLOR_BLACK));
 				else
