diff --git a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/CSourceLocator.java b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/CSourceLocator.java
index d4eb106dae..6a0b3054c1 100644
--- a/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/CSourceLocator.java
+++ b/debug/org.eclipse.cdt.debug.core/src/org/eclipse/cdt/debug/internal/core/sourcelookup/CSourceLocator.java
@@ -150,7 +150,7 @@ public class CSourceLocator implements ICSourceLocator, IPersistableSourceLocato
 			}
 			if (resource instanceof IFile) {
 				try {
-					Object result = locations[i].findSourceElement(resource.getLocation().toOSString());
+					Object result = locations[i].findSourceElement(resource.getLocation().toString());
 					if (result instanceof IFile && ((IFile) result).equals(resource))
 						return true;
 					if (result instanceof List && ((List<?>) result).contains(resource))
