diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/ExecutablesViewerComparator.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/ExecutablesViewerComparator.java
index 28e0361b5f..a7f77cfd89 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/ExecutablesViewerComparator.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/views/executables/ExecutablesViewerComparator.java
@@ -57,7 +57,7 @@ class ExecutablesViewerComparator extends ViewerComparator {
 			}
 
 			if (sortType == ExecutablesView.LOCATION) {
-				return getComparator().compare(getPath(e1).toOSString(), getPath(e2).toOSString()) * columnOrder;
+				return getComparator().compare(getPath(e1).toString(), getPath(e2).toString()) * columnOrder;
 			}
 
 			if (sortType == ExecutablesView.MODIFIED) {
