diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/importexecutable/ImportExecutablePageOne.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/importexecutable/ImportExecutablePageOne.java
index e05e9b78c9..310deeac80 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/importexecutable/ImportExecutablePageOne.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/importexecutable/ImportExecutablePageOne.java
@@ -547,7 +547,7 @@ public class ImportExecutablePageOne extends WizardPage {
 		if (dirName.length() > 0) {
 			File path = new File(dirName);
 			if (path.exists())
-				dialog.setFilterPath(new Path(dirName).toOSString());
+				dialog.setFilterPath(new Path(dirName).toString());
 		}
 
 		String selectedDirectory = dialog.open();
