diff --git a/dsf-gdb/org.eclipse.cdt.dsf.gdb.ui/src/org/eclipse/cdt/dsf/gdb/internal/ui/commands/GdbConnectCommand.java b/dsf-gdb/org.eclipse.cdt.dsf.gdb.ui/src/org/eclipse/cdt/dsf/gdb/internal/ui/commands/GdbConnectCommand.java
index 538bc46e8d..17e49bff9c 100644
--- a/dsf-gdb/org.eclipse.cdt.dsf.gdb.ui/src/org/eclipse/cdt/dsf/gdb/internal/ui/commands/GdbConnectCommand.java
+++ b/dsf-gdb/org.eclipse.cdt.dsf.gdb.ui/src/org/eclipse/cdt/dsf/gdb/internal/ui/commands/GdbConnectCommand.java
@@ -607,7 +607,7 @@ public class GdbConnectCommand extends RefreshableDebugCommand implements IConne
 											IPath binaryPath = backend.getProgramPath();
 											if (binaryPath != null && !binaryPath.isEmpty()) {
 												fProcessNameToBinaryMap.put(binaryPath.lastSegment(),
-														binaryPath.toOSString());
+														binaryPath.toString());
 											}
 										}
 
