diff --git a/dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/service/GDBBackend.java b/dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/service/GDBBackend.java
index 3faebeb1a4..f26ee0fca7 100644
--- a/dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/service/GDBBackend.java
+++ b/dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/service/GDBBackend.java
@@ -185,7 +185,7 @@ public class GDBBackend extends AbstractDsfService implements IGDBBackend, IMIBa
 		// All configuration should be done in the final launch sequence
 		// to allow for more flexibility.
 
-		String cmd = getGDBPath().toOSString() + " --interpreter" + //$NON-NLS-1$
+		String cmd = getGDBPath().toString() + " --interpreter" + //$NON-NLS-1$
 		// We currently work with MI version 2. Don't use just 'mi' because it
 		// points to the latest MI version, while we want mi2 specifically.
 				" mi2" + //$NON-NLS-1$
