diff --git a/dsf-gdb/org.eclipse.cdt.dsf.gdb.ui/src/org/eclipse/cdt/dsf/gdb/internal/ui/launching/SolibSearchPathBlock.java b/dsf-gdb/org.eclipse.cdt.dsf.gdb.ui/src/org/eclipse/cdt/dsf/gdb/internal/ui/launching/SolibSearchPathBlock.java
index c64bd742b2..1f4398a91b 100644
--- a/dsf-gdb/org.eclipse.cdt.dsf.gdb.ui/src/org/eclipse/cdt/dsf/gdb/internal/ui/launching/SolibSearchPathBlock.java
+++ b/dsf-gdb/org.eclipse.cdt.dsf.gdb.ui/src/org/eclipse/cdt/dsf/gdb/internal/ui/launching/SolibSearchPathBlock.java
@@ -260,7 +260,7 @@ public class SolibSearchPathBlock extends Observable implements IMILaunchConfigu
 			@Override
 			public String getText(Object element) {
 				if (element instanceof IPath)
-					return ((IPath) element).toOSString();
+					return ((IPath) element).toString();
 				return super.getText(element);
 			}
 		};
@@ -364,7 +364,7 @@ public class SolibSearchPathBlock extends Observable implements IMILaunchConfigu
 			ArrayList<String> values = new ArrayList<>(elements.size());
 			Iterator<IPath> it = elements.iterator();
 			while (it.hasNext()) {
-				values.add((it.next()).toOSString());
+				values.add((it.next()).toString());
 			}
 			configuration.setAttribute(IGDBLaunchConfigurationConstants.ATTR_DEBUGGER_SOLIB_PATH, values);
 		}
