diff --git a/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/ui/sourcelookup/DsfSourceDisplayAdapter.java b/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/ui/sourcelookup/DsfSourceDisplayAdapter.java
index d2c9f815c5..19b8f7fb24 100644
--- a/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/ui/sourcelookup/DsfSourceDisplayAdapter.java
+++ b/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/ui/sourcelookup/DsfSourceDisplayAdapter.java
@@ -471,7 +471,7 @@ public class DsfSourceDisplayAdapter implements ISourceDisplay, ISteppingControl
 						ITranslationUnit tunit = (ITranslationUnit) artifact;
 						IPath tuPath = tunit.getLocation();
 						if (tuPath != null)
-							missingFile = tuPath.toOSString();
+							missingFile = tuPath.toString();
 					} else {
 						missingFile = ""; //$NON-NLS-1$
 					}
