diff --git a/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/ui/actions/AbstractDisassemblyBreakpointsTarget.java b/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/ui/actions/AbstractDisassemblyBreakpointsTarget.java
index 6e8926e599..686f56d554 100644
--- a/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/ui/actions/AbstractDisassemblyBreakpointsTarget.java
+++ b/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/ui/actions/AbstractDisassemblyBreakpointsTarget.java
@@ -333,10 +333,10 @@ public abstract class AbstractDisassemblyBreakpointsTarget
 				if (location == null) {
 					return;
 				}
-				filePath = location.toOSString();
+				filePath = location.toString();
 			} else {
 				resource = ResourcesPlugin.getWorkspace().getRoot();
-				filePath = URIUtil.toPath(fileUri).toOSString();
+				filePath = URIUtil.toPath(fileUri).toString();
 			}
 			int srcLine = selection.getSourceLine();
 			if (interactive) {
