diff --git a/jsoncdb/org.eclipse.cdt.jsoncdb.core.tests/src/org/eclipse/cdt/jsoncdb/core/internal/MacrosFile_GCCTest.java b/jsoncdb/org.eclipse.cdt.jsoncdb.core.tests/src/org/eclipse/cdt/jsoncdb/core/internal/MacrosFile_GCCTest.java
index 0b72350ad7..d7f11b774a 100644
--- a/jsoncdb/org.eclipse.cdt.jsoncdb.core.tests/src/org/eclipse/cdt/jsoncdb/core/internal/MacrosFile_GCCTest.java
+++ b/jsoncdb/org.eclipse.cdt.jsoncdb.core.tests/src/org/eclipse/cdt/jsoncdb/core/internal/MacrosFile_GCCTest.java
@@ -82,7 +82,7 @@ public class MacrosFile_GCCTest {
 		parsed = result.getMacroFiles().get(0);
 		assertEquals("name", name, parsed);
 
-		name = (new Path("A:an\\Macros/file.inc")).toOSString();
+		name = (new Path("A:an\\Macros/file.inc")).toString();
 		// -imacrosA:an\Macros/file.inc
 		result = new ParseContext();
 		assertEquals(8 + name.length(), testee.processArgument(result, cwd, "-imacros" + name + more));
@@ -131,7 +131,7 @@ public class MacrosFile_GCCTest {
 		parsed = result.getMacroFiles().get(0);
 		assertEquals("name", name, parsed);
 
-		name = (new Path("A:an\\Ma cr os/fi le.inc")).toOSString();
+		name = (new Path("A:an\\Ma cr os/fi le.inc")).toString();
 		// -imacros'A:an\Ma cr os/fi le.inc'
 		result = new ParseContext();
 		assertEquals(8 + name.length() + 2,
