diff --git a/jsoncdb/org.eclipse.cdt.jsoncdb.nvidia.tests/src/org/eclipse/cdt/jsoncdb/nvidia/NvccSystemIncludePathTest.java b/jsoncdb/org.eclipse.cdt.jsoncdb.nvidia.tests/src/org/eclipse/cdt/jsoncdb/nvidia/NvccSystemIncludePathTest.java
index 76ac388c2b..7a3f2c7e1e 100644
--- a/jsoncdb/org.eclipse.cdt.jsoncdb.nvidia.tests/src/org/eclipse/cdt/jsoncdb/nvidia/NvccSystemIncludePathTest.java
+++ b/jsoncdb/org.eclipse.cdt.jsoncdb.nvidia.tests/src/org/eclipse/cdt/jsoncdb/nvidia/NvccSystemIncludePathTest.java
@@ -55,7 +55,7 @@ public class NvccSystemIncludePathTest {
 		assertEquals("#entries", 1, result.getSystemIncludePaths().size());
 		assertEquals("name", name, result.getSystemIncludePaths().get(0));
 
-		name = (new Path("A:an\\In CLU  de/Pat h")).toOSString();
+		name = (new Path("A:an\\In CLU  de/Pat h")).toString();
 		// -isystem="A:an\In CLU de/Pat h"
 		result = tcp.processArgs(cwd, "-isystem" + "=" + "\"" + name + "\"" + more);
 		assertEquals("#entries", 1, result.getSystemIncludePaths().size());
