diff --git a/jsoncdb/org.eclipse.cdt.jsoncdb.core.tests/src/org/eclipse/cdt/jsoncdb/core/internal/IncludeFile_GCCTest.java b/jsoncdb/org.eclipse.cdt.jsoncdb.core.tests/src/org/eclipse/cdt/jsoncdb/core/internal/IncludeFile_GCCTest.java
index e3fdc91de3..23c15acd2d 100644
--- a/jsoncdb/org.eclipse.cdt.jsoncdb.core.tests/src/org/eclipse/cdt/jsoncdb/core/internal/IncludeFile_GCCTest.java
+++ b/jsoncdb/org.eclipse.cdt.jsoncdb.core.tests/src/org/eclipse/cdt/jsoncdb/core/internal/IncludeFile_GCCTest.java
@@ -82,7 +82,7 @@ public class IncludeFile_GCCTest {
 		parsed = result.getIncludeFiles().get(0);
 		assertEquals("name", name, parsed);
 
-		name = (new Path("A:an\\Include/file.inc")).toOSString();
+		name = (new Path("A:an\\Include/file.inc")).toString();
 		// -includeA:an\Include/file.inc
 		result = new ParseContext();
 		assertEquals(8 + name.length(), testee.processArgument(result, cwd, "-include" + name + more));
@@ -131,7 +131,7 @@ public class IncludeFile_GCCTest {
 		parsed = result.getIncludeFiles().get(0);
 		assertEquals("name", name, parsed);
 
-		name = (new Path("A:an\\In CLU  de/fi le.inc")).toOSString();
+		name = (new Path("A:an\\In CLU  de/fi le.inc")).toString();
 		// -include'A:an\In CLU de/fi le.inc'
 		result = new ParseContext();
 		assertEquals(8 + name.length() + 2,
