diff --git a/jsoncdb/org.eclipse.cdt.jsoncdb.core/src/org/eclipse/cdt/jsoncdb/core/participant/ResponseFileArglets.java b/jsoncdb/org.eclipse.cdt.jsoncdb.core/src/org/eclipse/cdt/jsoncdb/core/participant/ResponseFileArglets.java
index 16f162eb2a..21a552ebbb 100644
--- a/jsoncdb/org.eclipse.cdt.jsoncdb.core/src/org/eclipse/cdt/jsoncdb/core/participant/ResponseFileArglets.java
+++ b/jsoncdb/org.eclipse.cdt.jsoncdb.core/src/org/eclipse/cdt/jsoncdb/core/participant/ResponseFileArglets.java
@@ -71,7 +71,7 @@ public class ResponseFileArglets {
 					IPath path = Path.fromOSString(fname);
 					if (!path.isAbsolute()) {
 						// relative path, prepend CWD
-						fname = parserHandler.getCompilerWorkingDirectory().append(path).toOSString();
+						fname = parserHandler.getCompilerWorkingDirectory().append(path).toString();
 					}
 
 					// parse file
