diff --git a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/LocalRunLaunchDelegate.java b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/LocalRunLaunchDelegate.java
index 62f8f3f9e5..b269512a66 100644
--- a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/LocalRunLaunchDelegate.java
+++ b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/LocalRunLaunchDelegate.java
@@ -101,7 +101,7 @@ public class LocalRunLaunchDelegate extends AbstractCLaunchDelegate2 {
 
 			String[] arguments = CommandLineUtil.argumentsToArray(args);
 			ArrayList<String> command = new ArrayList<>(1 + arguments.length);
-			command.add(exePath.toOSString());
+			command.add(exePath.toString());
 			command.addAll(Arrays.asList(arguments));
 			monitor.worked(2);
 
