diff --git a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/CommonTabLite.java b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/CommonTabLite.java
index d1e5557547..fc90639f61 100644
--- a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/CommonTabLite.java
+++ b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/CommonTabLite.java
@@ -243,7 +243,7 @@ public class CommonTabLite extends AbstractLaunchConfigurationTab {
 				for (int i = 0; i < res.length; i++) {
 					proj = res[i].getProject();
 					if (proj != null && proj.isAccessible()) {
-						return proj.getFullPath().toOSString();
+						return proj.getFullPath().toString();
 					}
 				}
 			}
@@ -274,7 +274,7 @@ public class CommonTabLite extends AbstractLaunchConfigurationTab {
 		Object[] results = dialog.getResult();
 		if ((results != null) && (results.length > 0) && (results[0] instanceof IPath)) {
 			IPath path = (IPath) results[0];
-			String containerName = path.toOSString();
+			String containerName = path.toString();
 			fSharedLocationText.setText(containerName);
 		}
 	}
@@ -302,7 +302,7 @@ public class CommonTabLite extends AbstractLaunchConfigurationTab {
 			if (file != null) {
 				IContainer parent = file.getParent();
 				if (parent != null) {
-					containerName = parent.getFullPath().toOSString();
+					containerName = parent.getFullPath().toString();
 				}
 			}
 			fSharedLocationText.setText(containerName);
