diff --git a/launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/internal/docker/launcher/ContainerLaunchConfigurationDelegate.java b/launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/internal/docker/launcher/ContainerLaunchConfigurationDelegate.java
index 15ba069db6..695aae6e26 100644
--- a/launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/internal/docker/launcher/ContainerLaunchConfigurationDelegate.java
+++ b/launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/internal/docker/launcher/ContainerLaunchConfigurationDelegate.java
@@ -180,7 +180,7 @@ public class ContainerLaunchConfigurationDelegate extends GdbLaunchDelegate {
 				// the project
 				if (workingDir == null && projectName != null) {
 					IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectName);
-					workingDir = project.getLocation().toOSString();
+					workingDir = project.getLocation().toString();
 				}
 
 				if (workingDir != null) {
@@ -313,7 +313,7 @@ public class ContainerLaunchConfigurationDelegate extends GdbLaunchDelegate {
 				// the project
 				if (workingDir == null && projectName != null) {
 					IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectName);
-					workingDir = project.getLocation().toOSString();
+					workingDir = project.getLocation().toString();
 				}
 				if (workingDir != null) {
 					IPath workingPath = new Path(workingDir);
