diff --git a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/CAbstractMainTab.java b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/CAbstractMainTab.java
index 98ae3a3c20..76da27035e 100644
--- a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/CAbstractMainTab.java
+++ b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/ui/CAbstractMainTab.java
@@ -188,7 +188,7 @@ public abstract class CAbstractMainTab extends CLaunchConfigurationTab {
 			// New project selected for the first time, set the program name default too.
 			IBinary[] bins = getBinaryFiles(project);
 			if (bins != null && bins.length == 1) {
-				fProgText.setText(bins[0].getResource().getProjectRelativePath().toOSString());
+				fProgText.setText(bins[0].getResource().getProjectRelativePath().toString());
 			}
 		}
 	}
