diff --git a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/refactoring/ProjectRenameChange.java b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/refactoring/ProjectRenameChange.java
index b9c748f6e8..e218425394 100644
--- a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/refactoring/ProjectRenameChange.java
+++ b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/refactoring/ProjectRenameChange.java
@@ -132,7 +132,7 @@ class ProjectRenameChange extends AbstractLaunchConfigChange {
 			pathProgName = pathProgName.removeLastSegments(1).append(newName);
 			if (progExtension != null)
 				pathProgName = pathProgName.addFileExtension(progExtension);
-			copy.setAttribute(ICDTLaunchConfigurationConstants.ATTR_PROGRAM_NAME, pathProgName.toOSString());
+			copy.setAttribute(ICDTLaunchConfigurationConstants.ATTR_PROGRAM_NAME, pathProgName.toString());
 		}
 
 		try {
