diff --git a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/ui/WorkingDirectoryBlock.java b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/ui/WorkingDirectoryBlock.java
index 1fceb6fc67..f37c80196e 100644
--- a/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/ui/WorkingDirectoryBlock.java
+++ b/launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/ui/WorkingDirectoryBlock.java
@@ -230,7 +230,7 @@ public class WorkingDirectoryBlock extends CLaunchConfigurationTab {
 			if (config != null) {
 				ICProject cProject = CDebugUtils.getCProject(config);
 				if (cProject != null) {
-					fWorkingDirText.setText("${workspace_loc:" + cProject.getPath().makeRelative().toOSString() + "}"); //$NON-NLS-1$ //$NON-NLS-2$
+					fWorkingDirText.setText("${workspace_loc:" + cProject.getPath().makeRelative().toString() + "}"); //$NON-NLS-1$ //$NON-NLS-2$
 					return;
 				}
 			}
