diff --git a/llvm/org.eclipse.cdt.managedbuilder.llvm.ui/src/org/eclipse/cdt/managedbuilder/llvm/util/LlvmToolOptionPathUtil.java b/llvm/org.eclipse.cdt.managedbuilder.llvm.ui/src/org/eclipse/cdt/managedbuilder/llvm/util/LlvmToolOptionPathUtil.java
index 6cbde1e50d..d57e0ad09d 100644
--- a/llvm/org.eclipse.cdt.managedbuilder.llvm.ui/src/org/eclipse/cdt/managedbuilder/llvm/util/LlvmToolOptionPathUtil.java
+++ b/llvm/org.eclipse.cdt.managedbuilder.llvm.ui/src/org/eclipse/cdt/managedbuilder/llvm/util/LlvmToolOptionPathUtil.java
@@ -171,7 +171,7 @@ public class LlvmToolOptionPathUtil {
 			IConfiguration[] configs;
 			String projectPath = null;
 			for (IProject proj : projects) {
-				projectPath = proj.getLocation().toOSString();
+				projectPath = proj.getLocation().toString();
 				if (projectPath != null) {
 					//only apply to C++ projects
 					if (FileUtil.containsCppFile(new File(projectPath))) {
