diff --git a/llvm/org.eclipse.cdt.managedbuilder.llvm.ui/src/org/eclipse/cdt/managedbuilder/llvm/ui/LlvmEnvironmentVariableSupplier.java b/llvm/org.eclipse.cdt.managedbuilder.llvm.ui/src/org/eclipse/cdt/managedbuilder/llvm/ui/LlvmEnvironmentVariableSupplier.java
index 4a0aec2207..03af008e37 100644
--- a/llvm/org.eclipse.cdt.managedbuilder.llvm.ui/src/org/eclipse/cdt/managedbuilder/llvm/ui/LlvmEnvironmentVariableSupplier.java
+++ b/llvm/org.eclipse.cdt.managedbuilder.llvm.ui/src/org/eclipse/cdt/managedbuilder/llvm/ui/LlvmEnvironmentVariableSupplier.java
@@ -324,7 +324,7 @@ public class LlvmEnvironmentVariableSupplier implements IConfigurationEnvironmen
 		// get mingw bin path
 		IPath mingwBinPath = getBinDir();
 		if (mingwBinPath != null) {
-			StringBuilder sB = new StringBuilder(mingwBinPath.toOSString());
+			StringBuilder sB = new StringBuilder(mingwBinPath.toString());
 			// drop bin
 			if (sB.length() >= 3) {
 				sB.delete(sB.length() - 3, sB.length());
