diff --git a/terminal/plugins/org.eclipse.tm.terminal.connector.local/src/org/eclipse/tm/terminal/connector/local/launcher/LocalLauncherDelegate.java b/terminal/plugins/org.eclipse.tm.terminal.connector.local/src/org/eclipse/tm/terminal/connector/local/launcher/LocalLauncherDelegate.java
index a5c775b06b..3e69ee9f13 100644
--- a/terminal/plugins/org.eclipse.tm.terminal.connector.local/src/org/eclipse/tm/terminal/connector/local/launcher/LocalLauncherDelegate.java
+++ b/terminal/plugins/org.eclipse.tm.terminal.connector.local/src/org/eclipse/tm/terminal/connector/local/launcher/LocalLauncherDelegate.java
@@ -132,7 +132,7 @@ public class LocalLauncherDelegate extends AbstractLauncherDelegate {
 							&& org.eclipse.core.resources.ResourcesPlugin.getWorkspace().getRoot()
 									.getLocation() != null) {
 						cwd = org.eclipse.core.resources.ResourcesPlugin.getWorkspace().getRoot().getLocation()
-								.toOSString();
+								.toString();
 					}
 				}
 			} else {
@@ -143,7 +143,7 @@ public class LocalLauncherDelegate extends AbstractLauncherDelegate {
 
 					IPath p = new Path(resolved);
 					if (p.toFile().canRead() && p.toFile().isDirectory()) {
-						cwd = p.toOSString();
+						cwd = p.toString();
 					}
 				} catch (CoreException ex) {
 					if (Platform.inDebugMode()) {
@@ -378,7 +378,7 @@ public class LocalLauncherDelegate extends AbstractLauncherDelegate {
 					&& org.eclipse.core.resources.ResourcesPlugin.getWorkspace().getRoot() != null
 					&& org.eclipse.core.resources.ResourcesPlugin.getWorkspace().getRoot().getLocation() != null) {
 				envpList.add("ECLIPSE_WORKSPACE=" + org.eclipse.core.resources.ResourcesPlugin.getWorkspace().getRoot() //$NON-NLS-1$
-						.getLocation().toOSString());
+						.getLocation().toString());
 			}
 		}
 
