diff --git a/terminal/plugins/org.eclipse.tm.terminal.view.ui/src/org/eclipse/tm/terminal/view/ui/local/showin/ExternalExecutablesManager.java b/terminal/plugins/org.eclipse.tm.terminal.view.ui/src/org/eclipse/tm/terminal/view/ui/local/showin/ExternalExecutablesManager.java
index ca5c403a08..45bbe7731b 100644
--- a/terminal/plugins/org.eclipse.tm.terminal.view.ui/src/org/eclipse/tm/terminal/view/ui/local/showin/ExternalExecutablesManager.java
+++ b/terminal/plugins/org.eclipse.tm.terminal.view.ui/src/org/eclipse/tm/terminal/view/ui/local/showin/ExternalExecutablesManager.java
@@ -30,8 +30,6 @@ import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Platform;
 import org.eclipse.tm.terminal.view.ui.activator.UIPlugin;
 import org.eclipse.tm.terminal.view.ui.internal.ExternalExecutablesState;
-import org.eclipse.tm.terminal.view.ui.local.showin.detectors.DetectGitBash;
-import org.eclipse.tm.terminal.view.ui.local.showin.detectors.DetectWSL;
 import org.eclipse.ui.PlatformUI;
 import org.eclipse.ui.services.ISourceProviderService;
 
@@ -40,7 +38,7 @@ import org.eclipse.ui.services.ISourceProviderService;
  */
 public class ExternalExecutablesManager {
 	// XXX: This may make a useful extension point?
-	private static List<IDetectExternalExecutable> detectors = List.of(new DetectGitBash(), new DetectWSL());
+	private static List<IDetectExternalExecutable> detectors = new ArrayList<>();
 
 	public static boolean hasEntries() {
 		IPath stateLocation = UIPlugin.getDefault().getStateLocation();
