diff --git a/terminal/plugins/org.eclipse.tm.terminal.view.ui/src/org/eclipse/tm/terminal/view/ui/preferences/PreferencePage.java b/terminal/plugins/org.eclipse.tm.terminal.view.ui/src/org/eclipse/tm/terminal/view/ui/preferences/PreferencePage.java
index ff85cec8e7..4905c15114 100644
--- a/terminal/plugins/org.eclipse.tm.terminal.view.ui/src/org/eclipse/tm/terminal/view/ui/preferences/PreferencePage.java
+++ b/terminal/plugins/org.eclipse.tm.terminal.view.ui/src/org/eclipse/tm/terminal/view/ui/preferences/PreferencePage.java
@@ -156,17 +156,17 @@ public class PreferencePage extends org.eclipse.jface.preference.PreferencePage
 						IPath p = new Path(text);
 
 						if (p.toFile().isFile() || !p.toFile().exists()) {
-							dialog.setFilterPath(p.removeLastSegments(1).toOSString());
+							dialog.setFilterPath(p.removeLastSegments(1).toString());
 							dialog.setFileName(p.lastSegment());
 						} else if (p.toFile().isDirectory()) {
-							dialog.setFilterPath(p.toOSString());
+							dialog.setFilterPath(p.toString());
 						}
 					}
 
 					String selected = dialog.open();
 					if (selected != null) {
 						IPath sp = new Path(selected);
-						command.setText(sp.toOSString());
+						command.setText(sp.toString());
 					}
 				}
 			});
@@ -174,7 +174,7 @@ public class PreferencePage extends org.eclipse.jface.preference.PreferencePage
 			String cmd = UIPlugin.getScopedPreferences()
 					.getString(IPreferenceKeys.PREF_LOCAL_TERMINAL_DEFAULT_SHELL_UNIX);
 			if (cmd != null && !"".equals(cmd)) { //$NON-NLS-1$
-				command.setText(new Path(cmd).toOSString());
+				command.setText(new Path(cmd).toString());
 			}
 
 			Composite argsPanel = new Composite(group, SWT.NONE);
@@ -291,11 +291,11 @@ public class PreferencePage extends org.eclipse.jface.preference.PreferencePage
 				// Determine the filter path
 				String text = workingDir.getText();
 				if (Messages.PreferencePage_workingDir_userhome_label.equals(text)) {
-					dialog.setFilterPath(uh.toOSString());
+					dialog.setFilterPath(uh.toString());
 				} else if (Messages.PreferencePage_workingDir_eclipsehome_label.equals(text)) {
-					dialog.setFilterPath(eh.toOSString());
+					dialog.setFilterPath(eh.toString());
 				} else if (Messages.PreferencePage_workingDir_eclipsews_label.equals(text)) {
-					dialog.setFilterPath(ew.toOSString());
+					dialog.setFilterPath(ew.toString());
 				} else if (text != null && !"".equals(text.trim())) { //$NON-NLS-1$
 					try {
 						// Resolve possible dynamic variables
@@ -320,7 +320,7 @@ public class PreferencePage extends org.eclipse.jface.preference.PreferencePage
 					} else if (ew.equals(sp)) {
 						workingDir.select(2);
 					} else {
-						workingDir.setText(sp.toOSString());
+						workingDir.setText(sp.toString());
 					}
 				}
 			}
@@ -363,7 +363,7 @@ public class PreferencePage extends org.eclipse.jface.preference.PreferencePage
 				|| "${workspace_loc}".equals(initialCwd)) { //$NON-NLS-1$
 			workingDir.select(2);
 		} else {
-			workingDir.setText(new Path(initialCwd).toOSString());
+			workingDir.setText(new Path(initialCwd).toString());
 		}
 
 		NoteCompositeHelper.createNoteComposite(group.getFont(), group, Messages.PreferencePage_workingDir_note_label,
@@ -617,7 +617,7 @@ public class PreferencePage extends org.eclipse.jface.preference.PreferencePage
 				|| "${workspace_loc}".equals(initialCwd)) { //$NON-NLS-1$
 			workingDir.select(2);
 		} else {
-			workingDir.setText(new Path(initialCwd).toOSString());
+			workingDir.setText(new Path(initialCwd).toString());
 		}
 
 		executables.clear();
@@ -635,7 +635,7 @@ public class PreferencePage extends org.eclipse.jface.preference.PreferencePage
 			String text = command.getText();
 			IPath p = new Path(text.trim());
 			UIPlugin.getScopedPreferences().putString(IPreferenceKeys.PREF_LOCAL_TERMINAL_DEFAULT_SHELL_UNIX,
-					p.toFile().isFile() && p.toFile().canRead() && p.toFile().canExecute() ? p.toOSString() : null);
+					p.toFile().isFile() && p.toFile().canRead() && p.toFile().canExecute() ? p.toString() : null);
 
 			text = arguments.getText();
 			UIPlugin.getScopedPreferences().putString(IPreferenceKeys.PREF_LOCAL_TERMINAL_DEFAULT_SHELL_UNIX_ARGS,
