diff --git a/terminal/plugins/org.eclipse.tm.terminal.connector.process/src/org/eclipse/tm/terminal/connector/process/ProcessConnector.java b/terminal/plugins/org.eclipse.tm.terminal.connector.process/src/org/eclipse/tm/terminal/connector/process/ProcessConnector.java
index 7b51918944..7f32b96223 100644
--- a/terminal/plugins/org.eclipse.tm.terminal.connector.process/src/org/eclipse/tm/terminal/connector/process/ProcessConnector.java
+++ b/terminal/plugins/org.eclipse.tm.terminal.connector.process/src/org/eclipse/tm/terminal/connector/process/ProcessConnector.java
@@ -172,7 +172,7 @@ public class ProcessConnector extends AbstractStreamsConnector {
 							pty);
 				} else {
 					// No PTY -> just execute via the standard Java Runtime implementation.
-					process = Runtime.getRuntime().exec(command.toString(), envp, workingDir);
+					process = ProcessFactory.getFactory().exec(command.toString(), envp, workingDir);
 				}
 			}
 
