diff --git a/windows/org.eclipse.cdt.msw.build/src/org/eclipse/cdt/internal/msw/build/ProcessOutputUtil.java b/windows/org.eclipse.cdt.msw.build/src/org/eclipse/cdt/internal/msw/build/ProcessOutputUtil.java
index 0cce6f5b3a..118ce5abb8 100644
--- a/windows/org.eclipse.cdt.msw.build/src/org/eclipse/cdt/internal/msw/build/ProcessOutputUtil.java
+++ b/windows/org.eclipse.cdt.msw.build/src/org/eclipse/cdt/internal/msw/build/ProcessOutputUtil.java
@@ -16,6 +16,8 @@ import java.io.InputStreamReader;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.eclipse.cdt.utils.spawner.ProcessFactory;
+
 /**
  * Simple util for getting output out of a command. Only meant to be used in a few places to help detect things in the environment.
  */
@@ -23,7 +25,8 @@ final class ProcessOutputUtil {
 
 	static String[] getAllOutputFromCommand(String... command) {
 		try {
-			Process p = new ProcessBuilder(command).start();
+			String[] cmdarray = command;
+			Process p = ProcessFactory.getFactory().exec(cmdarray, null/*envp*/);
 			List<String> lines = new ArrayList<>();
 			try (BufferedReader input = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
 				String line = input.readLine();
