diff --git a/clang/lib/Headers/limits.h b/clang/lib/Headers/limits.h
index 354e031a9d7b..fac0b26304ab 100644
--- a/clang/lib/Headers/limits.h
+++ b/clang/lib/Headers/limits.h
@@ -113,7 +113,7 @@
    that we don't have something like #pragma poison that could be used to
    deprecate a macro - the code should just use LLONG_MAX and friends.
  */
-#if defined(__GNU_LIBRARY__) ? defined(__USE_GNU) : !defined(__STRICT_ANSI__)
+#if defined(__GNU_LIBRARY__) ? defined(__USE_GNU) : defined(__NEWLIB__) ? __GNU_VISIBLE : !defined(__STRICT_ANSI__)
 
 #undef   LONG_LONG_MIN
 #undef   LONG_LONG_MAX
