diff --git a/clang/lib/CodeGen/CGCleanup.h b/clang/lib/CodeGen/CGCleanup.h
index 079a3e25d6dc..8ed4200308f3 100644
--- a/clang/lib/CodeGen/CGCleanup.h
+++ b/clang/lib/CodeGen/CGCleanup.h
@@ -603,7 +603,9 @@ struct EHPersonality {
   static const EHPersonality GNU_ObjC_SJLJ;
   static const EHPersonality GNU_ObjC_SEH;
   static const EHPersonality GNUstep_ObjC;
+  static const EHPersonality GNUstep_ObjC_SEH;
   static const EHPersonality GNU_ObjCXX;
+  static const EHPersonality GNU_ObjCXX_SEH;
   static const EHPersonality NeXT_ObjC;
   static const EHPersonality GNU_CPlusPlus;
   static const EHPersonality GNU_CPlusPlus_SJLJ;
diff --git a/clang/lib/CodeGen/CGException.cpp b/clang/lib/CodeGen/CGException.cpp
index 8b0776b9a52b..5fd7bd3c51c2 100644
--- a/clang/lib/CodeGen/CGException.cpp
+++ b/clang/lib/CodeGen/CGException.cpp
@@ -116,8 +116,12 @@ EHPersonality::GNU_ObjC_SEH = {"__gnu_objc_personality_seh0", "objc_exception_th
 const EHPersonality
 EHPersonality::GNU_ObjCXX = { "__gnustep_objcxx_personality_v0", nullptr };
 const EHPersonality
+EHPersonality::GNU_ObjCXX_SEH = { "__gnustep_objcxx_personality_seh0", nullptr };
+const EHPersonality
 EHPersonality::GNUstep_ObjC = { "__gnustep_objc_personality_v0", nullptr };
 const EHPersonality
+EHPersonality::GNUstep_ObjC_SEH = { "__gnustep_objc_personality_seh0", nullptr };
+const EHPersonality
 EHPersonality::MSVC_except_handler = { "_except_handler3", nullptr };
 const EHPersonality
 EHPersonality::MSVC_C_specific_handler = { "__C_specific_handler", nullptr };
@@ -157,6 +161,8 @@ static const EHPersonality &getObjCPersonality(const TargetInfo &Target,
     return EHPersonality::NeXT_ObjC;
   case ObjCRuntime::GNUstep:
     if (L.ObjCRuntime.getVersion() >= VersionTuple(1, 7))
+      if (L.hasSjLjExceptions())
+        return EHPersonality::GNUstep_ObjC_SEH;
       return EHPersonality::GNUstep_ObjC;
     [[fallthrough]];
   case ObjCRuntime::GCC:
@@ -210,6 +216,8 @@ static const EHPersonality &getObjCXXPersonality(const TargetInfo &Target,
     return getObjCPersonality(Target, L);
 
   case ObjCRuntime::GNUstep:
+    if (L.hasSjLjExceptions())
+      return EHPersonality::GNU_ObjCXX_SEH;
     return EHPersonality::GNU_ObjCXX;
 
   // The GCC runtime's personality function inherently doesn't support
