diff --git a/intl/configure b/intl/configure
index e75f072f8fa..028ed36434b 100755
--- a/intl/configure
+++ b/intl/configure
@@ -6892,6 +6892,12 @@ case "${CFLAGS}" in
 	;;
 esac
 
+case $host in
+  *-cygwin | *-mingw*)
+    PICFLAG=
+    ;;
+esac
+
 
 
 BISON3_YES='#'
diff --git a/intl/configure.ac b/intl/configure.ac
index 77e2fd2d8c5..c11a0690685 100644
--- a/intl/configure.ac
+++ b/intl/configure.ac
@@ -49,6 +49,13 @@ esac
 # intl is sometimes linked into shared libraries even without --enable-shared
 # (e.g. gdbsupport's inprocess agent): so always PICify, just in case.
 GCC_PICFLAG
+
+case $host in
+  *-cygwin | *-mingw*)
+    PICFLAG=
+    ;;
+esac
+
 AC_SUBST(PICFLAG)
 
 BISON3_YES='#'
diff --git a/libdecnumber/configure b/libdecnumber/configure
index 87f3bd5e457..8267cc77d6f 100755
--- a/libdecnumber/configure
+++ b/libdecnumber/configure
@@ -5049,6 +5049,12 @@ else
   PICFLAG=
 fi
 
+case $host in
+  *-cygwin | *-mingw*)
+    PICFLAG=
+    ;;
+esac
+
 
 
 # Output.
diff --git a/libdecnumber/configure.ac b/libdecnumber/configure.ac
index 16c7465b327..5f048dd1166 100644
--- a/libdecnumber/configure.ac
+++ b/libdecnumber/configure.ac
@@ -101,6 +101,13 @@ AC_ARG_ENABLE(host-shared,
 [AS_HELP_STRING([--enable-host-shared],
 		[build host code as shared libraries])],
 [PICFLAG=-fPIC], [PICFLAG=])
+
+case $host in
+  *-cygwin | *-mingw*)
+    PICFLAG=
+    ;;
+esac
+
 AC_SUBST(PICFLAG)
 
 # Output.
diff --git a/libiberty/configure b/libiberty/configure
index dd896270dc6..f5129a451d4 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -5405,6 +5405,12 @@ if [ "${shared}" != "yes" ]; then
   PICFLAG=
 fi
 
+case $host in
+  *-cygwin | *-mingw*)
+    PICFLAG=
+    ;;
+esac
+
 
 NOASANFLAG=
 case " ${CFLAGS} " in
diff --git a/libiberty/configure.ac b/libiberty/configure.ac
index 0748c592704..94da4691b7e 100644
--- a/libiberty/configure.ac
+++ b/libiberty/configure.ac
@@ -253,6 +253,13 @@ fi
 if [[ "${shared}" != "yes" ]]; then
   PICFLAG=
 fi
+
+case $host in
+  *-cygwin | *-mingw*)
+    PICFLAG=
+    ;;
+esac
+
 AC_SUBST(PICFLAG)
 
 NOASANFLAG=
diff --git a/zlib/configure b/zlib/configure
index e65ade3badd..b685277b954 100755
--- a/zlib/configure
+++ b/zlib/configure
@@ -11540,6 +11540,12 @@ else
 fi
 
 
+case $host in
+  *-cygwin | *-mingw*)
+    PICFLAG=
+    ;;
+esac
+
 
 ac_config_files="$ac_config_files Makefile"
 
diff --git a/zlib/configure.ac b/zlib/configure.ac
index ec73c219228..540fc8a0e75 100644
--- a/zlib/configure.ac
+++ b/zlib/configure.ac
@@ -125,6 +125,13 @@ AC_ARG_ENABLE(host-shared,
 [AS_HELP_STRING([--enable-host-shared],
 		[build host code as shared libraries])],
 [PICFLAG=-fPIC], [PICFLAG=])
+
+case $host in
+  *-cygwin | *-mingw*)
+    PICFLAG=
+    ;;
+esac
+
 AC_SUBST(PICFLAG)
 
 AC_CONFIG_FILES([Makefile])
