diff --git a/bfd/configure b/bfd/configure
index 31dfac508a2..84f85b4ddfe 100755
--- a/bfd/configure
+++ b/bfd/configure
@@ -12193,7 +12193,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -13021,7 +13021,7 @@ _ACEOF
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
diff --git a/binutils/configure b/binutils/configure
index 5a3e5017b46..47243a0268d 100755
--- a/binutils/configure
+++ b/binutils/configure
@@ -12047,7 +12047,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -13647,7 +13647,7 @@ fi
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
@@ -14578,7 +14578,7 @@ do
 	esac
 
 	case $targ in
-	arm-wince-pe* | arm-*-wince | arm*-*-cegcc* | arm*-*-mingw32ce*)
+	arm-wince-pe* | arm-*-wince | arm*-*-cegcc* | arm*-*-mingw*ce*)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_ARM_WINCE"
@@ -14606,7 +14606,7 @@ do
 	  BUILD_WINDMC='$(WINDMC_PROG)$(EXEEXT)'
 	  BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
 	  ;;
-	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw32**)
+	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw**)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_I386"
diff --git a/binutils/configure.ac b/binutils/configure.ac
index 6243a2b0c2d..8fa143e8f1a 100644
--- a/binutils/configure.ac
+++ b/binutils/configure.ac
@@ -355,7 +355,7 @@ do
 	esac
 
 	case $targ in
-	arm-wince-pe* | arm-*-wince | arm*-*-cegcc* | arm*-*-mingw32ce*)
+	arm-wince-pe* | arm-*-wince | arm*-*-cegcc* | arm*-*-mingw*ce*)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_ARM_WINCE"
@@ -384,7 +384,7 @@ do
 	  BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
 	  ;;
 changequote(,)dnl
-	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw32**)
+	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw**)
 changequote([,])dnl
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
diff --git a/gas/configure b/gas/configure
index 4e9bbd9dfdc..cfdc053c50d 100755
--- a/gas/configure
+++ b/gas/configure
@@ -11701,7 +11701,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -11779,7 +11779,7 @@ fi
 # If we are on a DOS filesystem, we must use gdb.ini rather than
 # .gdbinit.
 case "${host}" in
-  *-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-windows*)
+  *-*-msdos* | *-*-go32* | *-*-mingw* | *-*-windows*)
     GDBINIT="gdb.ini"
     ac_config_files="$ac_config_files gdb.ini:gdbinit.in"
 
@@ -13979,7 +13979,7 @@ _ACEOF
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
diff --git a/gas/configure.ac b/gas/configure.ac
index 787ee53f021..804fc8a2a74 100644
--- a/gas/configure.ac
+++ b/gas/configure.ac
@@ -185,7 +185,7 @@ AH_VERBATIM([00_CONFIG_H_CHECK],
 # If we are on a DOS filesystem, we must use gdb.ini rather than
 # .gdbinit.
 case "${host}" in
-  *-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-windows*)
+  *-*-msdos* | *-*-go32* | *-*-mingw* | *-*-windows*)
     GDBINIT="gdb.ini"
     AC_CONFIG_FILES(gdb.ini:gdbinit.in)
     ;;
diff --git a/gdb/configure b/gdb/configure
index 40dbec7efa4..308e81a6dfc 100755
--- a/gdb/configure
+++ b/gdb/configure
@@ -23707,7 +23707,7 @@ fi
 
   WIN32APILIBS=
   case ${host} in
-    *mingw32*)
+    *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
@@ -27287,7 +27287,7 @@ if test "$GDB_COMPILER_TYPE" != gcc; then
 fi
 
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     # Enable -Wno-format by default when using gcc on mingw since many
     # GCC versions complain about %I64.
     build_warnings="$build_warnings -Wno-format" ;;
@@ -27520,7 +27520,7 @@ SER_HARDWIRE="ser-base.o ser-unix.o ser-pipe.o ser-tcp.o"
 case ${host} in
   *go32* ) SER_HARDWIRE=ser-go32.o ;;
   *djgpp* ) SER_HARDWIRE=ser-go32.o ;;
-  *mingw32*) SER_HARDWIRE="ser-base.o ser-tcp.o ser-mingw.o" ;;
+  *mingw*) SER_HARDWIRE="ser-base.o ser-tcp.o ser-mingw.o" ;;
   *) SER_HARDWIRE="$SER_HARDWIRE ser-uds.o" ;;
 esac
 
diff --git a/gdb/configure.ac b/gdb/configure.ac
index 6d749defda5..6805055dd02 100644
--- a/gdb/configure.ac
+++ b/gdb/configure.ac
@@ -1785,7 +1785,7 @@ SER_HARDWIRE="ser-base.o ser-unix.o ser-pipe.o ser-tcp.o"
 case ${host} in
   *go32* ) SER_HARDWIRE=ser-go32.o ;;
   *djgpp* ) SER_HARDWIRE=ser-go32.o ;;
-  *mingw32*) SER_HARDWIRE="ser-base.o ser-tcp.o ser-mingw.o" ;;
+  *mingw*) SER_HARDWIRE="ser-base.o ser-tcp.o ser-mingw.o" ;;
   *) SER_HARDWIRE="$SER_HARDWIRE ser-uds.o" ;;
 esac
 AC_SUBST(SER_HARDWIRE)
diff --git a/gdbserver/configure b/gdbserver/configure
index 050264ed7da..3f8ec95fb1c 100755
--- a/gdbserver/configure
+++ b/gdbserver/configure
@@ -6679,7 +6679,7 @@ fi
 
   WIN32APILIBS=
   case ${host} in
-    *mingw32*)
+    *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
@@ -9811,7 +9811,7 @@ if test "$GDB_COMPILER_TYPE" != gcc; then
 fi
 
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     # Enable -Wno-format by default when using gcc on mingw since many
     # GCC versions complain about %I64.
     build_warnings="$build_warnings -Wno-format" ;;
diff --git a/gdbsupport/configure b/gdbsupport/configure
index 38541e87c15..c49b4574510 100755
--- a/gdbsupport/configure
+++ b/gdbsupport/configure
@@ -7695,7 +7695,7 @@ fi
 
   WIN32APILIBS=
   case ${host} in
-    *mingw32*)
+    *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
@@ -10308,7 +10308,7 @@ if test "$GDB_COMPILER_TYPE" != gcc; then
 fi
 
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     # Enable -Wno-format by default when using gcc on mingw since many
     # GCC versions complain about %I64.
     build_warnings="$build_warnings -Wno-format" ;;
@@ -10463,7 +10463,7 @@ $as_echo "#define DEVELOPMENT 1" >>confdefs.h
 fi
 
 case ${host} in
-  *mingw32*)
+  *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
diff --git a/gdbsupport/configure.ac b/gdbsupport/configure.ac
index 3b461bc82de..f7b3a604209 100644
--- a/gdbsupport/configure.ac
+++ b/gdbsupport/configure.ac
@@ -72,7 +72,7 @@ if test "$development" = true ; then
 fi
 
 case ${host} in
-  *mingw32*)
+  *mingw*)
     AC_DEFINE(USE_WIN32API, 1,
               [Define if we should use the Windows API, instead of the
 	       POSIX API.  On Windows, we use the Windows API when
diff --git a/gold/configure b/gold/configure
index 5c6fe2a5607..0f10c427d0b 100755
--- a/gold/configure
+++ b/gold/configure
@@ -8029,7 +8029,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
diff --git a/gprof/configure b/gprof/configure
index cb150353050..fa471e46c7a 100755
--- a/gprof/configure
+++ b/gprof/configure
@@ -11970,7 +11970,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
diff --git a/ld/configure b/ld/configure
index a4d30abfb1c..a99db1877e2 100755
--- a/ld/configure
+++ b/ld/configure
@@ -15947,7 +15947,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -16745,7 +16745,7 @@ done
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
diff --git a/libiberty/configure b/libiberty/configure
index 79ef15312e2..a42d00a49a1 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -6761,8 +6761,8 @@ esac
 
     ;;
 
-  *-*-mingw32*)
-    # Under mingw32, sys_nerr and sys_errlist exist, but they are
+  *-*-mingw*)
+    # Under mingw, sys_nerr and sys_errlist exist, but they are
     # macros, so the test below won't find them.
     libiberty_cv_var_sys_nerr=yes
     libiberty_cv_var_sys_errlist=yes
diff --git a/libiberty/configure.ac b/libiberty/configure.ac
index 6c1ff9c6093..2c1af825361 100644
--- a/libiberty/configure.ac
+++ b/libiberty/configure.ac
@@ -642,8 +642,8 @@ if test -z "${setobjs}"; then
     AC_LIBOBJ([vsnprintf])
     ;;
 
-  *-*-mingw32*)
-    # Under mingw32, sys_nerr and sys_errlist exist, but they are
+  *-*-mingw*)
+    # Under mingw, sys_nerr and sys_errlist exist, but they are
     # macros, so the test below won't find them.
     libiberty_cv_var_sys_nerr=yes
     libiberty_cv_var_sys_errlist=yes
diff --git a/opcodes/configure b/opcodes/configure
index 9dc87d6c19c..c20c7a648c4 100755
--- a/opcodes/configure
+++ b/opcodes/configure
@@ -11525,7 +11525,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
diff --git a/sim/configure b/sim/configure
index c31ff71c805..8291fb52b6c 100755
--- a/sim/configure
+++ b/sim/configure
@@ -16198,7 +16198,7 @@ build_warnings="-Wall -Wdeclaration-after-statement -Wpointer-arith
 # Enable -Wno-format by default when using gcc on mingw since many
 # GCC versions complain about %I64.
 case "${host}" in
-  *-*-mingw32*) build_warnings="$build_warnings -Wno-format" ;;
+  *-*-mingw*) build_warnings="$build_warnings -Wno-format" ;;
   *) build_warnings="$build_warnings -Wformat-nonliteral" ;;
 esac
 
