diff --git a/ld/configure b/ld/configure
index b7f29ca68b7..aa091a7b9e9 100755
--- a/ld/configure
+++ b/ld/configure
@@ -16061,7 +16061,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -18741,7 +18741,7 @@ done
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
