diff --git a/configure b/configure
index 93d286bb3b7..d65312f4bf7 100755
--- a/configure
+++ b/configure
@@ -3583,10 +3583,10 @@ case "${target}" in
   ia64*-*-*vms*)
     noconfigdirs="$noconfigdirs target-libffi"
     ;;
-  i[3456789]86-w64-mingw*)
+  i[3456789]86-*-mingw64*)
     noconfigdirs="$noconfigdirs target-libffi"
     ;;
-  i[3456789]86-*-mingw*)
+  i[3456789]86-*-mingw32*)
     noconfigdirs="$noconfigdirs target-libffi"
     ;;
   x86_64-*-mingw*)
@@ -3932,9 +3932,9 @@ case "${target}" in
     # No ld support yet.
     noconfigdirs="$noconfigdirs libgui itcl ld"
     ;;
-  i[3456789]86-w64-mingw*)
+  i[3456789]86-*-mingw64*)
     ;;
-  i[3456789]86-*-mingw*)
+  i[3456789]86-*-mingw32*)
     target_configdirs="$target_configdirs target-winsup"
     ;;
   *-*-cygwin*)
@@ -10468,12 +10468,12 @@ case " $target_configdirs " in
 esac
 
 case "$target" in
-  x86_64-*mingw* | *-w64-mingw*)
+  *-mingw64*)
   # MinGW-w64 does not use newlib, nor does it use winsup. It may,
   # however, use a symlink named 'mingw' in ${prefix} .
     FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L${prefix}/${target}/lib -L${prefix}/mingw/lib -isystem ${prefix}/${target}/include -isystem ${prefix}/mingw/include'
     ;;
-  *-mingw*)
+  *-mingw32*)
   # MinGW can't be handled as Cygwin above since it does not use newlib.
     FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L$$r/$(TARGET_SUBDIR)/winsup/mingw -L$$r/$(TARGET_SUBDIR)/winsup/w32api/lib -isystem $$s/winsup/mingw/include -isystem $$s/winsup/w32api/include'
     ;;
