diff --git a/configure b/configure
index 93d286bb3b7..d65312f4bf7 100755
--- a/configure
+++ b/configure
@@ -3583,10 +3583,10 @@ case "${target}" in
   ia64*-*-*vms*)
     noconfigdirs="$noconfigdirs target-libffi"
     ;;
-  i[3456789]86-w64-mingw*)
+  i[3456789]86-*-mingw64*)
     noconfigdirs="$noconfigdirs target-libffi"
     ;;
-  i[3456789]86-*-mingw*)
+  i[3456789]86-*-mingw32*)
     noconfigdirs="$noconfigdirs target-libffi"
     ;;
   x86_64-*-mingw*)
@@ -3932,9 +3932,9 @@ case "${target}" in
     # No ld support yet.
     noconfigdirs="$noconfigdirs libgui itcl ld"
     ;;
-  i[3456789]86-w64-mingw*)
+  i[3456789]86-*-mingw64*)
     ;;
-  i[3456789]86-*-mingw*)
+  i[3456789]86-*-mingw32*)
     target_configdirs="$target_configdirs target-winsup"
     ;;
   *-*-cygwin*)
