diff --git a/bfd/configure b/bfd/configure
index 151f5cdf80a..2856543eb60 100755
--- a/bfd/configure
+++ b/bfd/configure
@@ -12167,7 +12167,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -13016,7 +13016,7 @@ _ACEOF
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
diff --git a/binutils/configure b/binutils/configure
index 531322ec521..242cd6ff6ea 100755
--- a/binutils/configure
+++ b/binutils/configure
@@ -12161,7 +12161,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -13717,7 +13717,7 @@ fi
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
@@ -14534,7 +14534,7 @@ do
 	esac
 
 	case $targ in
-	arm-wince-pe* | arm-*-wince | arm*-*-cegcc* | arm*-*-mingw32ce*)
+	arm-wince-pe* | arm-*-wince | arm*-*-cegcc* | arm*-*-mingw*ce*)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_ARM_WINCE"
@@ -14562,7 +14562,7 @@ do
 	  BUILD_WINDMC='$(WINDMC_PROG)$(EXEEXT)'
 	  BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
 	  ;;
-	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw32**)
+	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw**)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_I386"
diff --git a/binutils/configure.ac b/binutils/configure.ac
index 7b70e765243..0ea4f0747aa 100644
--- a/binutils/configure.ac
+++ b/binutils/configure.ac
@@ -330,7 +330,7 @@ do
 	esac
 
 	case $targ in
-	arm-wince-pe* | arm-*-wince | arm*-*-cegcc* | arm*-*-mingw32ce*)
+	arm-wince-pe* | arm-*-wince | arm*-*-cegcc* | arm*-*-mingw*ce*)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_ARM_WINCE"
@@ -359,7 +359,7 @@ do
 	  BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
 	  ;;
 changequote(,)dnl
-	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw32**)
+	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw**)
 changequote([,])dnl
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
diff --git a/gas/configure b/gas/configure
index 12ca81bb16b..bd5c0ab18ce 100755
--- a/gas/configure
+++ b/gas/configure
@@ -11644,7 +11644,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -11722,7 +11722,7 @@ fi
 # If we are on a DOS filesystem, we must use gdb.ini rather than
 # .gdbinit.
 case "${host}" in
-  *-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-windows*)
+  *-*-msdos* | *-*-go32* | *-*-mingw* | *-*-windows*)
     GDBINIT="gdb.ini"
     ac_config_files="$ac_config_files gdb.ini:gdbinit.in"
 
@@ -13919,7 +13919,7 @@ _ACEOF
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
diff --git a/gas/configure.ac b/gas/configure.ac
index 572e56e281c..7289f6c3134 100644
--- a/gas/configure.ac
+++ b/gas/configure.ac
@@ -176,7 +176,7 @@ AH_VERBATIM([00_CONFIG_H_CHECK],
 # If we are on a DOS filesystem, we must use gdb.ini rather than
 # .gdbinit.
 case "${host}" in
-  *-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-windows*)
+  *-*-msdos* | *-*-go32* | *-*-mingw* | *-*-windows*)
     GDBINIT="gdb.ini"
     AC_CONFIG_FILES(gdb.ini:gdbinit.in)
     ;;
diff --git a/gdb/configure b/gdb/configure
index e751edc3f9a..247058c1d48 100755
--- a/gdb/configure
+++ b/gdb/configure
@@ -13487,7 +13487,7 @@ fi
 
   WIN32APILIBS=
   case ${host} in
-    *mingw32*)
+    *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
@@ -17105,7 +17105,7 @@ if test "$GDB_COMPILER_TYPE" != gcc; then
 fi
 
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     # Enable -Wno-format by default when using gcc on mingw since many
     # GCC versions complain about %I64.
     build_warnings="$build_warnings -Wno-format" ;;
@@ -17338,7 +17338,7 @@ SER_HARDWIRE="ser-base.o ser-unix.o ser-pipe.o ser-tcp.o"
 case ${host} in
   *go32* ) SER_HARDWIRE=ser-go32.o ;;
   *djgpp* ) SER_HARDWIRE=ser-go32.o ;;
-  *mingw32*) SER_HARDWIRE="ser-base.o ser-tcp.o ser-mingw.o" ;;
+  *mingw*) SER_HARDWIRE="ser-base.o ser-tcp.o ser-mingw.o" ;;
   *) SER_HARDWIRE="$SER_HARDWIRE ser-uds.o" ;;
 esac
 
diff --git a/gdb/configure.ac b/gdb/configure.ac
index 5a380ce38d9..58b8c6ff665 100644
--- a/gdb/configure.ac
+++ b/gdb/configure.ac
@@ -1850,7 +1850,7 @@ SER_HARDWIRE="ser-base.o ser-unix.o ser-pipe.o ser-tcp.o"
 case ${host} in
   *go32* ) SER_HARDWIRE=ser-go32.o ;;
   *djgpp* ) SER_HARDWIRE=ser-go32.o ;;
-  *mingw32*) SER_HARDWIRE="ser-base.o ser-tcp.o ser-mingw.o" ;;
+  *mingw*) SER_HARDWIRE="ser-base.o ser-tcp.o ser-mingw.o" ;;
   *) SER_HARDWIRE="$SER_HARDWIRE ser-uds.o" ;;
 esac
 AC_SUBST(SER_HARDWIRE)
diff --git a/gdbserver/configure b/gdbserver/configure
index bcb15941ed1..dedf24187ea 100755
--- a/gdbserver/configure
+++ b/gdbserver/configure
@@ -6679,7 +6679,7 @@ fi
 
   WIN32APILIBS=
   case ${host} in
-    *mingw32*)
+    *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
@@ -9812,7 +9812,7 @@ if test "$GDB_COMPILER_TYPE" != gcc; then
 fi
 
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     # Enable -Wno-format by default when using gcc on mingw since many
     # GCC versions complain about %I64.
     build_warnings="$build_warnings -Wno-format" ;;
diff --git a/gdbsupport/configure b/gdbsupport/configure
index afaba88ee94..3b198ee143c 100755
--- a/gdbsupport/configure
+++ b/gdbsupport/configure
@@ -7692,7 +7692,7 @@ fi
 
   WIN32APILIBS=
   case ${host} in
-    *mingw32*)
+    *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
@@ -10297,7 +10297,7 @@ if test "$GDB_COMPILER_TYPE" != gcc; then
 fi
 
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     # Enable -Wno-format by default when using gcc on mingw since many
     # GCC versions complain about %I64.
     build_warnings="$build_warnings -Wno-format" ;;
@@ -10443,7 +10443,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
 case ${host} in
-  *mingw32*)
+  *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
diff --git a/gdbsupport/configure.ac b/gdbsupport/configure.ac
index 55477d55c82..32f413ac32e 100644
--- a/gdbsupport/configure.ac
+++ b/gdbsupport/configure.ac
@@ -61,7 +61,7 @@ AM_GDB_COMPILER_TYPE
 AM_GDB_WARNINGS
 
 case ${host} in
-  *mingw32*)
+  *mingw*)
     AC_DEFINE(USE_WIN32API, 1,
               [Define if we should use the Windows API, instead of the
 	       POSIX API.  On Windows, we use the Windows API when
diff --git a/gold/configure b/gold/configure
index 5f5b8c3602b..d23519f2ef9 100755
--- a/gold/configure
+++ b/gold/configure
@@ -7979,7 +7979,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
diff --git a/gprof/configure b/gprof/configure
index 8d74f9b22a5..cffd9c899ed 100755
--- a/gprof/configure
+++ b/gprof/configure
@@ -11946,7 +11946,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
diff --git a/ld/configure b/ld/configure
index f5c329c1f12..210662870e4 100755
--- a/ld/configure
+++ b/ld/configure
@@ -15601,7 +15601,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
@@ -16399,7 +16399,7 @@ done
 
 
 case "${host}" in
-*-*-msdos* | *-*-go32* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+*-*-msdos* | *-*-go32* | *-*-mingw* | *-*-cygwin* | *-*-windows*)
 
 $as_echo "#define USE_BINARY_FOPEN 1" >>confdefs.h
  ;;
diff --git a/libiberty/configure b/libiberty/configure
index e09ddd91e58..c89fe6e3324 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -6755,8 +6755,8 @@ esac
 
     ;;
 
-  *-*-mingw32*)
-    # Under mingw32, sys_nerr and sys_errlist exist, but they are
+  *-*-mingw*)
+    # Under mingw, sys_nerr and sys_errlist exist, but they are
     # macros, so the test below won't find them.
     libiberty_cv_var_sys_nerr=yes
     libiberty_cv_var_sys_errlist=yes
diff --git a/libiberty/configure.ac b/libiberty/configure.ac
index f59f35e1f20..fefe8246167 100644
--- a/libiberty/configure.ac
+++ b/libiberty/configure.ac
@@ -641,8 +641,8 @@ if test -z "${setobjs}"; then
     AC_LIBOBJ([vsnprintf])
     ;;
 
-  *-*-mingw32*)
-    # Under mingw32, sys_nerr and sys_errlist exist, but they are
+  *-*-mingw*)
+    # Under mingw, sys_nerr and sys_errlist exist, but they are
     # macros, so the test below won't find them.
     libiberty_cv_var_sys_nerr=yes
     libiberty_cv_var_sys_errlist=yes
diff --git a/opcodes/configure b/opcodes/configure
index d3cef604c85..2214421544f 100755
--- a/opcodes/configure
+++ b/opcodes/configure
@@ -11501,7 +11501,7 @@ fi
 
 # Disable -Wformat by default when using gcc on mingw
 case "${host}" in
-  *-*-mingw32*)
+  *-*-mingw*)
     if test "${GCC}" = yes -a -z "${ERROR_ON_WARNING}" ; then
       GCC_WARN_CFLAGS="$GCC_WARN_CFLAGS -Wno-format"
       GCC_WARN_CFLAGS_FOR_BUILD="$GCC_WARN_CFLAGS_FOR_BUILD -Wno-format"
diff --git a/sim/configure b/sim/configure
index 51ac74a4ce4..3a71598aeeb 100755
--- a/sim/configure
+++ b/sim/configure
@@ -16010,7 +16010,7 @@ build_warnings="-Wall -Wdeclaration-after-statement -Wpointer-arith
 # Enable -Wno-format by default when using gcc on mingw since many
 # GCC versions complain about %I64.
 case "${host}" in
-  *-*-mingw32*) build_warnings="$build_warnings -Wno-format" ;;
+  *-*-mingw*) build_warnings="$build_warnings -Wno-format" ;;
   *) build_warnings="$build_warnings -Wformat-nonliteral" ;;
 esac
 
diff --git a/sim/ppc/configure b/sim/ppc/configure
index 1e5283771fa..7c01650793b 100755
--- a/sim/ppc/configure
+++ b/sim/ppc/configure
@@ -3460,7 +3460,7 @@ fi
 
 
 case ${host} in
-  *mingw32*)
+  *mingw*)
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
diff --git a/sim/ppc/configure.ac b/sim/ppc/configure.ac
index 75232006e94..5dd79753add 100644
--- a/sim/ppc/configure.ac
+++ b/sim/ppc/configure.ac
@@ -410,7 +410,7 @@ fi],[sim_xor_endian=""])dnl
 
 
 case ${host} in
-  *mingw32*)
+  *mingw*)
     AC_DEFINE(USE_WIN32API, 1,
               [Define if we should use the Windows API, instead of the 
 	       POSIX API.  On Windows, we use the Windows API when 
