diff --git a/gcc/config/i386/x-mingw32 b/gcc/config/i386/x-mingw32
index 8900bfccb6c..5ebe0882f4d 100644
--- a/gcc/config/i386/x-mingw32
+++ b/gcc/config/i386/x-mingw32
@@ -21,6 +21,9 @@
 #
 local_includedir=$(libsubdir)/$(unlibsubdir)/..`echo $(exec_prefix) | sed -e 's|^$(prefix)||' -e 's|/[^/]*|/..|g'`/include
 
+# Add Windows socket library.
+LIBS += -lws2_32
+
 # On MinGW, we use "%IA64d" to print 64-bit integers, and the format-checking
 # code does not handle that, so we have to disable checking here.
 WERROR_FLAGS += -Wno-format
diff --git a/gcc/diagnostics/sarif-sink.cc b/gcc/diagnostics/sarif-sink.cc
index 76aec2a886c..1356919ff80 100644
--- a/gcc/diagnostics/sarif-sink.cc
+++ b/gcc/diagnostics/sarif-sink.cc
@@ -18,6 +18,13 @@ You should have received a copy of the GNU General Public License
 along with GCC; see the file COPYING3.  If not see
 <http://www.gnu.org/licenses/>.  */
 
+#ifdef __MINGW32__
+#include <winsock2.h>
+#include <afunix.h>
+#else
+#include <sys/un.h>
+#include <sys/socket.h>
+#endif
 
 #include "config.h"
 #define INCLUDE_LIST
@@ -55,8 +62,6 @@ along with GCC; see the file COPYING3.  If not see
 #include "backtrace.h"
 #include "xml.h"
 #include "intl.h"
-#include <sys/un.h>
-#include <sys/socket.h>
 
 namespace diagnostics {
 
