diff --git a/misc.c b/misc.c
index ce77ec943..418c499c6 100644
--- a/misc.c
+++ b/misc.c
@@ -1220,6 +1220,7 @@ tilde_expand(const char *filename, uid_t uid, char **retp)
 	struct passwd *pw;
 	size_t len;
 	int ret = -1, r, slash;
+	const char *home_env = NULL;
 
 	*retp = NULL;
 	if (*filename != '~') {
@@ -1248,14 +1249,47 @@ tilde_expand(const char *filename, uid_t uid, char **retp)
 		}
 		/* else					~user */
 	}
+
 	if (user != NULL) {
 		if ((pw = getpwnam(user)) == NULL) {
 			error_f("No such user %s", user);
 			goto out;
 		}
-	} else if ((pw = getpwuid(uid)) == NULL) {
-		error_f("No such uid %ld", (long)uid);
-		goto out;
+	} else {
+
+#ifdef __CYGWIN__
+		/*
+			* On Cygwin, prefer $HOME if it is an absolute path.
+			*/
+		home_env = getenv("HOME");
+		if (home_env != NULL && *home_env == '/') {
+
+			slash = (len = strlen(home_env)) == 0 ||
+				home_env[len - 1] != '/';
+
+			if ((r = xasprintf(&s, "%s%s%s",
+				home_env,
+				slash ? "/" : "",
+				path != NULL ? path : "")) == -1) {
+				error_f("xasprintf failed");
+				goto out;
+			}
+			if (r >= PATH_MAX) {
+				error_f("Path too long");
+				goto out;
+			}
+
+			ret = 0;
+			*retp = s;
+			s = NULL;
+			goto out;
+		}
+#endif
+
+		if ((pw = getpwuid(uid)) == NULL) {
+			error_f("No such uid %ld", (long)uid);
+			goto out;
+		}
 	}
 
 	/* Make sure directory has a trailing '/' */
