diff --git a/misc.c b/misc.c
index ce77ec943..67b20d3c6 100644
--- a/misc.c
+++ b/misc.c
@@ -1258,10 +1259,19 @@ tilde_expand(const char *filename, uid_t uid, char **retp)
 		goto out;
 	}
 
+	const char * home_dir = pw->pw_dir;
+#ifdef __CYGWIN__
+	// On Cygwin, prefer $HOME if it is an absolute path.
+	const char * home_env = getenv("HOME");
+	if (home_env != NULL && *home_env == '/') {
+		home_dir = home_env;
+	}
+#endif
+
 	/* Make sure directory has a trailing '/' */
-	slash = (len = strlen(pw->pw_dir)) == 0 || pw->pw_dir[len - 1] != '/';
+	slash = (len = strlen(home_dir)) == 0 || home_dir[len - 1] != '/';
 
-	if ((r = xasprintf(&s, "%s%s%s", pw->pw_dir,
+	if ((r = xasprintf(&s, "%s%s%s", home_dir,
 	    slash ? "/" : "", path != NULL ? path : "")) <= 0) {
 		error_f("xasprintf failed");
 		goto out;
