diff --git a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/importexecutable/AbstractImportExecutableWizard.java b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/importexecutable/AbstractImportExecutableWizard.java
index 9464b21ea8..c7da0788dc 100644
--- a/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/importexecutable/AbstractImportExecutableWizard.java
+++ b/debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/importexecutable/AbstractImportExecutableWizard.java
@@ -105,12 +105,18 @@ public abstract class AbstractImportExecutableWizard extends Wizard implements I
 					IFolder exeFolder = null;
 					int segmentCount = locationRelative.segmentCount();
 					if (segmentCount >= 2) {
-						exeFolder = project.getFolder(locationRelative.segment(0));
+						String segment = locationRelative.segment(0);
+						//Eclipse hate directory name like ".libs" on Windows. So replace "." with "_"
+						segment = segment.replace('.', '_');
+						exeFolder = project.getFolder(segment);
 						if (!exeFolder.exists()) {
 							exeFolder.create(IResource.VIRTUAL, false, null);
 						}
 						for (int j = 1; j < segmentCount - 1; j++) {
-							exeFolder = exeFolder.getFolder(locationRelative.segment(j));
+							segment = locationRelative.segment(j);
+							//Eclipse hate directory name like ".libs" on Windows. So replace "." with "_"
+							segment = segment.replace('.', '_');
+							exeFolder = exeFolder.getFolder(segment);
 							if (!exeFolder.exists()) {
 								exeFolder.create(IResource.VIRTUAL, false, null);
 							}
