diff --git a/lldb/CMakeLists.txt b/lldb/CMakeLists.txt
index 59cdc4593463..a79e059a2d04 100644
--- a/lldb/CMakeLists.txt
+++ b/lldb/CMakeLists.txt
@@ -77,7 +77,7 @@ if (LLDB_ENABLE_PYTHON)
     endif()
   endforeach()
   # Make sure lldb extension has "_d" suffix on Windows in Debug mode.
-  if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL Debug)
+  if(MSVC AND CMAKE_BUILD_TYPE STREQUAL Debug)
     string(SUBSTRING ${LLDB_PYTHON_EXT_SUFFIX} 0 2 FIRST_2_CHARS)
     # Add "_d" manually if LLDB_PYTHON_EXT_SUFFIX lacks it due to release Python configuration.
     if(NOT FIRST_2_CHARS STREQUAL "_d")
