diff --git a/lldb/CMakeLists.txt b/lldb/CMakeLists.txt
index 20d584c48bad..5fe7290e4536 100644
--- a/lldb/CMakeLists.txt
+++ b/lldb/CMakeLists.txt
@@ -53,6 +53,11 @@ if (WIN32)
   add_definitions(-D_ENABLE_EXTENDED_ALIGNED_STORAGE)
 endif()
 
+if (CYGWIN)
+  add_definitions(-D_GNU_SOURCE)
+  add_definitions(-D__addr_t_defined)
+endif()
+
 if (LLDB_ENABLE_PYTHON)
   set(cachestring_LLDB_PYTHON_RELATIVE_PATH
     "Path where Python modules are installed, relative to LLDB's install prefix")
