diff --git a/lldb/unittests/Core/ModuleSpecTest.cpp b/lldb/unittests/Core/ModuleSpecTest.cpp
index f9e19ed35acc..d4f5e03f5055 100644
--- a/lldb/unittests/Core/ModuleSpecTest.cpp
+++ b/lldb/unittests/Core/ModuleSpecTest.cpp
@@ -22,7 +22,7 @@
 using namespace lldb;
 using namespace lldb_private;
 
-extern const char *TestMainArgv0;
+static const char *TestMainArgv0;
 
 // This test file intentionally doesn't initialize the FileSystem.
 // Everything in this file should be able to run without requiring
diff --git a/lldb/unittests/Host/FileSystemTest.cpp b/lldb/unittests/Host/FileSystemTest.cpp
index 58887f6b2467..c2b18c6c2d20 100644
--- a/lldb/unittests/Host/FileSystemTest.cpp
+++ b/lldb/unittests/Host/FileSystemTest.cpp
@@ -12,7 +12,7 @@
 #include "lldb/Host/FileSystem.h"
 #include "llvm/Support/Errc.h"
 
-extern const char *TestMainArgv0;
+static const char *TestMainArgv0;
 
 using namespace lldb_private;
 using namespace llvm;
diff --git a/lldb/unittests/TestingSupport/TestUtilities.cpp b/lldb/unittests/TestingSupport/TestUtilities.cpp
index efdc6c5eb234..f185b7352255 100644
--- a/lldb/unittests/TestingSupport/TestUtilities.cpp
+++ b/lldb/unittests/TestingSupport/TestUtilities.cpp
@@ -17,7 +17,7 @@
 
 using namespace lldb_private;
 
-extern const char *TestMainArgv0;
+static const char *TestMainArgv0;
 
 std::once_flag TestUtilities::g_debugger_initialize_flag;
 std::string lldb_private::GetInputFilePath(const llvm::Twine &name) {
