diff --git a/clang/include/clang/Support/Compiler.h b/clang/include/clang/Support/Compiler.h
index c35815e106d2..084581ae6479 100644
--- a/clang/include/clang/Support/Compiler.h
+++ b/clang/include/clang/Support/Compiler.h
@@ -39,7 +39,7 @@
 #define CLANG_ABI
 #define CLANG_TEMPLATE_ABI
 #define CLANG_EXPORT_TEMPLATE
-#elif defined(_WIN32) && !defined(__MINGW32__)
+#elif defined(_MSC_VER)
 #if defined(CLANG_EXPORTS)
 #define CLANG_ABI __declspec(dllexport)
 #define CLANG_TEMPLATE_ABI
@@ -49,7 +49,7 @@
 #define CLANG_TEMPLATE_ABI __declspec(dllimport)
 #define CLANG_EXPORT_TEMPLATE
 #endif
-#elif defined(__ELF__) || defined(__MINGW32__) || defined(_AIX)
+#elif defined(__ELF__) || defined(__CYGWIN__) || defined(__MINGW32__) || defined(_AIX)
 #define CLANG_ABI LLVM_ATTRIBUTE_VISIBILITY_DEFAULT
 #define CLANG_TEMPLATE_ABI LLVM_ATTRIBUTE_VISIBILITY_DEFAULT
 #define CLANG_EXPORT_TEMPLATE
