diff --git a/llvm/lib/Target/X86/X86ISelLowering.cpp b/llvm/lib/Target/X86/X86ISelLowering.cpp
index ee8d268379fe..6e1c3000200e 100644
--- a/llvm/lib/Target/X86/X86ISelLowering.cpp
+++ b/llvm/lib/Target/X86/X86ISelLowering.cpp
@@ -18185,6 +18185,7 @@ unsigned X86TargetLowering::getGlobalWrapperKind(
 
   // The following OpFlags under RIP-rel PIC use RIP.
   if (Subtarget.isPICStyleRIPRel() &&
+      !(Subtarget.isTargetWindowsCygwin() && Subtarget.isTargetWin64()) &&
       (OpFlags == X86II::MO_NO_FLAG || OpFlags == X86II::MO_COFFSTUB ||
        OpFlags == X86II::MO_DLLIMPORT))
     return X86ISD::WrapperRIP;
