diff --git a/Utilities/cmcurl/lib/rand.c b/Utilities/cmcurl/lib/rand.c
index c62b1a4032..63743373ea 100644
--- a/Utilities/cmcurl/lib/rand.c
+++ b/Utilities/cmcurl/lib/rand.c
@@ -21,10 +21,14 @@
  * SPDX-License-Identifier: curl
  *
  ***************************************************************************/
+#ifdef __CYGWIN__
+#define _BSD_SOURCE
+#endif
 
 #include "curl_setup.h"
 
 #include <limits.h>
+#include <stdlib.h>
 
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
