diff --git a/Source/cmLocalGenerator.cxx b/Source/cmLocalGenerator.cxx
index 2638af1d28..eb7de3938c 100644
--- a/Source/cmLocalGenerator.cxx
+++ b/Source/cmLocalGenerator.cxx
@@ -1257,6 +1294,25 @@ std::vector<BT<std::string>> cmLocalGenerator::GetIncludeDirectoriesImplicit(
       // Only exclude this hard coded path for backwards compatibility.
       implicitExclude.emplace("/usr/include");
     }
+#ifdef __MINGW32__
+	const char * path = getenv("INCLUDE");
+	if (path) {
+		implicitExclude.insert(path);
+	} else {
+		const char * msystem = getenv("MSYSTEM");
+		if (msystem) {
+			if (strcmp(msystem, "UCRT64") == 0) {
+				implicitExclude.insert(cygpath_m("/ucrt64/include"));
+			} else if (strcmp(msystem, "MINGW64") == 0) {
+				implicitExclude.insert(cygpath_m("/mingw64/include"));
+			} else {
+
+			}
+		} else {
+
+		}
+	}
+#endif
 
     for (std::string const& i : impDirVec) {
       if (implicitSet.insert(this->GlobalGenerator->GetRealPath(i)).second) {
