diff --git a/gcc/incpath.cc b/gcc/incpath.cc
index 28cf2efac4e..bafa8eecfea 100644
--- a/gcc/incpath.cc
+++ b/gcc/incpath.cc
@@ -179,6 +179,49 @@ add_env_var_paths (const char *env_var, incpath_kind chain)
 #endif
 }
 
+static void
+dump_static_array_cpp_include_defaults()
+{
+	size_t i = 0;
+	const struct default_include * p;
+	for (p = cpp_include_defaults; p->fname; p++) {
+		fprintf(stderr, "%zu. %s\n", i, p->fname);
+		i++;
+	}
+}
+
+static void
+check_static_array_cpp_include_defaults()
+{
+	static bool checked = false;
+	if (!checked) {
+
+		const struct default_include * p;
+		for (p = cpp_include_defaults; p->fname; p++) {
+#ifdef _WIN32
+			if (p->fname[0] == '/') {
+				fprintf(stderr, "array cpp_include_defaults cantain path start with '/'.\n");
+				dump_static_array_cpp_include_defaults();
+				exit(1);
+			}
+			if (!(filename_ncmp(p->fname, cpp_PREFIX, cpp_PREFIX_len) == 0)) {
+				fprintf(stderr, "array cpp_include_defaults cantain path not start with PREFIX %s.\n", cpp_PREFIX);
+				dump_static_array_cpp_include_defaults();
+				exit(1);
+			}
+#else
+			if (p->fname[0] != '/') {
+				fprintf(stderr, "array cpp_include_defaults cantain path not start with '/'.\n");
+				dump_static_array_cpp_include_defaults();
+				exit(1);
+			}
+#endif
+		}
+
+		checked = true;
+	}
+}
+
 /* Append the standard include chain defined in cppdefault.cc.  */
 static void
 add_standard_paths (const char *sysroot, const char *iprefix,
@@ -188,6 +231,8 @@ add_standard_paths (const char *sysroot, const char *iprefix,
   int relocated = cpp_relocated ();
   size_t len;
 
+  check_static_array_cpp_include_defaults();
+
   if (iprefix && (len = cpp_GCC_INCLUDE_DIR_len) != 0)
     {
       /* Look for directories that start with the standard prefix.
