diff --git a/configure b/configure
index 51bf1d1add1..89c4dc02a38 100755
--- a/configure
+++ b/configure
@@ -10686,18 +10686,6 @@ case " $target_configdirs " in
   ;;
 esac
 
-case "$target" in
-  x86_64-*mingw* | *-w64-mingw*)
-  # MinGW-w64 does not use newlib, nor does it use winsup. It may,
-  # however, use a symlink named 'mingw' in ${prefix} .
-    FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L${prefix}/${target}/lib -L${prefix}/mingw/lib -isystem ${prefix}/${target}/include -isystem ${prefix}/mingw/include'
-    ;;
-  *-mingw*)
-  # MinGW can't be handled as Cygwin above since it does not use newlib.
-    FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L$$r/$(TARGET_SUBDIR)/winsup/mingw -L$$r/$(TARGET_SUBDIR)/winsup/w32api/lib -isystem $$s/winsup/mingw/include -isystem $$s/winsup/w32api/include'
-    ;;
-esac
-
 # Allow the user to override the flags for
 # our build compiler if desired.
 if test x"${build}" = x"${host}" ; then
