diff --git a/libgcc/config/i386/cygming-crtbegin.c b/libgcc/config/i386/cygming-crtbegin.c
index 5014b923905..b08a54630e0 100644
--- a/libgcc/config/i386/cygming-crtbegin.c
+++ b/libgcc/config/i386/cygming-crtbegin.c
@@ -24,6 +24,8 @@ a copy of the GCC Runtime Library Exception along with this program;
 see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 <http://www.gnu.org/licenses/>.  */
 
+#if ! defined(inhibit_libc)
+
 /* Target machine header files require this define. */
 #define IN_LIBGCC2
 
@@ -159,3 +161,5 @@ __gcc_deregister_frame (void)
     FreeLibrary (hmod_libgcc);
 #endif
 }
+
+#endif
diff --git a/libgcc/config/i386/cygming-crtend.c b/libgcc/config/i386/cygming-crtend.c
index 106806e4f60..f15712c6e1e 100644
--- a/libgcc/config/i386/cygming-crtend.c
+++ b/libgcc/config/i386/cygming-crtend.c
@@ -24,6 +24,8 @@ a copy of the GCC Runtime Library Exception along with this program;
 see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 <http://www.gnu.org/licenses/>.  */
 
+#if ! defined(inhibit_libc)
+
 /* Target machine header files require this define. */
 #define IN_LIBGCC2
 
@@ -79,3 +81,5 @@ deregister_frame_dtor (void)
   __gcc_deregister_frame ();
 }
 #endif
+
+#endif
diff --git a/libgcc/config/i386/enable-execute-stack-mingw32.c b/libgcc/config/i386/enable-execute-stack-mingw32.c
index 99006d97a4e..df445507c54 100644
--- a/libgcc/config/i386/enable-execute-stack-mingw32.c
+++ b/libgcc/config/i386/enable-execute-stack-mingw32.c
@@ -22,6 +22,7 @@
    see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    <http://www.gnu.org/licenses/>.  */
 
+#if ! defined(inhibit_libc)
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 
@@ -37,3 +38,4 @@ __enable_execute_stack (void *addr)
   VirtualProtect (b.BaseAddress, b.RegionSize, PAGE_EXECUTE_READWRITE,
 		  &b.Protect);
 }
+#endif
diff --git a/libgcc/libgcc2.c b/libgcc/libgcc2.c
index e04d169f888..dd2c728f387 100644
--- a/libgcc/libgcc2.c
+++ b/libgcc/libgcc2.c
@@ -2951,7 +2951,7 @@ __clear_cache (void *beg __attribute__((__unused__)),
 
 #endif /* L_clear_cache */
 
-#ifdef L_trampoline
+#if defined(L_trampoline) && ! defined(inhibit_libc)
 
 /* Jump to a trampoline, loading the static chain address.  */
 
diff --git a/libgcc/unwind-c.c b/libgcc/unwind-c.c
index 63c1deebdf2..1fd73b1fd71 100644
--- a/libgcc/unwind-c.c
+++ b/libgcc/unwind-c.c
@@ -24,6 +24,8 @@ a copy of the GCC Runtime Library Exception along with this program;
 see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 <http://www.gnu.org/licenses/>.  */
 
+#if ! defined(inhibit_libc)
+
 #include "tconfig.h"
 #include "tsystem.h"
 #include "auto-target.h"
@@ -244,3 +246,4 @@ __gcc_personality_seh0 (PEXCEPTION_RECORD ms_exc, void *this_frame,
 				ms_disp, __gcc_personality_imp);
 }
 #endif /* SEH */
+#endif
diff --git a/libgcc/unwind-seh.c b/libgcc/unwind-seh.c
index 8ef0257b616..a7738f13fa2 100644
--- a/libgcc/unwind-seh.c
+++ b/libgcc/unwind-seh.c
@@ -22,6 +22,8 @@
    see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    <http://www.gnu.org/licenses/>.  */
 
+#if ! defined(inhibit_libc)
+
 #include "tconfig.h"
 #include "tsystem.h"
 #include "coretypes.h"
@@ -481,3 +483,4 @@ _Unwind_Backtrace(_Unwind_Trace_Fn trace,
     }
 }
 #endif /* __SEH__  && !defined (__USING_SJLJ_EXCEPTIONS__)  */
+#endif
diff --git a/libgcc/unwind-sjlj.c b/libgcc/unwind-sjlj.c
index 178f4d0af01..425e93fcf04 100644
--- a/libgcc/unwind-sjlj.c
+++ b/libgcc/unwind-sjlj.c
@@ -22,6 +22,8 @@
    see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    <http://www.gnu.org/licenses/>.  */
 
+#if ! defined(inhibit_libc)
+
 #include "tconfig.h"
 #include "tsystem.h"
 #include "coretypes.h"
@@ -325,3 +327,4 @@ uw_identify_context (struct _Unwind_Context *context)
 #include "unwind.inc"
 
 #endif /* USING_SJLJ_EXCEPTIONS */
+#endif
