diff --git a/gcc/gcc.cc b/gcc/gcc.cc
index 32520c17778..ce2f0457904 100644
--- a/gcc/gcc.cc
+++ b/gcc/gcc.cc
@@ -3158,6 +3158,16 @@ add_prefix (struct path_prefix *pprefix, const char *prefix,
 
   prefix = update_path (prefix, component);
   len = strlen (prefix);
+  char prefix1[len + 10];
+  strcpy(prefix1, prefix);
+  free((char *)prefix);
+  prefix = 0;
+  len = convert_path(prefix1, len);
+  if (len > 0 && prefix1[len - 1] != '/') {
+    prefix1[len++] = '/';
+    prefix1[len] = 0;
+  }
+  prefix = xstrdup(prefix1);
   if (len > pprefix->max_len)
     pprefix->max_len = len;
 
