diff --git a/libstdc++-v3/configure.host b/libstdc++-v3/configure.host
index 253e5a9ad0d..9d1e30834a1 100644
--- a/libstdc++-v3/configure.host
+++ b/libstdc++-v3/configure.host
@@ -280,13 +280,13 @@ case "${host_os}" in
   hpux*)
     os_include_dir="os/hpux"
     ;;
-  mingw32*)
+  mingw*)
     case "$host" in
-      *-w64-*)
+      *-mingw64*)
         os_include_dir="os/mingw32-w64"
         error_constants_dir="os/mingw32-w64"
         ;;
-      *)
+      *-mingw32*)
         os_include_dir="os/mingw32"
         error_constants_dir="os/mingw32"
         ;;
