diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index 947abb9b02b5..64398b3e7bb2 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -579,7 +579,7 @@ toolchains::MinGW::GetExceptionModel(const ArgList &Args) const {
   if (getArch() == llvm::Triple::x86_64 || getArch() == llvm::Triple::aarch64 ||
       getArch() == llvm::Triple::arm || getArch() == llvm::Triple::thumb)
     return llvm::ExceptionHandling::WinEH;
-  return llvm::ExceptionHandling::DwarfCFI;
+  return llvm::ExceptionHandling::SjLj;
 }
 
 SanitizerMask toolchains::MinGW::getSupportedSanitizers() const {
