--- origsrc/llvm-8.0.1.src/unittests/Passes/CMakeLists.txt	2019-01-14 13:25:35.000000000 -0500
+++ src/llvm-8.0.1.src/unittests/Passes/CMakeLists.txt	2019-08-26 07:34:59.067715400 -0400
@@ -14,7 +14,7 @@ add_llvm_unittest(PluginsTests
 export_executable_symbols(PluginsTests)
 target_link_libraries(PluginsTests PRIVATE LLVMTestingSupport)
 
-set(LLVM_LINK_COMPONENTS)
+set(LLVM_LINK_COMPONENTS Support Core)
 add_llvm_library(TestPlugin MODULE BUILDTREE_ONLY
   TestPlugin.cpp
   )
