--- origsrc/llvm-8.0.1.src/tools/llvm-cfi-verify/lib/CMakeLists.txt	2018-09-15 15:04:27.000000000 -0400
+++ src/llvm-8.0.1.src/tools/llvm-cfi-verify/lib/CMakeLists.txt	2019-07-21 17:11:44.112119200 -0400
@@ -1,4 +1,4 @@
-add_library(LLVMCFIVerify
+llvm_add_library(LLVMCFIVerify
   STATIC
   FileAnalysis.cpp
   FileAnalysis.h
@@ -7,7 +7,7 @@ add_library(LLVMCFIVerify
   )
 
 llvm_update_compile_flags(LLVMCFIVerify)
-llvm_map_components_to_libnames(libs
+set(LLVM_LINK_COMPONENTS
   DebugInfoDWARF
   MC
   MCParser
@@ -15,5 +15,4 @@ llvm_map_components_to_libnames(libs
   Support
   Symbolize
   )
-target_link_libraries(LLVMCFIVerify ${libs})
 set_target_properties(LLVMCFIVerify PROPERTIES FOLDER "Libraries")
--- origsrc/llvm-8.0.1.src/tools/llvm-exegesis/lib/AArch64/CMakeLists.txt	2018-09-15 15:04:27.000000000 -0400
+++ src/llvm-8.0.1.src/tools/llvm-exegesis/lib/AArch64/CMakeLists.txt	2019-07-21 17:12:48.875918200 -0400
@@ -3,16 +3,15 @@ include_directories(
   ${LLVM_BINARY_DIR}/lib/Target/AArch64
   )
 
-add_library(LLVMExegesisAArch64
+llvm_add_library(LLVMExegesisAArch64
   STATIC
   Target.cpp
   )
 
 llvm_update_compile_flags(LLVMExegesisAArch64)
-llvm_map_components_to_libnames(libs
+set(LLVM_LINK_COMPONENTS
   AArch64
-  Exegesis
   )
 
-target_link_libraries(LLVMExegesisAArch64 ${libs})
+target_link_libraries(LLVMExegesisAArch64 PRIVATE LLVMExegesis)
 set_target_properties(LLVMExegesisAArch64 PROPERTIES FOLDER "Libraries")
--- origsrc/llvm-8.0.1.src/tools/llvm-exegesis/lib/CMakeLists.txt	2018-11-08 11:51:42.000000000 -0500
+++ src/llvm-8.0.1.src/tools/llvm-exegesis/lib/CMakeLists.txt	2019-07-21 17:14:29.755756100 -0400
@@ -15,7 +15,7 @@ endif()
 
 set(LLVM_EXEGESIS_TARGETS "${LLVM_EXEGESIS_TARGETS} ${TARGETS_TO_APPEND}" PARENT_SCOPE)
 
-add_library(LLVMExegesis
+llvm_add_library(LLVMExegesis
   STATIC
   Analysis.cpp
   Assembler.cpp
@@ -35,7 +35,7 @@ add_library(LLVMExegesis
   )
 
 llvm_update_compile_flags(LLVMExegesis)
-llvm_map_components_to_libnames(libs
+set(LLVM_LINK_COMPONENTS
   Analysis
   CodeGen
   Core
@@ -53,5 +53,4 @@ if(LLVM_ENABLE_LIBPFM AND HAVE_LIBPFM)
   list(APPEND libs pfm)
 endif()
 
-target_link_libraries(LLVMExegesis ${libs})
 set_target_properties(LLVMExegesis PROPERTIES FOLDER "Libraries")
--- origsrc/llvm-8.0.1.src/tools/llvm-exegesis/lib/PowerPC/CMakeLists.txt	2018-11-08 11:51:42.000000000 -0500
+++ src/llvm-8.0.1.src/tools/llvm-exegesis/lib/PowerPC/CMakeLists.txt	2019-07-21 17:13:19.204917200 -0400
@@ -3,16 +3,15 @@ include_directories(
   ${LLVM_BINARY_DIR}/lib/Target/PowerPC
   )
 
-add_library(LLVMExegesisPowerPC
+llvm_add_library(LLVMExegesisPowerPC
   STATIC
   Target.cpp
   )
 
 llvm_update_compile_flags(LLVMExegesisPowerPC)
-llvm_map_components_to_libnames(libs
+set(LLVM_LINK_COMPONENTS
   PowerPC
-  Exegesis
   )
 
-target_link_libraries(LLVMExegesisPowerPC ${libs})
+target_link_libraries(LLVMExegesisPowerPC PRIVATE LLVMExegesis)
 set_target_properties(LLVMExegesisPowerPC PROPERTIES FOLDER "Libraries")
--- origsrc/llvm-8.0.1.src/tools/llvm-exegesis/lib/X86/CMakeLists.txt	2018-09-15 15:04:27.000000000 -0400
+++ src/llvm-8.0.1.src/tools/llvm-exegesis/lib/X86/CMakeLists.txt	2019-07-21 17:13:52.820879300 -0400
@@ -3,16 +3,15 @@ include_directories(
   ${LLVM_BINARY_DIR}/lib/Target/X86
   )
 
-add_library(LLVMExegesisX86
+llvm_add_library(LLVMExegesisX86
   STATIC
   Target.cpp
   )
 
 llvm_update_compile_flags(LLVMExegesisX86)
-llvm_map_components_to_libnames(libs
+set(LLVM_LINK_COMPONENTS
   X86
-  Exegesis
   )
 
-target_link_libraries(LLVMExegesisX86 ${libs})
+target_link_libraries(LLVMExegesisX86 PRIVATE LLVMExegesis)
 set_target_properties(LLVMExegesisX86 PROPERTIES FOLDER "Libraries")
--- origsrc/llvm-8.0.1.src/unittests/Passes/CMakeLists.txt	2019-01-14 13:25:35.000000000 -0500
+++ src/llvm-8.0.1.src/unittests/Passes/CMakeLists.txt	2019-07-21 17:14:56.448277400 -0400
@@ -14,7 +14,9 @@ add_llvm_unittest(PluginsTests
 export_executable_symbols(PluginsTests)
 target_link_libraries(PluginsTests PRIVATE LLVMTestingSupport)
 
-set(LLVM_LINK_COMPONENTS)
+set(LLVM_LINK_COMPONENTS
+  Support
+  )
 add_llvm_library(TestPlugin MODULE BUILDTREE_ONLY
   TestPlugin.cpp
   )
