diff --git a/llvm/cmake/modules/HandleLLVMOptions.cmake b/llvm/cmake/modules/HandleLLVMOptions.cmake
index 1b75ee842b47..f78cd193161c 100644
--- a/llvm/cmake/modules/HandleLLVMOptions.cmake
+++ b/llvm/cmake/modules/HandleLLVMOptions.cmake
@@ -472,7 +472,7 @@ elseif(LLVM_ENABLE_PIC)
 endif()
 
 if((NOT ("${CMAKE_SYSTEM_NAME}" MATCHES "AIX")) AND
-   (NOT (WIN32 OR CYGWIN)))
+   (NOT ((WIN32 OR CYGWIN) AND NOT LLVM_BUILD_LLVM_DYLIB)))
   # GCC for MinGW does nothing about -fvisibility-inlines-hidden, but warns
   # about use of the attributes. As long as we don't use the attributes (to
   # override the default) we shouldn't set the command line options either.
