diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index d681b1ccab62..9520fa174e01 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -855,6 +855,23 @@ else()
   option(LLVM_BUILD_LLVM_DYLIB "Build libllvm dynamic library" ${LLVM_BUILD_LLVM_DYLIB_default})
 endif()
 
+if (CYGWIN OR MINGW)
+	if (NOT BUILD_SHARED_LIBS)
+		set(LLVM_BUILD_LLVM_DYLIB OFF)
+		set(LLVM_LINK_LLVM_DYLIB OFF)
+		# On Cygwin and MinGW, it is important to explicitly set LLVM_ENABLE_PIC to OFF
+		# otherwise, there is link error
+		# /usr/bin/ld: error: export ordinal too large: 101192
+		# caused by commit de07b1e84d8de948304766df602fee2b845e9532
+		set(LLVM_ENABLE_PIC OFF)
+	else()
+		# NOTE: Important! the following 2 lines must be commented,  the 2 variables must have no initial values. Tested many times.
+		# set(LLVM_BUILD_LLVM_DYLIB OFF)
+		# set(LLVM_LINK_LLVM_DYLIB OFF)
+		# set(LLVM_ENABLE_PIC OFF)
+	endif()
+endif()
+
 if (LLVM_LINK_LLVM_DYLIB AND BUILD_SHARED_LIBS)
   message(FATAL_ERROR "Cannot enable BUILD_SHARED_LIBS with LLVM_LINK_LLVM_DYLIB.  We recommend disabling BUILD_SHARED_LIBS.")
 endif()
