diff --git a/llvm/cmake/modules/HandleLLVMOptions.cmake b/llvm/cmake/modules/HandleLLVMOptions.cmake
index bdbd36174fc7..87821669e3bf 100644
--- a/llvm/cmake/modules/HandleLLVMOptions.cmake
+++ b/llvm/cmake/modules/HandleLLVMOptions.cmake
@@ -400,13 +400,13 @@ if( LLVM_USE_LINKER )
   endif()
 endif()
 
-if( LLVM_ENABLE_PIC )
+if(WIN32 OR CYGWIN)
+  # On Windows all code is PIC. MinGW warns if -fPIC is used.
+elseif(LLVM_ENABLE_PIC)
   if( XCODE )
     # Xcode has -mdynamic-no-pic on by default, which overrides -fPIC. I don't
     # know how to disable this, so just force ENABLE_PIC off for now.
     message(WARNING "-fPIC not supported with Xcode.")
-  elseif( WIN32 OR CYGWIN)
-    # On Windows all code is PIC. MinGW warns if -fPIC is used.
   else()
     add_flag_or_print_warning("-fPIC" FPIC)
     # Enable interprocedural optimizations for non-inline functions which would
