diff --git a/llvm/lib/Target/X86/X86ISelDAGToDAG.cpp b/llvm/lib/Target/X86/X86ISelDAGToDAG.cpp
index 77a997588c4f..ef8981f0e7e3 100644
--- a/llvm/lib/Target/X86/X86ISelDAGToDAG.cpp
+++ b/llvm/lib/Target/X86/X86ISelDAGToDAG.cpp
@@ -2944,6 +2944,18 @@ bool X86DAGToDAGISel::selectMOV64Imm32(SDValue N, SDValue &Imm) {
     return false;
 
   Imm = N;
+  if (Subtarget->isTargetWindowsCygwin() && Subtarget->isTargetWin64()) {
+    if (N->getOpcode() != ISD::TargetGlobalAddress)
+      return TM.getCodeModel() == CodeModel::Small;
+
+    std::optional<ConstantRange> CR =
+        cast<GlobalAddressSDNode>(N)->getGlobal()->getAbsoluteSymbolRange();
+    if (!CR)
+      return TM.getCodeModel() == CodeModel::Small;
+
+    return CR->getUnsignedMax().ult(1ull << 32);
+  }
+
   // Small/medium code model can reference non-TargetGlobalAddress objects with
   // 32 bit constants.
   if (N->getOpcode() != ISD::TargetGlobalAddress) {
