diff --git a/llvm/lib/CodeGen/AssignmentTrackingAnalysis.cpp b/llvm/lib/CodeGen/AssignmentTrackingAnalysis.cpp
index eb372655e5f1..0800b6f4d6b7 100644
--- a/llvm/lib/CodeGen/AssignmentTrackingAnalysis.cpp
+++ b/llvm/lib/CodeGen/AssignmentTrackingAnalysis.cpp
@@ -2578,6 +2578,7 @@ DebugAssignmentTrackingAnalysis::run(Function &F,
   return Results;
 }
 
+#if !defined(__CYGWIN__)
 AnalysisKey DebugAssignmentTrackingAnalysis::Key;
 
 PreservedAnalyses
@@ -2586,6 +2587,7 @@ DebugAssignmentTrackingPrinterPass::run(Function &F,
   FAM.getResult<DebugAssignmentTrackingAnalysis>(F).print(OS, F);
   return PreservedAnalyses::all();
 }
+#endif
 
 bool AssignmentTrackingAnalysis::runOnFunction(Function &F) {
   if (!isAssignmentTrackingEnabled(*F.getParent()))
diff --git a/llvm/lib/Passes/PassRegistry.def b/llvm/lib/Passes/PassRegistry.def
index fbaca001d1fe..2026bbbbae8d 100644
--- a/llvm/lib/Passes/PassRegistry.def
+++ b/llvm/lib/Passes/PassRegistry.def
@@ -235,7 +235,9 @@ FUNCTION_ANALYSIS("block-freq", BlockFrequencyAnalysis())
 FUNCTION_ANALYSIS("branch-prob", BranchProbabilityAnalysis())
 FUNCTION_ANALYSIS("cycles", CycleAnalysis())
 FUNCTION_ANALYSIS("da", DependenceAnalysis())
+#if !defined(__CYGWIN__)
 FUNCTION_ANALYSIS("debug-ata", DebugAssignmentTrackingAnalysis())
+#endif
 FUNCTION_ANALYSIS("demanded-bits", DemandedBitsAnalysis())
 FUNCTION_ANALYSIS("domfrontier", DominanceFrontierAnalysis())
 FUNCTION_ANALYSIS("domtree", DominatorTreeAnalysis())
@@ -385,7 +387,9 @@ FUNCTION_PASS("print<branch-prob>", BranchProbabilityPrinterPass(dbgs()))
 FUNCTION_PASS("print<cost-model>", CostModelPrinterPass(dbgs()))
 FUNCTION_PASS("print<cycles>", CycleInfoPrinterPass(dbgs()))
 FUNCTION_PASS("print<da>", DependenceAnalysisPrinterPass(dbgs()))
+#if !defined(__CYGWIN__)
 FUNCTION_PASS("print<debug-ata>", DebugAssignmentTrackingPrinterPass(dbgs()))
+#endif
 FUNCTION_PASS("print<delinearization>", DelinearizationPrinterPass(dbgs()))
 FUNCTION_PASS("print<demanded-bits>", DemandedBitsPrinterPass(dbgs()))
 FUNCTION_PASS("print<domfrontier>", DominanceFrontierPrinterPass(dbgs()))
