diff --git a/clang/tools/driver/CMakeLists.txt b/clang/tools/driver/CMakeLists.txt
index 018605c2fd4f..a9cbf2ad75e4 100644
--- a/clang/tools/driver/CMakeLists.txt
+++ b/clang/tools/driver/CMakeLists.txt
@@ -48,7 +48,7 @@ clang_target_link_libraries(clang
   clangSerialization
   )
 
-if(WIN32 AND NOT CYGWIN)
+if(WIN32 OR CYGWIN)
   # Prevent versioning if the buildhost is targeting for Win32.
 else()
   set_target_properties(clang PROPERTIES VERSION ${CLANG_EXECUTABLE_VERSION})
diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index d681b1ccab62..b3d3501f9c72 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -332,6 +332,9 @@ option(LLVM_INSTALL_CCTOOLS_SYMLINKS
 # If you have the correct setup on Windows you can use this option to enable
 # symlinks and save a lot of diskspace.
 option(LLVM_USE_SYMLINKS "Use symlinks instead of copying binaries" ${CMAKE_HOST_UNIX})
+if(WIN32 OR CYGWIN)
+  set(LLVM_USE_SYMLINKS OFF)
+endif()
 
 option(LLVM_INSTALL_UTILS "Include utility binaries in the 'install' target." OFF)
 
