diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index d681b1ccab62..231a3173b728 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -331,7 +331,7 @@ option(LLVM_INSTALL_CCTOOLS_SYMLINKS
 # By default we use symlinks on Unix platforms and copy binaries on Windows
 # If you have the correct setup on Windows you can use this option to enable
 # symlinks and save a lot of diskspace.
-option(LLVM_USE_SYMLINKS "Use symlinks instead of copying binaries" ${CMAKE_HOST_UNIX})
+option(LLVM_USE_SYMLINKS "Use symlinks instead of copying binaries" OFF)
 
 option(LLVM_INSTALL_UTILS "Include utility binaries in the 'install' target." OFF)
 
