diff --git a/lld/MinGW/Driver.cpp b/lld/MinGW/Driver.cpp
index c7d7b9cfca38..30756880af46 100644
--- a/lld/MinGW/Driver.cpp
+++ b/lld/MinGW/Driver.cpp
@@ -529,6 +529,16 @@ bool link(ArrayRef<const char *> argsArr, llvm::raw_ostream &stdoutOS,
     add("-libpath:" + StringRef(a->getValue()));
   }
 
+#ifdef __CYGWIN__
+	std::vector<std::string> cygwin_lib_search_paths{"/lib", "/lib/w32api"};
+	size_t cygwin_lib_search_paths_length = cygwin_lib_search_paths.size();
+	for (size_t i = 0; i < cygwin_lib_search_paths_length; i++) {
+		const std::string& path = cygwin_lib_search_paths[i];
+		searchPaths.push_back(path);
+		add("-libpath:" + StringRef(path));
+	}
+#endif
+
   StringRef prefix = "";
   bool isStatic = false;
   for (auto *a : args) {
