diff --git a/clang/lib/CodeGen/CGCall.h b/clang/lib/CodeGen/CGCall.h
index eaaf10c4eec6..9922612c6bad 100644
--- a/clang/lib/CodeGen/CGCall.h
+++ b/clang/lib/CodeGen/CGCall.h
@@ -254,7 +254,7 @@ public:
 
 /// CallArgList - Type for representing both the value and type of
 /// arguments in a call.
-class CallArgList : public SmallVector<CallArg, 8> {
+class CallArgList : public SmallVector<CallArg, 2> {
 public:
   CallArgList() : StackBase(nullptr) {}
 
