diff --git a/clang/lib/Basic/Version.cpp b/clang/lib/Basic/Version.cpp
index 4823f566bd77..4d45f9ab97b2 100644
--- a/clang/lib/Basic/Version.cpp
+++ b/clang/lib/Basic/Version.cpp
@@ -17,7 +17,7 @@
 #include <cstdlib>
 #include <cstring>
 
-#include "VCSVersion.inc"
+#include "llvm/Support/VCSRevision.h"
 
 namespace clang {
 
@@ -25,8 +25,8 @@ std::string getClangRepositoryPath() {
 #if defined(CLANG_REPOSITORY_STRING)
   return CLANG_REPOSITORY_STRING;
 #else
-#ifdef CLANG_REPOSITORY
-  return CLANG_REPOSITORY;
+#ifdef LLVM_REPOSITORY
+  return LLVM_REPOSITORY;
 #else
   return "";
 #endif
@@ -42,8 +42,8 @@ std::string getLLVMRepositoryPath() {
 }
 
 std::string getClangRevision() {
-#ifdef CLANG_REVISION
-  return CLANG_REVISION;
+#ifdef LLVM_REVISION
+  return LLVM_REVISION;
 #else
   return "";
 #endif
