diff --git a/clang/lib/Driver/ToolChains/MinGW.cpp b/clang/lib/Driver/ToolChains/MinGW.cpp
index c81a7ed17029..41ed9df2448e 100644
--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -167,7 +167,16 @@ void tools::MinGW::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     else
       CmdArgs.push_back("DllMainCRTStartup");
     CmdArgs.push_back("--enable-auto-image-base");
+  } else {
+    if (Args.hasArg(options::OPT_no_pie)) {
+      CmdArgs.push_back("--disable-dynamicbase");
+    }
+    if (TC.getArch() == llvm::Triple::x86) {
+		CmdArgs.push_back("--large-address-aware");
+	}
   }
+  CmdArgs.push_back("--allow-multiple-definition");
+  CmdArgs.push_back("--dll-search-prefix=lib");
 
   if (Args.hasArg(options::OPT_Z_Xlinker__no_demangle))
     CmdArgs.push_back("--no-demangle");
