diff --git a/lldb/source/API/CMakeLists.txt b/lldb/source/API/CMakeLists.txt
index f3f1168f6095..97eca809ec48 100644
--- a/lldb/source/API/CMakeLists.txt
+++ b/lldb/source/API/CMakeLists.txt
@@ -325,9 +325,11 @@ foreach(header
   list(APPEND lldb_staged_headers ${staged_header})
 endforeach()
 
+if(NOT CYGWIN)
 add_custom_command(TARGET liblldb POST_BUILD
   COMMAND ${LLDB_SOURCE_DIR}/scripts/version-header-fix.py ${LLDB_SOURCE_DIR}/include/lldb/lldb-defines.h ${lldb_header_staging_dir}/lldb-defines.h ${LLDB_VERSION_MAJOR} ${LLDB_VERSION_MINOR} ${LLDB_VERSION_PATCH}
 )
+endif()
 add_custom_target(liblldb-header-staging DEPENDS ${lldb_staged_headers})
 add_dependencies(liblldb liblldb-header-staging)
 
