diff --git a/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp b/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp
index ea74a500518e..fd85cf0a5c70 100644
--- a/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp
+++ b/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp
@@ -360,7 +360,7 @@ Status ScriptInterpreterLua::RegisterBreakpointCallback(
     StructuredData::ObjectSP extra_args_sp) {
   Status error;
   auto data_up = std::make_unique<CommandDataLua>(extra_args_sp);
-  error = m_lua->RegisterBreakpointCallback(data_up.get(), command_body_text);
+  error = Status::FromError(m_lua->RegisterBreakpointCallback(data_up.get(), command_body_text));
   if (error.Fail())
     return error;
   auto baton_sp =
@@ -381,7 +381,7 @@ Status ScriptInterpreterLua::RegisterWatchpointCallback(
     StructuredData::ObjectSP extra_args_sp) {
   Status error;
   auto data_up = std::make_unique<WatchpointOptions::CommandData>();
-  error = m_lua->RegisterWatchpointCallback(data_up.get(), command_body_text);
+  error = Status::FromError(m_lua->RegisterWatchpointCallback(data_up.get(), command_body_text));
   if (error.Fail())
     return error;
   auto baton_sp =
